--- src/BruteX/brutex	2015-10-01 15:35:40.429063305 +0200
+++ brutex	2015-10-01 15:36:12.168891356 +0200
@@ -14,13 +14,13 @@
 # 3. SNMPWalk
 # 4. DNSEnum
 #
-# HYDRA SERVICES: 
+# HYDRA SERVICES:
 # asterisk cisco cisco-enable cvs ftp ftps http[s]-{head|get} http[s]-{get|post}-form http-proxy http-proxy-urlenum icq imap[s] irc ldap2[s] ldap3[-{cram|digest}md5][s] mssql mysql(v4) nntp oracle-listener oracle-sid pcanywhere pcnfs pop3[s] postgres rdp redis rexec rlogin rsh rtsp s7-300 sip smb smtp[s] smtp-enum snmp socks5 ssh sshkey teamspeak telnet[s] vmauthd vnc xmpp
 
 
 TARGET="$1"
 PORT="$2"
-LOOT_DIR=loot
+LOOT_DIR=/tmp/loot
 USER_FILE="simple-users.txt"
 PASS_FILE="password.lst"
 DNS_FILE="namelist.txt"
@@ -64,8 +64,13 @@
 echo ""
 echo ""
 
+if [ ! -d "${LOOT_DIR}" ]
+then
+  mkdir "${LOOT_DIR}"
+fi
+
 echo -e "$COLOR3################################### Running Port Scan ##############################$RESET"
-if [ -z "$PORT" ] 
+if [ -z "$PORT" ]
 then
 	nmap -T4 --open $TARGET -oX $LOOT_DIR/nmap-$TARGET.xml
 else
@@ -103,7 +108,7 @@
 then
 	echo -e "$COLOR1 + -- --=[Port 21 closed... skipping.$RESET"
 else
-	echo -e "$COLOR2 + -- --=[Port 21 opened... running tests...$RESET"	
+	echo -e "$COLOR2 + -- --=[Port 21 opened... running tests...$RESET"
 	hydra -L $USER_FILE -P $PASS_FILE $TARGET ftp -t $THREADS -e ns
 fi
 
@@ -144,7 +149,7 @@
 then
 	echo -e "$COLOR1 + -- --=[Port 110 closed... skipping.$RESET"
 else
-	echo -e "$COLOR2 + -- --=[Port 110 opened... running tests...$RESET"	
+	echo -e "$COLOR2 + -- --=[Port 110 opened... running tests...$RESET"
 	hydra -L $USER_FILE -P $PASS_FILE $TARGET pop3 -t $THREADS -e ns
 fi
 
@@ -170,7 +175,7 @@
 	echo -e "$COLOR1 + -- --=[Port 443 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 443 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET https-get -s 443 -t $THREADS -f -q -e ns -m $HTTP_LOCATION 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET https-get -s 443 -t $THREADS -f -q -e ns -m $HTTP_LOCATION
 fi
 
 if [ -z "$port_445" ]
@@ -178,7 +183,7 @@
 	echo -e "$COLOR1 + -- --=[Port 445 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 445 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET smb -S 445 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET smb -S 445 -t $THREADS -e ns
 fi
 
 if [ -z "$port_512" ]
@@ -186,7 +191,7 @@
 	echo -e "$COLOR1 + -- --=[Port 512 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 512 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET rexec -S 512 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET rexec -S 512 -t $THREADS -e ns
 fi
 
 if [ -z "$port_513" ]
@@ -194,7 +199,7 @@
 	echo -e "$COLOR1 + -- --=[Port 513 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 513 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET rlogin -S 513 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET rlogin -S 513 -t $THREADS -e ns
 fi
 
 if [ -z "$port_514" ]
@@ -202,7 +207,7 @@
 	echo -e "$COLOR1 + -- --=[Port 514 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 514 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET rsh -S 514 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET rsh -S 514 -t $THREADS -e ns
 fi
 
 if [ -z "$port_993" ]
@@ -210,7 +215,7 @@
 	echo -e "$COLOR1 + -- --=[Port 993 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 993 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET imap -S 993 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET imap -S 993 -t $THREADS -e ns
 fi
 
 if [ -z "$port_1433" ]
@@ -218,7 +223,7 @@
 	echo -e "$COLOR1 + -- --=[Port 1433 closed... skipping.$RESET"
 else
 	echo -e "$COLOR2 + -- --=[Port 1433 opened... running tests...$RESET"
-	hydra -L $USER_FILE -P $PASS_FILE $TARGET mssql -S 1433 -t $THREADS -e ns 
+	hydra -L $USER_FILE -P $PASS_FILE $TARGET mssql -S 1433 -t $THREADS -e ns
 fi
 
 if [ -z "$port_3306" ]
