From b7e0def45a584e95a22c5804d60ace76b756d5ec Mon Sep 17 00:00:00 2001
From: Roy Hills <royhills@hotmail.com>
Date: Wed, 14 Sep 2022 15:10:43 +0000
Subject: [PATCH] Use printf %ld format for timeval tv_usec field

---
 ike-scan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ike-scan.c b/ike-scan.c
index 29f510c..b14c82a 100644
--- a/ike-scan.c
+++ b/ike-scan.c
@@ -1511,7 +1511,7 @@ display_packet(int n, unsigned char *packet_in, host_entry *he,
       clock_seconds = time_tv.tv_sec;
       time_tm = localtime(&clock_seconds);
       cp = msg;
-      msg = make_message("%s%02d:%02d:%02d.%06u ", cp,
+      msg = make_message("%s%02d:%02d:%02d.%06ld ", cp,
                          time_tm->tm_hour, time_tm->tm_min, time_tm->tm_sec,
                          time_tv.tv_usec);
       free(cp);
-- 
2.43.0

