From 9949ce4bdf9f4bcb616b2a5d273708a7ea9ee93d Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Mon, 26 Sep 2022 05:22:18 +0100
Subject: [PATCH] acinclude.m4: fix -Wimplicit-int errors (Clang 16+ compat)

---
 acinclude.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index ec67906..cc21ff6 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -144,7 +144,7 @@ int does_int64_work()
     return 0;
   return 1;
 }
-main() {
+int main() {
   return ! does_int64_work();
 }],
 [Ac_cachevar=yes],
@@ -200,7 +200,7 @@ int does_int64_snprintf_work()
     return 0;			/* either multiply or snprintf is busted */
   return 1;
 }
-main() {
+int main() {
   return ! does_int64_snprintf_work();
 }],
 [pgac_cv_snprintf_long_long_int_format=$pgac_format; break],
-- 
2.43.0

