diff -rupN driftnet-0.1.6/display.c driftnet-0.1.6.new/display.c
--- driftnet-0.1.6/display.c	2002-07-09 20:26:41.000000000 +0100
+++ driftnet-0.1.6.new/display.c	2014-05-30 18:10:21.540001050 +0100
@@ -11,9 +11,9 @@
 
 static const char rcsid[] = "$Id: display.c,v 1.15 2002/06/13 20:06:42 chris Exp $";
 
-#include <gtk/gtk.h>
-#include <gdk/gdk.h>
-#include <gdk/gdkx.h>
+#include <gtk-2.0/gtk/gtk.h>
+#include <gtk-2.0/gdk/gdk.h>
+#include <gtk-2.0/gdk/gdkx.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
diff -rupN driftnet-0.1.6/Makefile driftnet-0.1.6.new/Makefile
--- driftnet-0.1.6/Makefile	2014-05-30 18:18:17.960001256 +0100
+++ driftnet-0.1.6.new/Makefile	2014-05-30 18:18:24.506667926 +0100
@@ -21,7 +21,7 @@ LDFLAGS += -g
 
 # You might need these if libpcap is installed somewhere random.
 CFLAGS += -I/usr/include/pcap
-#LDFLAGS += -L/path/to/libpcap.so
+LDFLAGS = /usr/lib/libgif.so
 
 # Required on Linux to get BSDish definitions of the TCP/IP structs.
 CFLAGS += -D_BSD_SOURCE
@@ -31,8 +31,8 @@ LDLIBS += -lpcap -lpthread
 
 # Optional C compiler and linker flags. Typical driftnet builds have support
 # for displaying captured images in an X window, and need the following flags:
-CFLAGS  += `gtk-config --cflags`
-LDLIBS  += -ljpeg  `gtk-config --libs`
+CFLAGS  += `pkg-config --cflags gtk+-2.0`
+LDLIBS  += -ljpeg  `pkg-config --libs gtk+-2.0`
 
 # Alternatively, you can build a version of driftnet which can only be used
 # in `adjunct' mode as the back end for some other image-processing program. To
@@ -103,7 +103,7 @@ tarball: nodepend $(SRCS) $(HDRS) $(TXTS
 	mv driftnet-$(VERSION).tar.gz ..
 	
 depend: endianness
-	makedepend -- $(CFLAGS) `cat endianness` -- $(SRCS)
+	gccmakedep -- $(CFLAGS) `cat endianness` -- $(SRCS)
 	touch depend
 	rm -f Makefile.bak
 
