--- gif.c	2002-07-10 04:26:41.000000000 +0900
+++ gif.c.new	2014-06-28 04:30:10.260406806 +0900
@@ -20,7 +20,8 @@
  */
 int gif_load_hdr(img I) {
     GifFileType *g;
-    g = I->us = DGifOpenFileHandle(fileno(I->fp));
+    int e;
+    g = I->us = DGifOpenFileHandle(fileno(I->fp), &e);
     if (!I->us) {
         I->err = IE_HDRFORMAT;
         return 0;
@@ -36,7 +37,8 @@
  * Abort loading a GIF file after the header is done.
  */
 int gif_abort_load(img I) {
-    DGifCloseFile((GifFileType*)I->us);
+    int e;
+    DGifCloseFile((GifFileType*)I->us, &e);
     return 1;
 }
 
@@ -47,6 +49,7 @@
     GifFileType *g = I->us;
     struct SavedImage *si;
     int ret = 0;
+    int e;
     unsigned char *p, *end;
     GifColorType *pal;
     pel *q;
@@ -114,7 +117,7 @@
     ret = 1;
 fail:
 
-    DGifCloseFile(g);
+    DGifCloseFile(g, &e);
     
     return ret;
 }
