diff -Naur driftnet-upstream-1.1.5.orig/src/display/gif.c driftnet-upstream-1.1.5/src/display/gif.c
--- driftnet-upstream-1.1.5.orig/src/display/gif.c	2014-10-28 11:28:46.084689763 -0400
+++ driftnet-upstream-1.1.5/src/display/gif.c	2014-10-28 11:32:07.601599279 -0400
@@ -32,6 +32,13 @@
     g = I->us = DGifOpenFileHandle(fileno(I->fp));
 #endif
 
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+# define GIF_CLOSE_FILE(g)  DGifCloseFile(g, NULL);
+#else
+# define GIF_CLOSE_FILE(g)  DGifCloseFile(g);
+#endif
+
+
     if (!I->us) {
         I->err = IE_HDRFORMAT;
         return 0;
@@ -47,7 +54,11 @@
  * Abort loading a GIF file after the header is done.
  */
 int gif_abort_load(img I) {
-    DGifCloseFile((GifFileType*)I->us);
+#if (GIFLIB_MAJOR > 5) || ((GIFLIB_MAJOR == 5) && (GIFLIB_MINOR >= 1))
+    DGifCloseFile(((GifFileType*)I->us), NULL);
+#else
+    DGifCloseFile(((GifFileType*)I->us);
+#endif
     return 1;
 }
 
@@ -125,7 +136,7 @@
     ret = 1;
 fail:
 
-    DGifCloseFile(g);
+    GIF_CLOSE_FILE(g);
 
     return ret;
 }
