diff -urN a/arch-nspawn.in b/arch-nspawn.in
--- a/arch-nspawn.in	2013-05-25 15:39:46.000000000 -0500
+++ b/arch-nspawn.in	2013-06-04 16:22:20.718826563 -0500
@@ -74,8 +74,8 @@
 }
 
 copy_hostconf () {
-	cp -a /etc/pacman.d/gnupg "$working_dir/etc/pacman.d"
-	echo "Server = $host_mirror" > $working_dir/etc/pacman.d/mirrorlist
+	#cp -a /etc/pacman.d/gnupg "$working_dir/etc/pacman.d"
+	#echo "Server = $host_mirror" > $working_dir/etc/pacman.d/mirrorlist
 
 	[[ -n $pac_conf ]] && cp $pac_conf $working_dir/etc/pacman.conf
 	[[ -n $makepkg_conf ]] && cp $makepkg_conf $working_dir/etc/makepkg.conf
