diff -urN a/makechrootpkg.in b/makechrootpkg.in
--- a/makechrootpkg.in	2013-06-04 16:19:08.453803346 -0500
+++ b/makechrootpkg.in	2013-06-04 16:29:16.961006377 -0500
@@ -306,7 +306,9 @@
 		exit 1
 	fi
 
-	sudo -u nobody makepkg $makepkg_args || exit 1
+	mkdir /build/.distcc
+	chown nobody /build/.distcc
+	sudo -u nobody DISTCC_IO_TIMEOUT=0 DISTCC_DIR='/build/.distcc' makepkg $makepkg_args || exit 1
 
 	if $run_namcap; then
 		pacman -S --needed --noconfirm namcap
