diff -urN a/arch-nspawn.in b/arch-nspawn.in
--- a/arch-nspawn.in	2013-06-04 16:22:20.718826563 -0500
+++ b/arch-nspawn.in	2013-06-04 17:42:52.282698873 -0500
@@ -98,7 +98,7 @@
 
 eval $(grep '^CARCH=' "$working_dir/etc/makepkg.conf")
 
-exec ${CARCH:+setarch "$CARCH"} systemd-nspawn 2>/dev/null \
+exec systemd-nspawn 2>/dev/null \
 	-D "$working_dir" \
 	--machine "${working_dir//\//-}" \
 	"${mount_args[@]}" \
diff -urN a/makechrootpkg.in b/makechrootpkg.in
--- a/makechrootpkg.in	2013-06-04 17:37:51.646420764 -0500
+++ b/makechrootpkg.in	2013-06-04 17:50:34.673206856 -0500
@@ -256,10 +256,10 @@
 	# Ensure sources are downloaded
 	if [[ -n $SUDO_USER ]]; then
 		sudo -u $SUDO_USER env SRCDEST="$SRCDEST" BUILDDIR="$builddir" \
-			makepkg --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --config="$copydir/etc/makepkg.conf" --verifysource -o
 	else
 		( export SRCDEST BUILDDIR="$builddir"
-			makepkg --asroot --config="$copydir/etc/makepkg.conf" --verifysource -o
+			makepkg -A --asroot --config="$copydir/etc/makepkg.conf" --verifysource -o
 		)
 	fi
 	(( $? != 0 )) && die "Could not download sources."
