diff -urN a/arch-nspawn.in b/arch-nspawn.in
--- a/arch-nspawn.in	2013-08-20 13:23:41.830745058 -0600
+++ b/arch-nspawn.in	2013-08-20 13:24:29.500531210 -0600
@@ -98,8 +98,11 @@
 
 eval $(grep '^CARCH=' "$working_dir/etc/makepkg.conf")
 
+machine_name="${working_dir//\//-}"
+machine_name="${machine_name#-}"
+
 exec systemd-nspawn 2>/dev/null \
 	-D "$working_dir" \
-	--machine "${working_dir//\//-}" \
+	--machine "$machine_name" \
 	"${mount_args[@]}" \
 	"$@"
