diff --git a/modules/screencapture.py b/modules/screencapture.py
index 8e14d5c..e6380db 100755
--- a/modules/screencapture.py
+++ b/modules/screencapture.py
@@ -11,13 +11,14 @@ import os
 # External Libraries
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.chrome.service import Service
 
 # Selenium Configuration
 chrome_opt = Options()
 chrome_opt.add_argument("--ignore-certificate-errors")
 chrome_opt.add_argument("--test-type")
 chrome_opt.add_argument("--headless")  # Comment out to Debug
-DRIVER = "chromedriver"
+DRIVER = "/usr/bin/chromedriver"
 
 
 def main(hostx, out_path):
@@ -37,7 +38,8 @@ def main(hostx, out_path):
 
 
 def take_screenshot(target, port, path):
-    driver = webdriver.Chrome(executable_path=DRIVER, options=chrome_opt)
+    service = Service(executable_path=DRIVER)
+    driver = webdriver.Chrome(service=service, options=chrome_opt)
     driver.set_page_load_timeout(9)
 
     if port == "443":
