--- pytacle-alpha2/pytacle.py	2013-10-30 17:18:38.000000000 +0000
+++ /usr/bin/pytacle	2013-12-10 10:41:58.937386275 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 #
 #       pytacle.py
@@ -41,6 +41,8 @@
 import gtk
 gtk.gdk.threads_init()
 
+import sys
+sys.path.append("/usr/lib/python2.7/pytacle/")
 from mcc_mnc import codes
 
 VERSION = "v0.2"
@@ -66,7 +68,7 @@
                     'on_scan_togglebutton_toggled'  : self.on_scan_togglebutton_toggled,
                     }
         builder = gtk.Builder()
-        builder.add_from_file("windows.glade")
+        builder.add_from_file("/usr/lib/python2.7/pytacle/windows.glade")
         builder.connect_signals(signals)
         self.main_window = builder.get_object("main_window")
         self.main_window.set_title(self.__class__.__name__)
@@ -530,7 +532,7 @@
                     stream = self.xor(clburst, ciburst)
                     try:
                         key = self.send_to_kraken(stream)
-                    except Exception, e:
+                    except Exception as e:
                         self.log_t("Error sending to kraken: %s\n" % str(e), "error")
                     if key != None:
                         self.log_t("Found key '%s' for fn %s\n" % (key, cifn), "info")
@@ -753,12 +755,12 @@
     signal.signal(signal.SIGINT, app.destroy_event)
     try:
         app.main()
-    except Exception, e:
-        print e
+    except Exception as e:
+        print(e)
         if DEBUG:
-            print '-'*60
+            print('-'*60)
             traceback.print_exc(file=sys.stdout)
-            print '-'*60
+            print('-'*60)
 
 if __name__ == '__main__':
     main()
