diff --git a/epan/Makefile.am b/epan/Makefile.am
index b2a569d..717e233 100644
--- a/epan/Makefile.am
+++ b/epan/Makefile.am
@@ -130,6 +130,7 @@ libwireshark_la_LIBADD = \
 	dissectors/libdirtydissectors.la $(wslua_lib) @SOCKET_LIBS@ @NSL_LIBS@ \
 	@C_ARES_LIBS@ @ADNS_LIBS@ @LIBGCRYPT_LIBS@ @LIBGNUTLS_LIBS@ \
 	@KRB5_LIBS@ @SSL_LIBS@ @LIBSMI_LDFLAGS@ @GEOIP_LIBS@ \
+	${top_builddir}/wiretap/libwiretap.la \
 	${top_builddir}/wsutil/libwsutil.la -lm
 
 libwireshark_la_DEPENDENCIES = \
@@ -137,6 +138,7 @@ libwireshark_la_DEPENDENCIES = \
 	libwireshark_asmopt.la crypt/libairpdcap.la ftypes/libftypes.la \
 	dfilter/libdfilter.la dissectors/libdissectors.la \
 	dissectors/libdirtydissectors.la $(wslua_lib) \
+	${top_builddir}/wiretap/libwiretap.la \
 	${top_builddir}/wsutil/libwsutil.la
 
 #EXTRA_PROGRAMS = reassemble_test
