--- src/dublin-traceroute/CMakeLists.txt	2019-11-19 14:33:04.876664047 +0100
+++ CMakeLists.txt	2019-11-19 14:43:02.656664044 +0100
@@ -69,33 +69,6 @@
     )
 install(DIRECTORY include/dublintraceroute DESTINATION include)
 
-# find setcap
-find_program(SETCAP_EXECUTABLE
-    NAMES
-        setcap
-    PATHS
-        /bin
-        /usr/bin
-        /usr/local/bin
-        /sbin
-)
-
-if (SETCAP_EXECUTABLE)
-    install(CODE "execute_process(
-        COMMAND
-            ${SETCAP_EXECUTABLE}
-            cap_net_raw+ep
-            ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/dublin-traceroute
-        RESULT_VARIABLE
-            SETCAP_RESULT
-        )
-        if (SETCAP_RESULT)
-            message(WARNING \"setcap failed (${SETCAP_RESULT})\")
-        endif()"
-    )
-endif()
-
-
 # Testing
 INCLUDE(ExternalProject)
 # Only include googletest if the git submodule has been fetched
