--- src/kube-hunter/setup.cfg	2020-05-06 01:17:12.879998633 +0200
+++ setup.cfg	2020-05-06 01:17:35.149998631 +0200
@@ -30,14 +30,13 @@
 install_requires =
     netaddr
     netifaces
-    scapy>=2.4.3
+    scapy
     requests
     PrettyTable
-    urllib3>=1.24.3
+    urllib3
     ruamel.yaml
     future
     packaging
-    dataclasses
 setup_requires =
     setuptools>=30.3.0
     setuptools_scm
