diff -Nru a/sdk/C/common/common.mk b/sdk/C/common/common.mk
--- a/sdk/C/common/common.mk	2010-03-03 14:14:46.000000000 +0100
+++ b/sdk/C/common/common.mk	2010-06-16 18:50:21.126984135 +0200
@@ -160,12 +160,12 @@
 	BINSUBDIR   := debug
 	LIBSUFFIX   := D
 else 
-	COMMONFLAGS += -O2 
+	COMMONFLAGS +=
 	BINSUBDIR   := release
 	LIBSUFFIX   := 
-	NVCCFLAGS   += --compiler-options -fno-strict-aliasing
-	CXXFLAGS    += -fno-strict-aliasing
-	CFLAGS      += -fno-strict-aliasing
+	NVCCFLAGS   += --compiler-options -fno-strict-aliasing --compiler-options -fno-inline -Xcompiler -D__builtin_stdarg_start=__builtin_va_start
+	CXXFLAGS    += -O2 -fno-strict-aliasing -fno-inline
+	CFLAGS      += -O2 -fno-strict-aliasing -fno-inline
 endif
 
 # architecture flag for cubin build
