diff -ur util/genmsgidx util/genmsgidx
--- util/genmsgidx	2008-11-27 23:34:21.000000000 +0100
+++ util/genmsgidx	2010-09-01 12:43:22.000000000 +0200
@@ -96,10 +96,10 @@
 haveallof variables BASEDIR || exit 1
 
 # Clear idx file and detect error
-> $BASEDIR/doc/explain.idx && {
-    echo "Error: Cannot write over the index file $BASEDIR/doc/explain.idx. Aborting"
-    exit 1
-}
+#> $BASEDIR/doc/explain.idx && {
+#    echo "Error: Cannot write over the index file $BASEDIR/doc/explain.idx. Aborting"
+#    exit 1
+#}
 
 $LS $BASEDIR/doc/*.txt |
 while read infile
