--- Makefile.in.orig	2008-09-09 04:06:55.000000000 +0200
+++ Makefile.in	2008-09-15 00:27:55.000000000 +0200
@@ -227,5 +227,6 @@
 	@echo "Copying manpages..."
-	$(INSTALL) -m 0444 man/tiger.8 $(DESTDIR)$(TIGERMANDIR)/man8
-	$(INSTALL) -m 0444 man/tigexp.8 $(DESTDIR)$(TIGERMANDIR)/man8
-	$(INSTALL) -m 0444 man/tigercron.8 $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -d $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -m 0644 man/tiger.8 $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -m 0644 man/tigexp.8 $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -m 0644 man/tigercron.8 $(DESTDIR)$(TIGERMANDIR)/man8
 
@@ -246,3 +247,3 @@
 	@for file in $(CONFIGFILES); do \
-	  $(INSTALL) -m 0640 $$file $(DESTDIR)$(TIGERCONFIG)/$$file; \
+	  $(INSTALL) -D -m 0640 $$file $(DESTDIR)$(TIGERCONFIG)/$$file; \
 	done
