From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 22 Jun 2024 22:08:43 +0200
Subject: [PATCH] Lower severity of XID collision warnings

---
 gdk/x11/gdkxid.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/x11/gdkxid.c b/gdk/x11/gdkxid.c
index 1005f9e40c0d..1523fa70b2d3 100644
--- a/gdk/x11/gdkxid.c
+++ b/gdk/x11/gdkxid.c
@@ -58,7 +58,7 @@ _gdk_xid_table_insert (GdkDisplay *display,
 					    (GEqualFunc) gdk_xid_equal);
 
   if (g_hash_table_lookup (display_x11->xid_ht, xid))
-    g_warning ("XID collision, trouble ahead");
+    g_debug ("XID collision, trouble ahead");
 
   g_hash_table_insert (display_x11->xid_ht, xid, data);
 }
