--- src/afl-0.44b/Makefile	2014-11-12 21:37:33.000000000 +0100
+++ Makefile	2014-11-13 00:14:35.539408584 +0100
@@ -3,13 +3,13 @@
 # ---------------------------------
 #
 # Written and maintained by Michal Zalewski <lcamtuf@google.com>
-# 
+#
 # Copyright 2013, 2014 Google Inc. All rights reserved.
-# 
+#
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at:
-# 
+#
 #   http://www.apache.org/licenses/LICENSE-2.0
 #
 
@@ -18,6 +18,7 @@
 
 BIN_PATH    = /usr/local/bin
 HELPER_PATH = /usr/local/lib/afl
+DESTDIR= ""
 
 PROGS       = afl-gcc afl-as afl-fuzz afl-showmap
 
@@ -97,9 +98,9 @@
 	rm -rf out_dir
 
 install: all
-	install afl-gcc afl-g++ afl-fuzz afl-showmap $(BIN_PATH)
-	mkdir -p -m 755 $(HELPER_PATH) 2>/dev/null
-	install afl-as as $(HELPER_PATH)
+	install afl-gcc afl-g++ afl-fuzz afl-showmap $(DESTDIR)$(BIN_PATH)
+	mkdir -p -m 755 $(DESTDIR)$(HELPER_PATH) 2>/dev/null
+	install afl-as as $(DESTDIR)$(HELPER_PATH)
 
 publish: clean
 	test "`basename $$PWD`" = "afl" || exit 1
