--- src/joomlavs/Gemfile	2019-07-24 16:54:53.576666679 +0200
+++ Gemfile	2019-07-24 16:54:38.586666679 +0200
@@ -2,17 +2,7 @@
 
 source 'https://rubygems.org'
 
-group :cli do
-  gem 'colorize', '>=0.8.1'
-  gem 'slop', '~>4.6.2'
-end
-
-group :http do
-  gem 'nokogiri', '~>1.8.4'
-  gem 'typhoeus', '~>1.3.0'
-end
-
-group :test do
-  gem 'coveralls', require: false
-  gem 'rspec', '~>3.7'
-end
+gem 'colorize'
+gem 'slop'
+gem 'nokogiri'
+gem 'typhoeus'
