--- src/frida-tools-9.0.0/setup.py	2020-10-22 01:03:18.000000000 +0200
+++ setup.py	2020-11-07 13:13:09.530000066 +0100
@@ -20,10 +20,10 @@
     author_email="oleavr@frida.re",
     url="https://frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 14.0.0, < 15.0.0",
-        "prompt-toolkit >= 2.0.0, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
