--- src/frida-tools-6.0.1/setup.py	2020-01-29 18:00:25.000000000 +0100
+++ setup.py	2020-01-30 17:48:56.126666370 +0100
@@ -12,10 +12,10 @@
     author_email="oleavr@frida.re",
     url="https://www.frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 12.8.5, < 13.0.0",
-        "prompt-toolkit >= 3.0.3, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
