--- src/frida-tools-10.3.0/setup.py	2021-09-23 11:29:54.000000000 +0200
+++ setup.py	2021-09-26 14:36:02.743332128 +0200
@@ -20,10 +20,10 @@
     author_email="oleavr@frida.re",
     url="https://frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 15.0.0, < 16.0.0",
-        "prompt-toolkit >= 2.0.0, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
