--- src/frida-tools-7.2.0/setup.py	2020-03-24 06:15:32.000000000 +0100
+++ setup.py	2020-03-25 14:56:46.176666520 +0100
@@ -12,10 +12,10 @@
     author_email="oleavr@frida.re",
     url="https://www.frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 12.8.12, < 13.0.0",
-        "prompt-toolkit >= 3.0.3, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
