--- src/frida-tools-9.1.0/setup.py	2020-12-19 04:39:45.000000000 +0100
+++ setup.py	2020-12-20 05:25:46.873332193 +0100
@@ -20,10 +20,10 @@
     author_email="oleavr@frida.re",
     url="https://frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 14.2.0, < 15.0.0",
-        "prompt-toolkit >= 2.0.0, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
