--- src/frida-tools-9.2.5/setup.py	2021-05-28 01:24:56.000000000 +0200
+++ setup.py	2021-06-21 12:51:05.193333224 +0200
@@ -20,10 +20,10 @@
     author_email="oleavr@frida.re",
     url="https://frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 14.2.9, < 15.0.0",
-        "prompt-toolkit >= 2.0.0, < 4.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
