--- src/frida-tools-5.4.0/setup.py	2020-01-02 21:29:31.000000000 +0100
+++ setup.py	2020-01-03 16:26:01.429999821 +0100
@@ -12,10 +12,10 @@
     author_email="oleavr@frida.re",
     url="https://www.frida.re",
     install_requires=[
-        "colorama >= 0.2.7, < 1.0.0",
-        "frida >= 12.7.3, < 13.0.0",
-        "prompt-toolkit >= 2.0.0, < 3.0.0",
-        "pygments >= 2.0.2, < 3.0.0"
+        "colorama",
+        "frida",
+        "prompt-toolkit",
+        "pygments"
     ],
     license="wxWindows Library Licence, Version 3.1",
     zip_safe=True,
