--- src/graphillion-2.1/setup.py	2025-04-04 13:14:38.000000000 +0200
+++ setup.py	2025-08-03 18:21:53.117547347 +0200
@@ -110,12 +110,12 @@
 # We add this option to suppress warning when compiling bddc.c
 # in SAPPOROBDD library. It is no problem because
 # the variables that the compiler warns are actually used.
-if platform.system() == 'Darwin': # macOS
-    extra_compile_args_list.append('-std=c++11')
-elif platform.system() == 'Windows':
-    pass
-else:
-    extra_compile_args_list.append('-Wno-maybe-uninitialized')
+#if platform.system() == 'Darwin': # macOS
+#    extra_compile_args_list.append('-std=c++11')
+#elif platform.system() == 'Windows':
+#    pass
+#else:
+#    extra_compile_args_list.append('-Wno-maybe-uninitialized')
 
 #extra_compile_args_list.append('-UNDEBUG')
 
