--- src/graphillion-1.9/setup.py	2024-10-04 13:13:07.000000000 +0200
+++ setup.py	2024-12-27 22:41:34.336667781 +0100
@@ -106,10 +106,11 @@
 # We add this option to suppress warning when compiling bddc.c
 # in SAPPOROBDD library. It is no problem because
 # the variables that the compiler warns are actually used.
-if os.uname().sysname == 'Darwin': # macOS
-    extra_compile_args_list.append('-std=c++11')
-else:
-    extra_compile_args_list.append('-Wno-maybe-uninitialized')
+#if os.uname().sysname == 'Darwin': # macOS
+#    extra_compile_args_list.append('-std=c++11')
+#else:
+#    extra_compile_args_list.append('-Wno-maybe-uninitialized')
+extra_compile_args_list.append('-Wno-maybe-uninitialized')
 
 setup(name='Graphillion',
       version=release.version,
