From 308890fb664cfbf10a858035894a4f73b7cfa47e Mon Sep 17 00:00:00 2001
From: Xiretza <xiretza@xiretza.xyz>
Date: Sat, 24 Apr 2021 23:30:16 +0200
Subject: [PATCH] makefile: install udev rules to correct location

---
 Makefile.am | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index d5ebce1..a41acdf 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -69,14 +69,8 @@ dist_pkgdata_DATA = $(udev_DATA)
 # sudo udevadm control --reload-rules
 #
 udevrule_DATA = $(top_srcdir)/95-perseus.rules
-udevruledir = $(bindir)
+udevruledir = $(libdir)/udev/rules.d/
 EXTRA_DIST = $(top_srcdir)/95-perseus.rules README.Windows $(top_srcdir)/build-aux/git-version-gen fpga_data.c
-
-install-data-hook:
-	-cp $(bindir)/95-perseus.rules /etc/udev/rules.d/ ; groupadd perseususb ; echo $(INST_USER) ;usermod -a -G perseususb $(INST_USER) ; udevadm control --reload-rules
-
-uninstall-hook:
-	-rm /etc/udev/rules.d/95-perseus.rules ; groupdel perseususb ; udevadm control --reload-rules
 endif
 
 #
-- 
2.31.1
