--- src/SFlock-0.3.10/setup.py	2019-04-21 13:02:51.000000000 +0200
+++ setup.py	2019-04-30 12:11:44.036666618 +0200
@@ -25,10 +25,10 @@
     description="Sample staging and detonation utility",
     include_package_data=True,
     install_requires=[
-        "click==6.6",
-        "cryptography>=2.1",
-        "olefile==0.43",
-        "peepdf>=0.4.1",
-        "python-magic==0.4.12",
+        "click",
+        "cryptography",
+        "olefile",
+        "peepdf",
+        "python-magic",
     ],
 )
