--- pf_ring.c.old	2018-10-29 06:21:20.362535743 +0000
+++ pf_ring.c	2018-10-29 06:21:45.365030745 +0000
@@ -8000,8 +8000,8 @@
       case NETDEV_POST_INIT:        what = "NETDEV_POST_INIT"; break;
 #ifdef NETDEV_UNREGISTER_FINAL
       case NETDEV_UNREGISTER_FINAL: what = "NETDEV_UNREGISTER_FINAL"; break;
-#else
-      case NETDEV_UNREGISTER_BATCH: what = "NETDEV_UNREGISTER_BATCH"; break;
+/*#else
+      case NETDEV_UNREGISTER_BATCH: what = "NETDEV_UNREGISTER_BATCH"; break;*/
 #endif
       case NETDEV_RELEASE:          what = "NETDEV_RELEASE"; break;
       case NETDEV_NOTIFY_PEERS:     what = "NETDEV_NOTIFY_PEERS"; break;
