--- src/comment_parser-1.2.4/setup.py	2022-01-26 05:49:14.000000000 +0100
+++ setup.py	2022-03-27 17:52:43.189998972 +0200
@@ -23,7 +23,7 @@
     long_description=readme(),
     long_description_content_type='text/markdown',
     packages=['comment_parser', 'comment_parser.parsers'],
-    install_requires=['python-magic==0.4.24'],
+    install_requires=['python-magic'],
     test_suite='nose.collector',
     tests_require=['nose'],
     zip_safe=False)
