--- res/lang/en.ts	2016-04-08 05:33:01.927645775 -0400
+++ res/lang/en.ts.new	2016-04-08 05:38:46.388668962 -0400
@@ -319,7 +319,7 @@
     </message>
     <message>
         <source>download_vendor</source>
-        <translation><![CDATA[You need to download some 3rd-party applications/files from <a href="%1">here</a> and place them in <strong>%2</strong> for APK Studio to work.]]></translation>
+        <translation><![CDATA[You need to run apkstudio-setup.sh first for APK Studio to work.]]></translation>
     </message>
     <message>
         <source>decode_failure</source>
