--- src/WiFi-Pumpkin/Core/Main.py	2016-01-19 22:24:18.027452915 +0100
+++ Main.py	2016-01-19 22:37:39.446444589 +0100
@@ -19,7 +19,7 @@
 from Plugins.sslstrip.StrippingProxy import StrippingProxy
 from Plugins.sslstrip.URLMonitor import URLMonitor
 from Plugins.sslstrip.CookieCleaner import CookieCleaner
-if search('/usr/share/',argv[0]):chdir('/usr/share/WiFi-Pumpkin/')
+if search('/usr/share/',argv[0]):chdir('/usr/share/wifi-pumpkin/')
 
 """
 Description:
