diff -upr clang-3.1.src.orig/lib/Driver/Tools.cpp clang-3.1.src/lib/Driver/Tools.cpp
--- clang-3.1.src.orig/lib/Driver/Tools.cpp	2012-04-19 00:32:25.000000000 +0300
+++ clang-3.1.src/lib/Driver/Tools.cpp	2012-05-23 21:12:08.000000000 +0300
@@ -193,7 +193,7 @@ static void addProfileRT(const ToolChain
   // libprofile_rt.so. We used to use the -l:libprofile_rt.a syntax, but that is
   // not supported by old linkers.
   std::string ProfileRT =
-    std::string(TC.getDriver().Dir) + "/../lib/libprofile_rt.a";
+    std::string(TC.getDriver().Dir) + "/../lib/llvm/libprofile_rt.a";
 
   CmdArgs.push_back(Args.MakeArgString(ProfileRT));
 }
