--- src/responder/odict.py	2021-12-16 00:05:52.353329132 +0100
+++ odict.py	2021-12-16 00:09:09.433329129 +0100
@@ -3,7 +3,7 @@
 	from UserDict import DictMixin
 except ImportError:
 	from collections import UserDict
-	from collections import MutableMapping as DictMixin
+	from collections.abc import MutableMapping as DictMixin
 
 class OrderedDict(dict, DictMixin):
 
@@ -19,7 +19,7 @@
     def clear(self):
         self.__end = end = []
         end += [None, end, end]
-        self.__map = {} 
+        self.__map = {}
         dict.clear(self)
 
     def __setitem__(self, key, value):
