diff -Nur nfs.orig/Makefile nfs/Makefile
--- nfs.orig/Makefile	1998-05-20 06:37:23.000000000 +1000
+++ nfs/Makefile	2010-08-05 23:42:20.794623280 +1000
@@ -22,10 +22,10 @@
 #

 # uncomment the following 4 lines for Solaris 2.x
-CC		= gcc
-CFLAGS		= -DSYSV -DREADLINE -I/usr/local/include
-LIBS		= -lsocket -L/usr/ucblib -R/usr/ucblib -lrpcsoc -lnsl \
-		  -L/usr/local/lib -lreadline -lhistory -ltermlib
+#CC		= gcc
+#CFLAGS		= -DSYSV -DREADLINE -I/usr/local/include
+#LIBS		= -lsocket -L/usr/ucblib -R/usr/ucblib -lrpcsoc -lnsl \
+#		  -L/usr/local/lib -lreadline -lhistory -ltermlib

 # uncomment the following 3 lines for AIX
 #CC		= gcc
@@ -33,9 +33,9 @@
 #LIBS		=

 # uncomment the following 3 lines for linux (tested on 2.0.33/redhat 5)
-#CC		= gcc
-#CFLAGS		= -DREADLINE -I/usr/local/include
-#LIBS		= -L/usr/local/lib -lreadline -lhistory -lncurses
+CC		= gcc
+CFLAGS		= -DREADLINE
+LIBS		= -lreadline -lhistory -ltirpc

 RPCGEN		= rpcgen
 RGFLAGS		= -C
diff -Nur nfs.orig/nfs.c nfs/nfs.c
--- nfs.orig/nfs.c	1998-05-20 06:36:10.000000000 +1000
+++ nfs/nfs.c	2010-08-05 23:42:09.724623249 +1000
@@ -189,7 +189,7 @@

 void interrupt(int);
 int command(char *);
-int getline(char *, int, int *, char **, int);
+int nfs_getline(char *, int, int *, char **, int);
 void do_host(int, char **);
 void do_setuid(int, char **);
 void do_setgid(int, char **);
@@ -266,7 +266,7 @@

     /* interpreter's main command loop */
     if (setjmp(intenv)) putchar('\n');
-    while (getline(buffer, BUFSIZ, &argcount, argvec, NARGVEC)) {
+    while (nfs_getline(buffer, BUFSIZ, &argcount, argvec, NARGVEC)) {
 	if (argcount == 0) continue;
 	if ((cmd = command(argvec[0])) == CMD_QUIT)
 	    break;
@@ -377,7 +377,7 @@
  * it up into an argument vector.
  */
 int
-getline(char *buf, int bufsize, int *argc, char **argv, int argvsize)
+nfs_getline(char *buf, int bufsize, int *argc, char **argv, int argvsize)
 {
     register char *p;

