--- src/misc.c	2005-08-31 08:52:12.000000000 +0200
+++ src/misc.c	2011-10-30 22:53:21.132879261 +0100
@@ -33,19 +33,22 @@
 
 char *search_vendor(unsigned char mac[6])
 {
-	char tmac[6];
+	char tmac[7];
 	int i = 0;
 	
 	sprintf(tmac, "%02x%02x%02x", mac[0], mac[1], mac[2]);
 
-    /* Convert mac prefix to upper */
-    for (i=0; i<6; i++)
-        tmac[i] = toupper(tmac[i]);
+	/* Convert mac prefix to upper */
+	for (i=0; i<6; i++)
+	   tmac[i] = toupper(tmac[i]);
 	
-	for (i=0; i<8436; i++)
+	i = 0;
+
+	while (oui_table[i].prefix != NULL)
 	{
 		if (strcmp(oui_table[i].prefix, tmac) == 0)
 			return oui_table[i].vendor;
+        	i++;
 	}
 	
 	return "Unknown vendor";
