--- a/setup.py	2024-10-08 10:49:28.976279733 +0200
+++ b/setup.py	2024-10-08 10:50:05.937183133 +0200
@@ -44,12 +44,12 @@
     install_requires=[
         "click",
         "jinja2",
-        "pefile>=2019.4.18, <2019.5.0"
-        "pyyaml>=5.1",
-        "sqlalchemy>=1.4, <1.5",
-        "alembic>=1.7.4, <1.8",
-        "requests>=2.22.0, <3",
-        "psutil>=5.4.8, <6"
+        "pefile",
+        "pyyaml",
+        "sqlalchemy",
+        "alembic",
+        "requests",
+        "psutil"
     ],
     extras_require={
         ":sys_platform == 'win32'": [
