--- a/system/db/sqlite3/default.sql	2025-10-01 00:40:49.261633590 +0200
+++ b/system/db/sqlite3/default.sql	2025-10-01 00:42:05.102631330 +0200
@@ -1,8 +1,8 @@
-INSERT INTO params VALUES(1, "register", 0, "");
-INSERT INTO params VALUES(2, "feedback", 0, "");
-INSERT INTO groups VALUES(1, "admin");
-INSERT INTO groups VALUES(2, "users");
+INSERT INTO params VALUES(1, 'register', 0, '');
+INSERT INTO params VALUES(2, 'feedback', 0, '');
+INSERT INTO groups VALUES(1, 'admin');
+INSERT INTO groups VALUES(2, 'users');
 
-INSERT INTO users VALUES(1, "admin", "4295173f5047ed4edf9eee2a047abc33", "demo1@xplico.org", 1, "none", "Xplico", "Team", "0", 0, 'FULL', 1, 1, 1, 0);
-INSERT INTO users VALUES(2, "xplico", "4295173f5047ed4edf9eee2a047abc33", "demo2@xplico.org", 1, "none", "Xplico", "Team", "0", 0, 'NORMAL', 1, 1, 2, 0);
+INSERT INTO users VALUES(1, 'admin', '4295173f5047ed4edf9eee2a047abc33', 'demo1@xplico.org', 1, 'none', 'Xplico', 'Team', '0', 0, 'FULL', 1, 1, 1, 0);
+INSERT INTO users VALUES(2, 'xplico', '4295173f5047ed4edf9eee2a047abc33', 'demo2@xplico.org', 1, 'none', 'Xplico', 'Team', '0', 0, 'NORMAL', 1, 1, 2, 0);
 
--- a/system/db/mysql/default.sql	2025-10-01 00:41:07.879382502 +0200
+++ b/system/db/mysql/default.sql	2025-10-01 00:42:24.334831735 +0200
@@ -1,10 +1,10 @@
-INSERT INTO params VALUES(1, "register", 0, "");
-INSERT INTO params VALUES(2, "feedback", 0, "");
-INSERT INTO groups VALUES(1, "admin");
-INSERT INTO groups VALUES(2, "users");
+INSERT INTO params VALUES(1, 'register', 0, '');
+INSERT INTO params VALUES(2, 'feedback', 0, '');
+INSERT INTO groups VALUES(1, 'admin');
+INSERT INTO groups VALUES(2, 'users');
 
-INSERT INTO users VALUES(1, "admin", "4295173f5047ed4edf9eee2a047abc33", "demo1@xplico.org", 1, "none", "Xplico", "Team", "0", 0, 'FULL', 1, 1, 1, 0);
-INSERT INTO users VALUES(2, "xplico", "4295173f5047ed4edf9eee2a047abc33", "demo2@xplico.org", 1, "none", "Xplico", "Team", "0", 0, 'NORMAL', 1, 1, 2, 0);
+INSERT INTO users VALUES(1, 'admin', '4295173f5047ed4edf9eee2a047abc33', 'demo1@xplico.org', 1, 'none', 'Xplico', 'Team', '0', 0, 'FULL', 1, 1, 1, 0);
+INSERT INTO users VALUES(2, 'xplico', '4295173f5047ed4edf9eee2a047abc33', 'demo2@xplico.org', 1, 'none', 'Xplico', 'Team', '0', 0, 'NORMAL', 1, 1, 2, 0);
 UPDATE users SET quota_used=0;
--- UPDATE params SET nvalue=1 WHERE name="register";
+-- UPDATE params SET nvalue=1 WHERE name='register';
 
