From: =?utf-8?q?Rapha=C3=ABl_Hertzog?= <hertzog@debian.org>
Date: Mon, 19 Aug 2019 09:35:21 +0200
Subject: Improve systemd service file

---
 xplico.service | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/xplico.service b/xplico.service
index fbab722..7e6775f 100644
--- a/xplico.service
+++ b/xplico.service
@@ -1,13 +1,14 @@
 [Unit]
 Description=Xplico
-After=network.target syslog.target
+After=network.target
+Documentation=https://www.xplico.org/docs
 
 [Service]
 Type=forking
 PrivateTmp=true
 ExecStart=/opt/xplico/bin/dema -d /opt/xplico -b sqlite
 Restart=always
-PIDFile=/var/run/dema.pid
+PIDFile=/run/dema.pid
 
 [Install]
 WantedBy=multi-user.target
