From c992641d97d0245fa419ca7c615488d0d8b77201 Mon Sep 17 00:00:00 2001
From: Astro Angelfish <astroangelfish@orangemc.moe>
Date: Fri, 28 Feb 2025 18:35:38 +0000
Subject: [PATCH] Add ./lib to library load path

---
 bin/seccomp-tools | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bin/seccomp-tools b/bin/seccomp-tools
index 1a3c2cf..20844dc 100755
--- a/bin/seccomp-tools
+++ b/bin/seccomp-tools
@@ -1,6 +1,8 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
+$LOAD_PATH << File.expand_path('../lib', File.dirname(__FILE__))
+
 require 'seccomp-tools/cli/cli'
 
 SeccompTools::CLI.work(ARGV)
-- 
2.48.1

