--- valgrind/include/vki/vki-amd64-linux.h.jj	2009-08-19 09:37:07.000000000 -0400
+++ valgrind/include/vki/vki-amd64-linux.h	2010-01-20 04:41:52.662552000 -0500
@@ -310,6 +310,9 @@ struct vki_stat {
 	long		st_blksize;
 	long		st_blocks;	/* Number 512-byte blocks allocated. */
 
+#undef st_atime
+#undef st_mtime
+#undef st_ctime
 	unsigned long	st_atime;
 	unsigned long 	st_atime_nsec; 
 	unsigned long	st_mtime;
--- valgrind/include/vki/vki-x86-linux.h.jj	2009-08-19 09:37:07.000000000 -0400
+++ valgrind/include/vki/vki-x86-linux.h	2010-01-20 04:42:19.649995000 -0500
@@ -348,6 +348,9 @@ struct vki_stat {
 	unsigned long  st_size;
 	unsigned long  st_blksize;
 	unsigned long  st_blocks;
+#undef st_atime
+#undef st_mtime
+#undef st_ctime
 	unsigned long  st_atime;
 	unsigned long  st_atime_nsec;
 	unsigned long  st_mtime;
--- valgrind/include/vki/vki-ppc32-linux.h.jj	2009-08-19 09:37:07.000000000 -0400
+++ valgrind/include/vki/vki-ppc32-linux.h	2010-01-20 04:42:44.861651000 -0500
@@ -392,6 +392,9 @@ struct vki_stat {
    long			st_size;
    unsigned long	st_blksize;
    unsigned long	st_blocks;
+#undef st_atime
+#undef st_mtime
+#undef st_ctime
    unsigned long	st_atime;
    unsigned long	st_atime_nsec;
    unsigned long	st_mtime;
--- valgrind/include/vki/vki-ppc64-linux.h.jj	2009-08-19 09:37:07.000000000 -0400
+++ valgrind/include/vki/vki-ppc64-linux.h	2010-01-20 04:43:05.579922000 -0500
@@ -428,6 +428,9 @@ struct vki_stat {
   long            st_size;
   unsigned long   st_blksize;
   unsigned long   st_blocks;
+#undef st_atime
+#undef st_mtime
+#undef st_ctime
   unsigned long   st_atime;
   unsigned long   st_atime_nsec;
   unsigned long   st_mtime;
