diff -ru restkit-4.0.0/setup.py restkit-4.0.0.new/setup.py
--- restkit-4.0.0/setup.py      2012-01-25 03:47:45.000000000 +0100
+++ restkit-4.0.0.new/setup.py  2012-01-26 21:17:10.211046123 +0100
@@ -39,8 +39,8 @@
         long_description = f.read()
 
     DATA_FILES = [
-        ('restkit', ["LICENSE", "MANIFEST.in", "NOTICE", "README.rst",
-                        "THANKS", "TODO.txt"])
+        ('share/doc/restkit', ["LICENSE", "MANIFEST.in", "NOTICE",
+                               "README.rst", "THANKS", "TODO.txt"])
         ]
 
     options=dict(
