diff --git a/stepic b/stepic
index 4c8ac97..f1dc424 100755
--- a/stepic
+++ b/stepic
@@ -21,7 +21,7 @@
 import optparse
 import sys
 import traceback
-import Image
+from PIL import Image
 import stepic
 
 
diff --git a/stepic.py b/stepic.py
index e8c9078..1fc292a 100644
--- a/stepic.py
+++ b/stepic.py
@@ -29,7 +29,7 @@ __version__ = '0.3'
 
 import warnings
 try:
-    import Image
+    from PIL import Image
 except:
     warnings.warn('Could not find PIL. Only encode_imdata and decode_imdata will work.',
                   ImportWarning, stacklevel=2)
