diff -up QScintilla_gpl-2.9.2/Qt4Qt5/features/qscintilla2.prf.qt5 QScintilla_gpl-2.9.2/Qt4Qt5/features/qscintilla2.prf
--- QScintilla_gpl-2.9.2/Qt4Qt5/features/qscintilla2.prf.qt5	2016-04-18 04:08:48.000000000 -0500
+++ QScintilla_gpl-2.9.2/Qt4Qt5/features/qscintilla2.prf	2016-04-18 08:13:23.935665305 -0500
@@ -19,9 +19,17 @@ CONFIG(debug, debug|release) {
         win32: {
             LIBS += -lqscintilla2d
         } else {
+        greaterThan(QT_MAJOR_VERSION, 4) {
+            LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+        } else {
             LIBS += -lqscintilla2
         }
+        }
     }
 } else {
+greaterThan(QT_MAJOR_VERSION, 4) {
+    LIBS += -lqscintilla2-qt$${QT_MAJOR_VERSION}
+} else {
     LIBS += -lqscintilla2
 }
+}
diff -up QScintilla_gpl-2.9.2/Qt4Qt5/qscintilla.pro.qt5 QScintilla_gpl-2.9.2/Qt4Qt5/qscintilla.pro
--- QScintilla_gpl-2.9.2/Qt4Qt5/qscintilla.pro.qt5	2016-04-18 08:13:23.951665381 -0500
+++ QScintilla_gpl-2.9.2/Qt4Qt5/qscintilla.pro	2016-04-18 08:14:20.807934597 -0500
@@ -23,7 +23,11 @@
 !win32:VERSION = 12.0.1
 
 TEMPLATE = lib
+greaterThan(QT_MAJOR_VERSION, 4) {
+TARGET = qscintilla2-qt$${QT_MAJOR_VERSION}
+} else {
 TARGET = qscintilla2
+}
 CONFIG += qt warn_off release thread exceptions hide_symbols
 INCLUDEPATH += . ../include ../lexlib ../src
 
