--- gcc/config/i386/linux.h.orig	2014-04-28 10:00:16.556121683 +0000
+++ gcc/config/i386/linux.h	2014-04-28 10:00:43.105959247 +0000
@@ -257,9 +257,9 @@
       {									\
 	struct rt_sigframe {						\
 	  int sig;							\
-	  struct siginfo *pinfo;					\
+	  siginfo_t *pinfo;					\
 	  void *puc;							\
-	  struct siginfo info;						\
+	  siginfo_t info;						\
 	  struct ucontext uc;						\
 	} *rt_ = (CONTEXT)->cfa;					\
 	sc_ = (struct sigcontext *) &rt_->uc.uc_mcontext;		\
