do not link against libraries that are not needed

Index: t1lib-5.1.1/lib/Makefile.in
===================================================================
--- t1lib-5.1.1.orig/lib/Makefile.in	2008-01-05 19:17:21.000000000 +0100
+++ t1lib-5.1.1/lib/Makefile.in	2008-01-05 19:17:38.000000000 +0100
@@ -24,7 +24,7 @@
 X_LIBS    = @X_LIBS@
 TOPSRC    = @top_srcdir@
 XPM_LIB   = -lXpm
-XLIB      = @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+XLIB      = -lX11
 LDFLAGS   = @LDFLAGS@
 LDLIBS    = @LDLIBS@
 AR        = ar rc
@@ -137,7 +137,7 @@
 	$(LIBTOOL) --mode=link \
 		$(CC) $(LDFLAGS) -o $@ $(T1LIBX_OBJS) \
 	         -version-info @T1LIB_LT_CURRENT@:@T1LIB_LT_REVISION@:@T1LIB_LT_AGE@ \
-	         libt1.la $(X_LIBS) $(XPM_LIB) $(XLIB)  -no-undefined -rpath $(libdir) 
+	         libt1.la $(X_LIBS) $(XLIB)  -no-undefined -rpath $(libdir) 
 	cp t1lib/t1libx.h .
 
 
Index: t1lib-5.1.1/type1afm/Makefile.in
===================================================================
--- t1lib-5.1.1.orig/type1afm/Makefile.in	2008-01-05 19:17:52.000000000 +0100
+++ t1lib-5.1.1/type1afm/Makefile.in	2008-01-05 19:18:02.000000000 +0100
@@ -70,7 +70,7 @@
 
 type1afm: $(OBJS) ../lib/t1lib.h
 	$(LIBTOOL) --mode=link \
-		$(CC) -o type1afm $(LDFLAGS) $(OBJS) $(T1LIB) $(LDLIBS)
+		$(CC) -o type1afm $(LDFLAGS) $(OBJS) $(T1LIB)
 
 .SUFFIXES: .lo
 .c.lo:
Index: t1lib-5.1.1/xglyph/Makefile.in
===================================================================
--- t1lib-5.1.1.orig/xglyph/Makefile.in	2008-01-05 19:18:15.000000000 +0100
+++ t1lib-5.1.1/xglyph/Makefile.in	2008-01-05 19:18:31.000000000 +0100
@@ -24,7 +24,7 @@
 X_LIBS    = @X_LIBS@
 TOPSRC    = @top_srcdir@
 XPM_LIB   = -lXpm
-XLIB      = @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+XLIB      = -lX11 @X_EXTRA_LIBS@
 LDFLAGS   = @LDFLAGS@
 LDLIBS    = @LDLIBS@ 
 AR        = ar rc
@@ -65,7 +65,7 @@
 
 T1LIB  = ../lib/libt1.la
 T1LIBX = ../lib/libt1x.la
-XAWLIB = -lXaw -lXt -lXmu
+XAWLIB = -lXaw -lXt
 
 
 all: xglyph
