diff -up src/Net-RawIP-0.25/eth.h.orig src/Net-RawIP-0.25/eth.h
--- src/Net-RawIP-0.25/eth.h.orig	2024-09-14 20:14:39.560000000 -0600
+++ src/Net-RawIP-0.25/eth.h	2024-09-14 20:16:03.380000000 -0600
@@ -0,0 +1,4 @@
+
+int tap(char *dev,unsigned int *my_eth_ip,unsigned char *my_eth_mac);
+int mac_disc(unsigned int addr,unsigned char * eth_mac);
+void send_eth_packet(int fd, char* eth_device, u_char *pkt, int len, int flag);
diff -up src/Net-RawIP-0.25/ifaddrlist.c.orig src/Net-RawIP-0.25/ifaddrlist.c
--- src/Net-RawIP-0.25/ifaddrlist.c.orig	2007-04-03 04:59:40.000000000 -0600
+++ src/Net-RawIP-0.25/ifaddrlist.c	2024-09-14 20:11:28.430000000 -0600
@@ -1,6 +1,5 @@
 #ifdef   _LINUX_
-#define  _BSD_SOURCE 1
-#define  __FAVOR_BSD 1
+#define  _DEFAULT_SOURCE 1
 #endif
 
 #include <sys/param.h>
diff -up src/Net-RawIP-0.25/RawIP.xs.orig src/Net-RawIP-0.25/RawIP.xs
--- src/Net-RawIP-0.25/RawIP.xs.orig	2024-09-14 20:16:57.420000000 -0600
+++ src/Net-RawIP-0.25/RawIP.xs	2024-09-14 20:17:05.910000000 -0600
@@ -31,6 +31,7 @@ extern "C" {
 #include <sys/cdefs.h>
 #endif
 #include "ifaddrlist.h"
+#include "eth.h"
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <pcap.h>
