*** sshuttleORIGIN/sshuttle	2011-05-21 22:58:24.156725075 +0200
--- sshuttle	2011-05-21 22:59:16.354698468 +0200
***************
*** 4,10 ****
  	[ -L "$EXE" ] || break
  	EXE=$(readlink "$EXE")
  done
! DIR=$(dirname "$EXE")
  if python2 -V 2>/dev/null; then
  	exec python2 "$DIR/main.py" python2 "$@"
  else
--- 4,10 ----
  	[ -L "$EXE" ] || break
  	EXE=$(readlink "$EXE")
  done
! DIR=/usr/share/sshuttle
  if python2 -V 2>/dev/null; then
  	exec python2 "$DIR/main.py" python2 "$@"
  else
