diff --git a/bettercap.gemspec b/bettercap.gemspec
index 91e1458..19fc0b1 100644
--- a/bettercap.gemspec
+++ b/bettercap.gemspec
@@ -14,8 +14,8 @@ Gem::Specification.new do |gem|
   gem.homepage = %q{https://github.com/evilsocket/bettercap}
 
   gem.add_dependency( 'colorize', '~> 0.8.0' )
-  gem.add_dependency( 'packetfu', '~> 1.1', '>= 1.1.10' )
-  gem.add_dependency( 'pcaprub', '~> 0.12', '>= 0.12.0', '<= 1.1.11' )
+  gem.add_dependency( 'packetfu', '~> 1.1', '>= 1.1.10' , '<= 1.1.11')
+  gem.add_dependency( 'pcaprub', '~> 0.12', '>= 0.12.0' )
   gem.add_dependency( 'network_interface', '~> 0.0', '>= 0.0.1' )
   gem.add_dependency( 'net-dns', '~> 0.8', '>= 0.8.0' )
   gem.add_dependency( 'rubydns', '~> 1.0', '>= 1.0.3' )
