diff -rauN qt-everywhere-src-5.15.17/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro qt-everywhere-src-5.15.17-assimp-patch/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro
--- qt-everywhere-src-5.15.17/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro	2024-11-20 14:56:46.000000000 +0100
+++ qt-everywhere-src-5.15.17-assimp-patch/qtquick3d/src/plugins/assetimporters/assimp/assimp.pro	2025-11-07 16:07:42.615686157 +0100
@@ -10,7 +10,7 @@
 include($$OUT_PWD/../qtassetimporters-config.pri)
 
 qtConfig(system-assimp):!if(cross_compile:host_build) {
-    QMAKE_USE_PRIVATE += assimp
+    QMAKE_USE_PRIVATE += quick3d-assimp
 } else {
     include(../../../3rdparty/assimp/assimp.pri)
 }
