--- autogen.sh.bad	2012-09-14 17:34:05.956148106 +0200
+++ autogen.sh	2012-09-14 17:34:26.959482287 +0200
@@ -19,7 +19,7 @@
 }
 
 grep "^A[CM]_PROG_LIBTOOL" configure.in >/dev/null && {
-  if which libtoolize && (libtooloze --version) < /dev/null > /dev/null 2>&1; then
+  if which libtoolize && (libtoolize --version) < /dev/null > /dev/null 2>&1; then
     LIBTOOLIZE=libtoolize
   else
       if which glibtoolize && (glibtoolize --version) < /dev/null > /dev/null 2>&1; then
