--- src/pwntools-4.6.0/setup.py	2021-07-12 23:00:40.000000000 +0200
+++ setup.py	2021-08-18 09:33:39.003332391 +0200
@@ -60,7 +60,7 @@
                         'psutil>=3.3.0',
                         'intervaltree>=3.0',
                         'sortedcontainers',
-                        'unicorn>=1.0.2rc1,<1.0.2rc4', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170, Gallopsled/pwntools#1538
+                        'unicorn', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170, Gallopsled/pwntools#1538
                         'six>=1.12.0',
                         'rpyc',
                         'colored_traceback',
