--- src/pwntools-4.1.2/setup.py	2020-06-06 03:21:21.000000000 +0200
+++ setup.py	2020-06-14 22:32:12.999998943 +0200
@@ -60,7 +60,7 @@
                         'psutil>=3.3.0',
                         'intervaltree>=3.0',
                         'sortedcontainers',
-                        'unicorn>=1.0.2rc1,<1.0.2rc4', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170, Gallopsled/pwntools#1538
+                        'unicorn', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170, Gallopsled/pwntools#1538
                         'six>=1.12.0',
 ]
 
