--- src/pwntools-3.12.1/setup.py	2018-09-18 00:11:45.000000000 +0200
+++ setup.py	2018-09-21 09:15:58.793333415 +0200
@@ -42,23 +42,23 @@
     if not flag:
         console_scripts.append(script)
 
-install_requires     = ['paramiko>=1.15.2',
-                        'mako>=1.0.0',
-                        'pyelftools>=0.2.4',
-                        'capstone>=3.0.5rc2', # See Gallopsled/pwntools#971, Gallopsled/pwntools#1160
-                        'ropgadget>=5.3',
-                        'pyserial>=2.7',
-                        'requests>=2.0',
-                        'pip>=6.0.8',
-                        'tox>=1.8.1',
-                        'pygments>=2.0',
+install_requires     = ['paramiko',
+                        'mako',
+                        'pyelftools',
+                        'capstone', # See Gallopsled/pwntools#971, Gallopsled/pwntools#1160
+                        'ropgadget',
+                        'pyserial',
+                        'requests',
+                        'pip',
+                        'tox',
+                        'pygments',
                         'pysocks',
                         'python-dateutil',
                         'pypandoc',
                         'packaging',
-                        'psutil>=3.3.0',
+                        'psutil',
                         'intervaltree',
-                        'sortedcontainers<2.0', # See Gallopsled/pwntools#1154
+                        'sortedcontainers', # See Gallopsled/pwntools#1154
                         'unicorn']
 
 # Check that the user has installed the Python development headers
