--- src/pwntools-4.0.1/setup.py	2020-01-22 21:57:27.000000000 +0100
+++ setup.py	2020-01-23 12:20:12.886666770 +0100
@@ -45,23 +45,23 @@
     if not flag:
         console_scripts.append(script)
 
-install_requires     = ['paramiko>=1.15.2',
-                        'mako>=1.0.0',
-                        'pyelftools>=0.2.4',
-                        'capstone>=3.0.5rc2', # See Gallopsled/pwntools#971, Gallopsled/pwntools#1160
-                        'ropgadget>=5.3',
-                        'pyserial>=2.7',
-                        'requests>=2.0',
-                        'pip>=6.0.8',
-                        'tox>=1.8.1',
-                        'pygments>=2.0',
+install_requires     = ['paramiko',
+                        'mako',
+                        'pyelftools',
+                        'capstone', # See Gallopsled/pwntools#971, Gallopsled/pwntools#1160
+                        'ropgadget',
+                        'pyserial',
+                        'requests',
+                        'pip',
+                        'tox',
+                        'pygments',
                         'pysocks',
                         'python-dateutil',
                         'packaging',
-                        'psutil>=3.3.0',
-                        'intervaltree>=3.0',
+                        'psutil',
+                        'intervaltree',
                         'sortedcontainers',
-                        'unicorn>=1.0.2rc1', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170
+                        'unicorn', # see unicorn-engine/unicorn#1100, unicorn-engine/unicorn#1170
 ]
 
 # Check that the user has installed the Python development headers
