--- src/scanssh-2.1/atomicio.c	2001-02-18 02:08:20.000000000 +0100
+++ atomicio.c	2025-12-16 12:02:46.442865425 +0100
@@ -34,26 +34,24 @@
  * ensure all of data on socket comes through. f==read || f==write
  */
 ssize_t
-atomicio(f, fd, _s, n)
-	ssize_t (*f) ();
-	int fd;
-	void *_s;
-	size_t n;
+atomicio(ssize_t (*f)(int, char *, size_t), int fd, void *_s, size_t n)
 {
-	char *s = _s;
-	ssize_t res, pos = 0;
+  char *s = _s;
+  ssize_t res, pos = 0;
 
-	while (n > pos) {
-		res = (f) (fd, s + pos, n - pos);
-		switch (res) {
-		case -1:
-			if (errno == EINTR || errno == EAGAIN)
-				continue;
-		case 0:
-			return (res);
-		default:
-			pos += res;
-		}
-	}
-	return (pos);
+  while (n > pos) {
+    res = f(fd, s + pos, n - pos);
+
+    switch (res) {
+    case -1:
+      if (errno == EINTR || errno == EAGAIN)
+        continue;
+      break;
+    case 0:
+      return res;
+    default:
+      pos += res;
+    }
+  }
+  return pos;
 }
