--- src/scanssh-2.1/Makefile.in	2005-03-05 20:28:31.000000000 +0100
+++ Makefile.in	2025-12-16 17:59:59.503410162 +0100
@@ -76,10 +76,10 @@
 
 bin_PROGRAMS = scanssh
 
-scanssh_SOURCES = scanssh.c atomicio.c exclude.c connecter.c xmalloc.c 	interface.c socks.c http.c telnet.c exclude.h interface.h 	scanssh.h socks.h xmalloc.h
+scanssh_SOURCES = scanssh.c atomicio.c exclude.c connecter.c xmalloc.c md5.c interface.c socks.c http.c telnet.c exclude.h interface.h scanssh.h socks.h xmalloc.h
 
 scanssh_LDADD = @LIBOBJS@ @PCAPLIB@ @EVENTLIB@ @DNETLIB@
-CFLAGS = -O2 -Wall -g
+CFLAGS = -O2 -Wall -g -Icompat
 
 INCLUDES = -I$(top_srcdir)/@DNETCOMPAT@ -I$(top_srcdir)/compat 	@EVENTINC@ @PCAPINC@ @DNETINC@
 
@@ -93,7 +93,7 @@
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = config.h
-CONFIG_CLEAN_FILES = 
+CONFIG_CLEAN_FILES =
 PROGRAMS =  $(bin_PROGRAMS)
 
 
@@ -104,7 +104,7 @@
 scanssh_OBJECTS =  scanssh.o atomicio.o exclude.o connecter.o xmalloc.o \
 interface.o socks.o http.o telnet.o
 scanssh_DEPENDENCIES =  @LIBOBJS@
-scanssh_LDFLAGS = 
+scanssh_LDFLAGS = -lssl -lcrypto -lmd
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@
@@ -129,14 +129,14 @@
 all: all-redirect
 .SUFFIXES:
 .SUFFIXES: .S .c .o .s
-$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
+$(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4)
 	cd $(top_srcdir) && $(AUTOMAKE) --foreign Makefile
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
 	cd $(top_builddir) \
 	  && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
-$(ACLOCAL_M4):  configure.in 
+$(ACLOCAL_M4):  configure.in
 	cd $(srcdir) && $(ACLOCAL)
 
 config.status: $(srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
