--- src/scanssh-2.1/md5.c	2004-03-31 09:30:21.000000000 +0200
+++ md5.c	2025-12-16 12:01:12.455842336 +0100
@@ -24,6 +24,7 @@
    public domain.  */
 
 #include <sys/types.h>
+#include <string.h>
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
@@ -125,7 +126,7 @@
 }
 
 /*
- * Final wrapup - pad to 64-byte boundary with the bit pattern 
+ * Final wrapup - pad to 64-byte boundary with the bit pattern
  * 1 0* (64-bit count of bits processed, MSB-first)
  */
 void
