--- ssl/ssldecode.c-orig	2002-08-17 03:33:17.000000000 +0200
+++ ssl/ssldecode.c
@@ -51,6 +51,7 @@
 #include <openssl/ssl.h>
 #include <openssl/hmac.h>
 #include <openssl/evp.h>
+#include <openssl/md5.h>
 #include <openssl/x509v3.h>
 #endif
 #include &quot;ssldecode.h&quot;
@@ -131,7 +132,8 @@ int ssl_decode_ctx_create(dp,keyfile,pas
     ssl_decode_ctx *d=0;
     int r,_status;
     
-    SSLeay_add_all_algorithms();
+    SSL_library_init();
+    OpenSSL_add_all_algorithms();
     if(!(d=(ssl_decode_ctx *)malloc(sizeof(ssl_decode_ctx))))
       ABORT(R_NO_MEMORY);
     if(!(d-&gt;ssl_ctx=SSL_CTX_new(SSLv23_server_method()))) 
