--- GNUmakefile.clean	2012-11-21 09:39:12.700606899 -0700
+++ GNUmakefile	2012-11-21 09:39:39.630613084 -0700
@@ -4,13 +4,13 @@
 build: pywmi-build 
 
 ifeq ($(PYTHON),)
-PYTHON=python
+PYTHON=python2
 endif
 
 PYLIBDIR=	$(ZENHOME)/lib/python
 ZENBIN=		$(ZENHOME)/bin
 PYINCLUDE=	$(shell $(PYTHON) pyinclude.py)
-VERSION:=$(shell cd pysamba ; python -c 'import version as v; print v.VERSION')
+VERSION:=$(shell cd pysamba ; python2 -c 'import version as v; print v.VERSION')
 
 LIBSUFFIX=so.0
 ifeq ($(shell uname), Darwin)
