diff -urN openvas-client-3.0.3.orig//openvas/attack.c openvas-client-3.0.3/openvas/attack.c
--- openvas-client-3.0.3.orig//openvas/attack.c	2010-11-11 15:45:28.000000000 +0100
+++ openvas-client-3.0.3/openvas/attack.c	2011-06-08 16:16:51.568801049 +0200
@@ -17,7 +17,7 @@
 
 #include <includes.h>
 
-#include <openvas/network.h>
+#include <openvas/misc/network.h>
 
 #include "openvas_plugin.h"
 #include "comm.h"
diff -urN openvas-client-3.0.3.orig//openvas/auth.c openvas-client-3.0.3/openvas/auth.c
--- openvas-client-3.0.3.orig//openvas/auth.c	2009-10-15 19:58:35.000000000 +0200
+++ openvas-client-3.0.3/openvas/auth.c	2011-06-08 16:17:12.262454507 +0200
@@ -23,7 +23,7 @@
 #include <includes.h>
 #include <stdarg.h>
 
-#include <openvas/network.h>
+#include <openvas/misc/network.h>
 #include "comm.h"
 #include "auth.h"
 #include "sighand.h"
diff -urN openvas-client-3.0.3.orig//openvas/backend.c openvas-client-3.0.3/openvas/backend.c
--- openvas-client-3.0.3.orig//openvas/backend.c	2009-10-15 19:58:35.000000000 +0200
+++ openvas-client-3.0.3/openvas/backend.c	2011-06-08 16:17:24.592645245 +0200
@@ -28,7 +28,7 @@
 #include <includes.h>
 #include <openvas/base/hash_table_util.h> /* for keys_as_string_list */
 #include <openvas/base/severity_filter.h> /* for severity_filter_apply */
-#include <openvas/plugutils.h> /* for addslashes */
+#include <openvas/misc/plugutils.h> /* for addslashes */
 
 #include "context.h"
 #include "openvas_i18n.h"
diff -urN openvas-client-3.0.3.orig//openvas/parser.c openvas-client-3.0.3/openvas/parser.c
--- openvas-client-3.0.3.orig//openvas/parser.c	2009-10-15 18:43:34.000000000 +0200
+++ openvas-client-3.0.3/openvas/parser.c	2011-06-08 16:17:03.615654074 +0200
@@ -23,7 +23,7 @@
 #include <includes.h>
 #include "openvas_i18n.h"
 
-#include <openvas/plugutils.h> /* for addslashes */
+#include <openvas/misc/plugutils.h> /* for addslashes */
 
 #include "auth.h"
 #include "error_dlg.h"
diff -urN openvas-client-3.0.3.orig//openvas/prefs_dialog/prefs_dialog_auth.c openvas-client-3.0.3/openvas/prefs_dialog/prefs_dialog_auth.c
--- openvas-client-3.0.3.orig//openvas/prefs_dialog/prefs_dialog_auth.c	2009-10-20 09:42:26.000000000 +0200
+++ openvas-client-3.0.3/openvas/prefs_dialog/prefs_dialog_auth.c	2011-06-08 16:17:52.623078699 +0200
@@ -9,10 +9,10 @@
  */
 
 /* for close_stream_connection() */
-#include <openvas/network.h>
+#include <openvas/misc/network.h>
 
 /* for emalloc, efree, estrdup */
-#include <openvas/system.h>
+#include <openvas/misc/system.h>
 
 #include "openvas_i18n.h"
 
diff -urN openvas-client-3.0.3.orig//openvas/sighand.c openvas-client-3.0.3/openvas/sighand.c
--- openvas-client-3.0.3.orig//openvas/sighand.c	2009-10-15 18:43:34.000000000 +0200
+++ openvas-client-3.0.3/openvas/sighand.c	2011-06-08 16:16:41.921985292 +0200
@@ -23,7 +23,7 @@
 
 #include <includes.h>
 
-#include <openvas/network.h> /* for close_stream_connection */
+#include <openvas/misc/network.h> /* for close_stream_connection */
 
 #include "openvas_i18n.h"
 #include "error_dlg.h"
diff -urN openvas-client-3.0.3.orig//openvas/subset.c openvas-client-3.0.3/openvas/subset.c
--- openvas-client-3.0.3.orig//openvas/subset.c	2009-10-15 18:43:34.000000000 +0200
+++ openvas-client-3.0.3/openvas/subset.c	2011-06-08 16:16:28.721781150 +0200
@@ -38,7 +38,7 @@
 #include <includes.h>
 #include <stdarg.h>
 
-#include <openvas/plugutils.h> /* for addslashes */
+#include <openvas/misc/plugutils.h> /* for addslashes */
 
 #include "subset.h"
 
diff -urN openvas-client-3.0.3.orig//src/gui/nvt_pref_sshlogin.c openvas-client-3.0.3/src/gui/nvt_pref_sshlogin.c
--- openvas-client-3.0.3.orig//src/gui/nvt_pref_sshlogin.c	2010-11-11 15:45:28.000000000 +0100
+++ openvas-client-3.0.3/src/gui/nvt_pref_sshlogin.c	2011-06-08 16:16:10.394831082 +0200
@@ -34,7 +34,7 @@
 #include <gtk/gtk.h>
 #include "glib.h"
 
-#include <openvas/system.h> /* for efree, estrdup */
+#include <openvas/misc/system.h> /* for efree, estrdup */
 
 #include "nvt_pref_sshlogin.h"
 
