--- source4/heimdal/cf/make-proto.pl.old        2012-07-16 13:24:28.758749581 +0200
+++ source4/heimdal/cf/make-proto.pl    2012-07-16 13:25:33.877177549 +0200
@@ -1,8 +1,8 @@
 # Make prototypes from .c files
 # $Id$

-##use Getopt::Std;
-require 'getopts.pl';
+use Getopt::Std;
+#require 'getopts.pl';

 my $comment = 0;
 my $if_0 = 0;
@@ -12,7 +12,7 @@
 my $oproto = 1;
 my $private_func_re = "^_";

-do Getopts('x:m:o:p:dqE:R:P:') || die "foo";
+getopts('x:m:o:p:dqE:R:P:') || die "foo";

 if($opt_d) {
     $debug = 1;

