--- Makefile-orig	2021-03-12 16:30:43.042271858 -0500
+++ Makefile	2021-03-12 16:32:16.732681429 -0500
@@ -1,6 +1,6 @@
-DESTDIR		= 
-PREFIX		= /usr/local
-SBINDIR		= $(PREFIX)/sbin
+DESTDIR		=
+PREFIX		= /usr
+SBINDIR		= $(PREFIX)/bin
 MANDIR		= $(PREFIX)/share/man
 
 SRC		= src
@@ -10,8 +10,7 @@
 	$(MAKE) -C $(SRC)
 
 install: all
-	PREFIX=$(DESTDIR)$(PREFIX)
-	$(MAKE) -C $(SRC) install
+	PREFIX=$(DESTDIR)$(PREFIX) $(MAKE) -C $(SRC) install
 	install -D -m 0644 man/mdk4.1 $(DESTDIR)$(MANDIR)/man8/mdk4.1
 	gzip -f $(DESTDIR)$(MANDIR)/man8/mdk4.1
 
@@ -23,3 +22,4 @@
 	$(MAKE) -C $(SRC) test
 
 distclean: clean
+

