--- src/mdk4/Makefile	2018-11-23 19:42:52.842361263 -0500
+++ Makefile	2018-11-23 19:39:41.066801366 -0500
@@ -1,5 +1,6 @@
-PREFIX		= /usr/local
-SBINDIR		= $(PREFIX)/sbin
+DESTDIR =
+PREFIX		= /usr
+SBINDIR		= $(PREFIX)/bin
 MANDIR		= $(PREFIX)/share/man
 
 SRC		= src
@@ -9,9 +10,9 @@
 	$(MAKE) -C $(SRC)
 
 install: all
-	PREFIX=$(PREFIX) $(MAKE) -C $(SRC) install
-	install -D -m 0644 man/mdk4.1 $(MANDIR)/man8/mdk4.1
-	gzip -f $(MANDIR)/man8/mdk4.1
+	PREFIX=$(DESTDIR)$(PREFIX) $(MAKE) -C $(SRC) install
+	install -D -m 0644 man/mdk4.1 $(DESTDIR)$(MANDIR)/man8/mdk4.1
+	gzip -f $(DESTDIR)$(MANDIR)/man8/mdk4.1
 
 .PHONY : clean
 clean:
