--- GConf-2.16.0/gconf/gconftool.c.orig	2006-10-14 17:37:14.000000000 +0000
+++ GConf-2.16.0/gconf/gconftool.c	2006-10-14 17:39:27.000000000 +0000
@@ -963,6 +963,8 @@ main (int argc, char** argv)
       
       gconf_engine_unref (conf);
 
+      g_spawn_command_line_sync ("/usr/bin/killall -q -TERM " GCONF_SERVERDIR "/" GCONFD, NULL, NULL, NULL, NULL);
+
       return retval;
     }
 
@@ -975,6 +977,8 @@ main (int argc, char** argv)
       
       gconf_engine_unref (conf);
 
+      g_spawn_command_line_sync ("/usr/bin/killall -q -TERM " GCONF_SERVERDIR "/" GCONFD, NULL, NULL, NULL, NULL);
+
       return retval;
     }
 
