--- src/vivisect/setup.py	2020-09-07 12:41:26.499999800 +0200
+++ setup.py	2020-09-07 12:42:31.889999800 +0200
@@ -27,11 +27,11 @@
         ]
     },
     install_requires=[
-        'pyasn1==0.4.5',
-        'pyasn1-modules==0.2.4',
-        'cxxfilt==0.2.1',
-        'msgpack==1.0.0',
-        'pycparser==2.20',
+        'pyasn1',
+        'pyasn1-modules',
+        'cxxfilt',
+        'msgpack',
+        'pycparser',
     ],
     classifiers=[
         'Topic :: Security',
