--- src/crackmapexec/pyproject.toml	2022-06-26 21:03:39.523332280 +0200
+++ pyproject.toml	2022-06-26 21:04:20.763332280 +0200
@@ -27,7 +27,7 @@
 [tool.poetry.dependencies]
 python = "^3.7.0"
 requests = ">=2.27.1"
-beautifulsoup4 = ">=4.11,<5"
+beautifulsoup4 = ">=4.10,<5"
 lsassy = ">=3.1.1"
 termcolor = "^1.1.0"
 msgpack = "^1.0.0"
@@ -39,7 +39,7 @@
 dsinternals = "^1.2.4"
 xmltodict = "^0.12.0"
 terminaltables = "^3.1.0"
-aioconsole = "^0.3.3"
+aioconsole = ">=0.3.3"
 pywerview = "^0.4.0"
 aardwolf = "^0.0.8"
 
