--- src/crackmapexec/setup.py	2019-12-04 23:09:53.886666487 +0100
+++ setup.py	2019-12-04 23:10:40.529999809 +0100
@@ -21,15 +21,15 @@
     ]),
     install_requires=[
         'pycryptodomex',
-        'pyasn1>=0.1.8',
-        'gevent>=1.2.0',
-        'requests>=2.9.1',
-        'requests-ntlm>=0.3.0',
+        'pyasn1',
+        'gevent',
+        'requests',
+        'requests-ntlm',
         'bs4',
         'netaddr',
         'pyOpenSSL',
         'termcolor',
-        'msgpack-python',
+        'msgpack',
         'pylnk',
         'splinter',
         'paramiko',
