--- src/CrackMapExec/setup.py	2019-02-24 19:41:16.029999708 +0100
+++ setup.py	2019-02-24 19:40:57.136666370 +0100
@@ -20,16 +20,14 @@
         "cme", "cme.*"
     ]),
     install_requires=[
-        'pycrypto>=2.6',
-        'pyasn1>=0.1.8',
-        'gevent>=1.2.0',
-        'requests>=2.9.1',
-        'requests-ntlm>=0.3.0',
+        'pyasn1',
+        'gevent',
+        'requests',
+        'requests-ntlm',
         'bs4',
         'netaddr',
         'pyOpenSSL',
         'termcolor',
-        'msgpack-python',
         'pylnk',
         'splinter',
         'paramiko',
