--- setup.py.orig	2021-04-29 00:29:06.534159583 -0500
+++ setup.py	2021-04-29 00:30:07.834607854 -0500
@@ -121,20 +121,20 @@
          'data/sessiongopher/*']}
 
 install_requires = \
-['bs4>=0.0.1,<0.0.2',
- 'gevent>=1.2.0',
- 'impacket>=0.9.21,<0.10.0',
- 'lsassy>=2.1.2,<3.0.0',
- 'msgpack>=1.0.0,<2.0.0',
- 'neo4j>=4.1.1,<5.0.0',
- 'paramiko>=2.7.2,<3.0.0',
- 'pylnk3>=0.3.0,<0.4.0',
- 'pypsrp>=0.5.0,<0.6.0',
- 'requests-ntlm>=0.3.0',
- 'requests>=2.9.1',
- 'termcolor>=1.1.0,<2.0.0',
- 'terminaltables>=3.1.0,<4.0.0',
- 'xmltodict>=0.12.0,<0.13.0']
+['bs4',
+ 'gevent',
+ 'impacket',
+ 'lsassy',
+ 'msgpack',
+ 'neo4j',
+ 'paramiko',
+ 'pylnk3',
+ 'pypsrp',
+ 'requests-ntlm',
+ 'requests',
+ 'termcolor',
+ 'terminaltables',
+ 'xmltodict']
 
 entry_points = \
 {'console_scripts': ['cme = cme.crackmapexec:main',
