--- src/tlspretense/tlspretense.gemspec	2017-12-07 09:47:53.000000000 +0100
+++ tlspretense.gemspec	2019-08-24 15:13:05.486666198 +0200
@@ -1,4 +1,4 @@
-require './lib/tlspretense/version'
+require_relative './lib/tlspretense/version'

 Gem::Specification.new do |s|
   s.name = "tlspretense"
@@ -29,7 +29,7 @@
   s.add_development_dependency("rake", ["~> 12.3"])
   s.add_development_dependency("rspec", ["~> 3.7"])
   s.add_development_dependency("rdoc", ["~> 3.12"])
-  s.add_development_dependency("bundler", ["~> 1.0"])
+  s.add_development_dependency("bundler")
   s.add_development_dependency("simplecov", ["~> 0.7"])
 end

