--- src/tempomail/main.go	2021-04-10 22:04:44.939999834 +0200
+++ main.go	2021-04-10 22:05:24.446666500 +0200
@@ -5,8 +5,9 @@
 	"fmt"
 	"github.com/fatih/color"
 	"io/ioutil"
-	"io"
-	"path/filepath"
+	//"io"
+	//"path/filepath"
+	"flag"
 	"log"
 	"math/rand"
 	"net/http"
@@ -17,7 +18,6 @@
 	"strings"
 	"syscall"
 	"time"
-	"flag"
 )
 
 var cyan = color.New(color.FgCyan)
@@ -27,7 +27,7 @@
 
 var path string
 
-func setPath(){
+func setPath() {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
 		os.Mkdir(path, 0777)
 	}
@@ -84,7 +84,6 @@
 	return numOfEmails
 }
 
-
 func saveMail(name, domainOnly string) { //use this function to save emails
 
 	for k, _ := range m {
@@ -211,4 +210,4 @@
 		time.Sleep(5 * time.Second)
 	}
 
-}
\ No newline at end of file
+}
