--- src/cloudfail/socks.py	2022-03-27 01:12:15.039999354 +0100
+++ socks.py	2022-03-27 01:12:04.533332684 +0100
@@ -59,7 +59,6 @@
 from errno import EOPNOTSUPP, EINVAL, EAGAIN
 from io import BytesIO
 from os import SEEK_CUR
-from collections import Callable
 from base64 import b64encode
 
 PROXY_TYPE_SOCKS4 = SOCKS4 = 1
@@ -224,14 +223,6 @@
 for name in ("sendto", "send", "recvfrom", "recv"):
     method = getattr(_BaseSocket, name, None)
 
-    # Determine if the method is not defined the usual way
-    # as a function in the class.
-    # Python 2 uses __slots__, so there are descriptors for each method,
-    # but they are not functions.
-    if not isinstance(method, Callable):
-        _BaseSocket._savenames.append(name)
-        setattr(_BaseSocket, name, _makemethod(name))
-
 class socksocket(_BaseSocket):
     """socksocket([family[, type[, proto]]]) -> socket object
 
