diff -purN btscanner-2.1.orig/btscanner.xml btscanner-2.1/btscanner.xml
--- btscanner-2.1.orig/btscanner.xml	2004-11-01 04:56:41.000000000 -0700
+++ btscanner-2.1/btscanner.xml	2013-04-10 23:54:38.000000000 -0600
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE btscanner SYSTEM "file:///usr/local/etc/btscanner.dtd">
+<!DOCTYPE btscanner SYSTEM "file:///etc/btscanner/btscanner.dtd">
 <btscanner>
 <files>
 	<log>btscanner.log</log>
-	<oui>/usr/local/share/oui.txt</oui>
+	<oui>/usr/share/btscanner/oui.txt</oui>
 	<store>~/bts</store>
 </files>
 
diff -purN btscanner-2.1.orig/configure btscanner-2.1/configure
--- btscanner-2.1.orig/configure	2005-11-21 06:57:26.000000000 -0700
+++ btscanner-2.1/configure	2013-04-10 23:55:37.000000000 -0600
@@ -5308,14 +5308,14 @@ fi
 ac_cv_lib_xml2=ac_cv_lib_xml2_main
 
 
-# CFLAGS
-CFLAGS="$CFLAGS -Wall -pthread -Wshadow -Wbad-function-cast \
-  -Wformat -Wimplicit-function-dec -Wparentheses -Wsign-compare \
-  -Wstrict-prototypes -Wtrigraphs -Wundef -Wuninitialized \
-  -W -Wunused -Wformat-security -Wmissing-braces -Wbad-function-cast \
-  -Wcast-qual -falign-functions -falign-labels -falign-loops\
-  -pedantic -fstrict-aliasing -D_GNU_SOURCE -std=c99"
-#  -fstack-check -fbounds-check"
+## CFLAGS
+#CFLAGS="$CFLAGS -Wall -pthread -Wshadow -Wbad-function-cast \
+#  -Wformat -Wparentheses -Wsign-compare \
+#  -Wstrict-prototypes -Wtrigraphs -Wundef -Wuninitialized \
+#  -W -Wunused -Wformat-security -Wmissing-braces -Wbad-function-cast \
+#  -Wcast-qual -falign-functions -falign-labels -falign-loops\
+#  -pedantic -fstrict-aliasing -D_GNU_SOURCE -std=c99"
+##  -fstack-check -fbounds-check"
 
 # cfg
 echo "$as_me:$LINENO: checking the location of the config file" >&5
