--- settings.old	2019-01-21 06:10:56.597428654 +0000
+++ settings	2019-01-21 06:14:32.576752551 +0000
@@ -17,7 +17,7 @@
 # chose your own 'personalized prompt'.
 # -▷   -▶   '>'  ☭   ✿   ☯   ☠   ☆   ➽  ♛  ⊶ 
 ------------------
-PROMPT_DISPLAY=--▷ 
+PROMPT_DISPLAY=> 
 ------------------
 
 
@@ -53,7 +53,7 @@
 # WARNING: dont use .im because option[19] use it as internal_domain
 # accepted values are: pt | org | ir | etc..   
 ---------
-PREFIX=pt
+PREFIX=org
 ---------
 
 
@@ -76,7 +76,7 @@
 # just change the next values to point
 # to your apache2 webroot install folders.
 ---------------------
-AP_PATH=/var/www/html
+AP_PATH=/srv/http
 ---------------------
 
 
@@ -98,9 +98,9 @@
 # just change the next values to point
 # to your dsniff packet install folders.
 --------------------------
-TCP_KILL=/usr/sbin/tcpkill
-URL_SNARF=/usr/sbin/urlsnarf
-MSG_SNARF=/usr/sbin/msgsnarf
+TCP_KILL=/usr/bin/tcpkill
+URL_SNARF=/usr/bin/urlsnarf
+MSG_SNARF=/usr/bin/msgsnarf
 DRI_NET=/usr/bin/driftnet
 -------------------------
