diff -rupN libidn.orig/.git/FETCH_HEAD libidn/.git/FETCH_HEAD
--- libidn.orig/.git/FETCH_HEAD	1970-01-01 01:00:00.000000000 +0100
+++ libidn/.git/FETCH_HEAD	2018-10-16 20:38:19.078641485 +0200
@@ -0,0 +1,3 @@
+5675376089606c962223114173beb31e53e416f8	not-for-merge	branch 'libidn_dot-problem' of /home/alucryd/Packages/src/libidn
+c0374862fc911c88febfab36aedfceaa9e5d7d50	not-for-merge	branch 'master' of /home/alucryd/Packages/src/libidn
+1ca3304276592490b7786f32bb3ecb79fac292a5	not-for-merge	branch 'with-nfc-api' of /home/alucryd/Packages/src/libidn
Binary files libidn.orig/.git/index and libidn/.git/index differ
diff -rupN libidn.orig/Makefile.am libidn/Makefile.am
--- libidn.orig/Makefile.am	2018-10-16 20:36:07.230954718 +0200
+++ libidn/Makefile.am	2018-10-16 20:38:19.038641175 +0200
@@ -16,8 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = gl lib/gl lib po src doc gltests lib/gltests tests examples	\
-	java csharp
+SUBDIRS = gl lib/gl lib po
 
 EXTRA_DIST = cfg.mk maint.mk .clcopying
 EXTRA_DIST += COPYING.LESSERv2 COPYING.LESSERv3 COPYINGv2 COPYINGv3
