diff -up Net-Pcap-0.17/Pcap.pm.orig Net-Pcap-0.17/Pcap.pm
--- Net-Pcap-0.17/Pcap.pm.orig	2013-03-15 23:19:38.000000000 -0600
+++ Net-Pcap-0.17/Pcap.pm	2013-03-15 23:21:32.000000000 -0600
@@ -704,6 +704,10 @@ Close the savefile associated with the d
 
 =over
 
+=item B<pcap_list_datalinks($pcap)>
+
+Returns a list of link layer types supported by the given pcap descriptor.
+
 
 =item B<pcap_datalink($pcap)>
 
