diff -up Net-Pcap-0.17/Makefile.PL.orig Net-Pcap-0.17/Makefile.PL
--- Net-Pcap-0.17/Makefile.PL.orig	2013-03-15 21:00:34.000000000 -0600
+++ Net-Pcap-0.17/Makefile.PL	2013-03-15 21:00:47.000000000 -0600
@@ -63,39 +63,6 @@ Or get and install the WinPcap developer
         - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 REASON
 
-} else { # other systems (Unix)
-    warn <<"REASON" and exit unless have_library('pcap');
-        - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-You appear to lack the pcap(3) library. \a
-
-If it is installed in a non-standard location, please try setting the LIBS 
-and INC values on the command line.
-
-Or get the sources and install the pcap library from http://www.tcpdump.org/
-
-If you install the pcap library using a system package, make sure to also 
-install the corresponding -devel package, which contains the C headers needed 
-to compile this module. 
-        - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-REASON
-
-    warn <<"REASON" unless have_library('pcap', 'pcap_lib_version');
-        - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-You appear to have an old version of the pcap library. \a
-
-This module need a recent version of the pcap library in order to provide 
-access to all its features. You can still compile it with your old pcap 
-library but some functions won't be available, and trying to use them in 
-Perl programs will generate errors. Programs only using the old functions 
-should perform as previously. If not, don't hesitate to fill a bug. 
-
-You can get the latest sources of the pcap library at http://www.tcpdump.org/
-
-If you install the pcap library using a system package, make sure to also 
-install the corresponding -devel package, which contains the C headers needed 
-to compile this module. 
-        - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-REASON
 }
 
 # Now trying to detect which functions are actually available. 
