From 08c4198a971d40abd6293bb54b3336f081b2bb6e Mon Sep 17 00:00:00 2001
From: "requires.io" <support@requires.io>
Date: Wed, 19 Jul 2017 16:55:03 +0200
Subject: [PATCH] [requires.io] dependency update

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index e85ba74..08fe557 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -3,5 +3,5 @@ argh>=0.24.1
 pathtools>=0.1.2
 PyYAML>=3.10
 tornado>=3.2
-port_for==0.3.1
+port-for==0.4
 livereload>=2.3.0
