Index: xamarin-pkg-msbuild/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj
===================================================================
--- xamarin-pkg-msbuild.orig/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj
+++ xamarin-pkg-msbuild/mono/build/sdks_and_nugets/update_sdks_and_nugets.proj
@@ -1,6 +1,6 @@
 <Project DefaultTargets="DeploySdksAndNuGets">
     <Import Project="$(MSBuildThisFileDirectory)\..\common.props" />
-    <Import Project="$(MSBuildThisFileDirectory)\..\DotnetBitsVersions.props" />
+    <Import Project="$(MSBuildThisFileDirectory)\..\DotNetBitsVersions.props" />
 
     <PropertyGroup>
         <HostOSName Condition="'$(HostOSName)' == ''">osx</HostOSName>
Index: xamarin-pkg-msbuild/mono/build/update_bundled_bits.proj
===================================================================
--- xamarin-pkg-msbuild.orig/mono/build/update_bundled_bits.proj
+++ xamarin-pkg-msbuild/mono/build/update_bundled_bits.proj
@@ -1,5 +1,5 @@
 <Project DefaultTargets="FetchAndUpdateSdksAndNuGets">
-    <Import Project="$(MSBuildThisFileDirectory)\DotnetBitsVersions.props" />
+    <Import Project="$(MSBuildThisFileDirectory)\DotNetBitsVersions.props" />
 
     <Target Name="Build" DependsOnTargets="FetchAndUpdateSdksAndNuGets" />
 
