diff -aur Kraken.orig/DeltaLookup.cpp Kraken/DeltaLookup.cpp
--- Kraken.orig/DeltaLookup.cpp	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/DeltaLookup.cpp	2014-06-04 19:12:55.489937896 -0400
@@ -3,6 +3,7 @@
 #include "DeltaLookup.h"
 #include "NcqDevice.h"
 #include <iostream>
+#include <unistd.h>
 #include <stdio.h>
 #include <list>
 #include <sys/time.h>
diff -aur Kraken.orig/DeltaLookup.h Kraken/DeltaLookup.h
--- Kraken.orig/DeltaLookup.h	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/DeltaLookup.h	2014-06-04 19:13:21.639440733 -0400
@@ -4,6 +4,7 @@
 #include <string>
 #include <stdint.h>
 #include "NcqDevice.h"
+#include <unistd.h>
 
 class DeltaLookup {
 public:
diff -aur Kraken.orig/Fragment.cpp Kraken/Fragment.cpp
--- Kraken.orig/Fragment.cpp	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/Fragment.cpp	2014-06-04 19:13:40.759077225 -0400
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <unistd.h>
 #include "Fragment.h"
 #include "Kraken.h"
 #include "../a5_cpu/A5CpuStubs.h"
diff -aur Kraken.orig/Fragment.h Kraken/Fragment.h
--- Kraken.orig/Fragment.h	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/Fragment.h	2014-06-04 19:13:58.725402308 -0400
@@ -13,6 +13,7 @@
 #include "NcqDevice.h"
 #include "DeltaLookup.h"
 #include <stdint.h>
+#include <unistd.h>
 
 class Fragment : public NcqRequestor {
 public:
diff -aur Kraken.orig/Kraken.cpp Kraken/Kraken.cpp
--- Kraken.orig/Kraken.cpp	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/Kraken.cpp	2014-06-04 19:14:22.561615792 -0400
@@ -3,6 +3,7 @@
 #include <stdio.h>
 #include <string>
 #include <assert.h>
+#include <unistd.h>
 #include <string.h>
 #include <stdlib.h>
 #include "../a5_cpu/A5CpuStubs.h"
diff -aur Kraken.orig/Kraken.h Kraken/Kraken.h
--- Kraken.orig/Kraken.h	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/Kraken.h	2014-06-04 19:14:40.291278711 -0400
@@ -4,6 +4,7 @@
 #include "NcqDevice.h"
 #include "DeltaLookup.h"
 #include "Fragment.h"
+#include <unistd.h>
 
 #include <vector>
 #include <list>
diff -aur Kraken.orig/NcqDevice.cpp Kraken/NcqDevice.cpp
--- Kraken.orig/NcqDevice.cpp	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/NcqDevice.cpp	2014-06-04 19:15:09.130730405 -0400
@@ -4,6 +4,7 @@
 #include <assert.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <unistd.h>
 #include <pthread.h>
 #include <sys/mman.h>
 #include <stdio.h>
diff -aur Kraken.orig/NcqDevice.h Kraken/NcqDevice.h
--- Kraken.orig/NcqDevice.h	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/NcqDevice.h	2014-06-04 19:15:25.293756443 -0400
@@ -4,6 +4,7 @@
 #include <stdint.h>
 #include <queue>
 #include <semaphore.h>
+#include <unistd.h>
 #if defined(__APPLE__)
 #define mmap64          mmap
 #endif /* __APPLE__ */
diff -aur Kraken.orig/ServerCore.cpp Kraken/ServerCore.cpp
--- Kraken.orig/ServerCore.cpp	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/ServerCore.cpp	2014-06-04 19:15:39.516819364 -0400
@@ -1,5 +1,6 @@
 #include "ServerCore.h"
 #include <sys/socket.h>
+#include <unistd.h>
 #include <netinet/in.h>
 #include <string.h>
 #include <sys/time.h>
diff -aur Kraken.orig/ServerCore.h Kraken/ServerCore.h
--- Kraken.orig/ServerCore.h	2014-06-04 19:12:10.697456171 -0400
+++ Kraken/ServerCore.h	2014-06-04 19:16:04.573009652 -0400
@@ -8,6 +8,7 @@
 #include <pthread.h>
 #include <map>
 #include <semaphore.h>
+#include <unistd.h>
 
 using namespace std;
 
