--- src/litecli-1.2.0/setup.py	2018-12-01 04:42:51.000000000 +0100
+++ setup.py	2019-11-20 02:01:36.069997373 +0100
@@ -25,12 +25,12 @@
 readme = open_file("README.md")
 
 install_requirements = [
-    "click >= 4.1",
-    "Pygments >= 1.6",
-    "prompt_toolkit>=2.0.0,<2.1.0",
-    "sqlparse>=0.2.2,<0.3.0",
-    "configobj >= 5.0.5",
-    "cli_helpers[styles] >= 1.0.1",
+    "click",
+    "Pygments",
+    "prompt_toolkit",
+    "sqlparse",
+    "configobj",
+    "cli_helpers[styles]",
 ]
 
 
