--- src/litecli-1.7.0/setup.py	2022-01-11 17:23:08.000000000 +0100
+++ setup.py	2022-03-27 16:32:23.239999028 +0200
@@ -23,12 +23,12 @@
 readme = open_file("README.md")
 
 install_requirements = [
-    "click >= 4.1",
-    "Pygments>=1.6,<=2.11.1",
-    "prompt_toolkit>=3.0.3,<4.0.0",
+    "click",
+    "Pygments",
+    "prompt_toolkit",
     "sqlparse",
-    "configobj >= 5.0.5",
-    "cli_helpers[styles] >= 1.0.1",
+    "configobj",
+    "cli_helpers[styles]",
 ]
 
 
