--- a/pg-gvm-22.6.5/CMakeLists.txt
+++ b/pg-gvm-22.6.5/CMakeLists.txt
@@ -185,14 +185,14 @@
   configure_file("${CMAKE_SOURCE_DIR}/pg-gvm-make-dev-links.in" "${CMAKE_BINARY_DIR}/pg-gvm-make-dev-links" @ONLY)
   install(
     FILES "${CMAKE_BINARY_DIR}/pg-gvm-make-dev-links"
-    DESTINATION "${CMAKE_INSTALL_DEV_PREFIX}/sbin"
+    DESTINATION "${CMAKE_INSTALL_DEV_PREFIX}/bin"
     PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
   )
   install(FILES "${CMAKE_BINARY_DIR}/${CONTROLOUT}" DESTINATION "${CMAKE_INSTALL_DEV_PREFIX}/share/postgresql/extension")
   install(FILES "${CMAKE_BINARY_DIR}/${SQLOUT}" DESTINATION "${CMAKE_INSTALL_DEV_PREFIX}/share/postgresql/extension")
   file(GLOB SQL_UPDATE_FILES ${CMAKE_SOURCE_DIR}/sql/update/*.sql)
   install(FILES ${SQL_UPDATE_FILES} DESTINATION "${CMAKE_INSTALL_DEV_PREFIX}/share/postgresql/extension")
-  message(NOTICE "${CMAKE_PROJECT_NAME} installed with prefix. Run ${CMAKE_INSTALL_DEV_PREFIX}/sbin/pg-gvm-make-dev-links as root to create symlinks to this installation.")
+  message(NOTICE "${CMAKE_PROJECT_NAME} installed with prefix. Run ${CMAKE_INSTALL_DEV_PREFIX}/bin/pg-gvm-make-dev-links as root to create symlinks to this installation.")
 else (CMAKE_INSTALL_DEV_PREFIX)
   install(TARGETS ${CMAKE_PROJECT_NAME} DESTINATION "${PostgreSQL_EXTLIB_DIR}")
   install(FILES "${CMAKE_BINARY_DIR}/${CONTROLOUT}" DESTINATION "${PostgreSQL_SHARE_DIR}/extension")

