diff --exclude='*~' -urN amap-4.8.orig/amap-inc.h amap-4.8/amap-inc.h
--- amap-4.8.orig/amap-inc.h	2005-02-21 21:47:47.251638320 -0500
+++ amap-4.8/amap-inc.h	2005-02-21 21:48:27.774477912 -0500
@@ -20,7 +20,7 @@
 #include <errno.h>
 #include <sys/resource.h>
 #include <sys/wait.h>
-#include "pcre-3.9/pcre.h"
+#include <pcre.h>
 #ifdef OPENSSL
  #include <openssl/ssl.h>
  #include <openssl/err.h>
diff --exclude='*~' -urN amap-4.8.orig/configure amap-4.8/configure
--- amap-4.8.orig/configure	2005-02-21 21:47:47.240639992 -0500
+++ amap-4.8/configure	2005-02-21 21:48:41.379409648 -0500
@@ -13,10 +13,6 @@
   exit 0
 fi
 
-cd pcre-3.9 || exit 1
-./configure || exit 1
-cd ..
-
 echo
 echo
 echo "Starting amap auto configuration ..."
diff --exclude='*~' -urN amap-4.8.orig/Makefile.am amap-4.8/Makefile.am
--- amap-4.8.orig/Makefile.am	2005-02-21 21:47:47.242639688 -0500
+++ amap-4.8/Makefile.am	2005-02-21 21:49:16.070135856 -0500
@@ -1,5 +1,5 @@
 CC=gcc
-LIB=-lpcre -L./pcre-3.9/ -I./pcre-3.9/
+LIB=-lpcre
 OPT=-O2
 OPT_DEBUG=-ggdb -Wall -lefence
 BINDIR=/bin
@@ -7,7 +7,7 @@
 MANDIR=/man/man1
 SCRIPTS=appdefs.trig appdefs.resp appdefs.rpc
 MANPAGE=amap.1
-all:	pcre strip
+all:	amap amapcrap
 
 pcre:
 	cd pcre-3.9 && make
