--- src/botocore-1.13.14/setup.py	2019-11-08 20:12:56.000000000 +0100
+++ setup.py	2019-11-15 23:35:31.346664982 +0100
@@ -23,31 +23,31 @@
     raise RuntimeError("Unable to find version string.")
 
 
-requires = ['jmespath>=0.7.1,<1.0.0',
-            'docutils>=0.10,<0.16']
+requires = ['jmespath',
+            'docutils']
 
 
 if sys.version_info[:2] == (2, 6):
     # For python2.6 we have a few other dependencies.
     # First we need an ordered dictionary so we use the
     # 2.6 backport.
-    requires.append('ordereddict==1.1')
+    requires.append('ordereddict')
     # Then we need simplejson.  This is because we need
     # a json version that allows us to specify we want to
     # use an ordereddict instead of a normal dict for the
     # JSON objects.  The 2.7 json module has this.  For 2.6
     # we need simplejson.
-    requires.append('simplejson==3.3.0')
-    requires.append('python-dateutil>=2.1,<2.7.0')
+    requires.append('simplejson')
+    requires.append('python-dateutil')
 else:
-    requires.append('python-dateutil>=2.1,<2.8.1')
+    requires.append('python-dateutil')
 
 if sys.version_info[:2] == (2, 6):
-    requires.append('urllib3>=1.20,<1.24')
+    requires.append('urllib3')
 elif sys.version_info[:2] == (3, 3):
-    requires.append('urllib3>=1.20,<1.23')
+    requires.append('urllib3')
 else:
-    requires.append('urllib3>=1.20,<1.26')
+    requires.append('urllib3')
 
 
 setup(
@@ -65,8 +65,8 @@
     install_requires=requires,
     extras_require={
         ':python_version=="2.6"': [
-            'ordereddict==1.1',
-            'simplejson==3.3.0',
+            'ordereddict',
+            'simplejson',
         ]
     },
     license="Apache License 2.0",
