--- src/domi-owned/domi_owned/bruteforce.py	2015-12-23 08:16:11.816701543 +0100
+++ bruteforce.py	2015-12-23 08:17:07.903064363 +0100
@@ -21,10 +21,13 @@
 import re
 import requests
 import time
+import urllib3
+import warnings
 
 from domi_owned import utility
 
-requests.packages.urllib3.disable_warnings()
+urllib3.disable_warnings()
+warnings.filterwarnings(action='ignore', module='requests')
 
 # Get authentication type for names.nsf
 def get_auth_type(target, header, usernames, password):
