--- src/domi-owned/domi_owned/fingerprint.py	2015-12-22 12:14:05.587442871 +0100
+++ fingerprint.py	2015-12-22 12:16:14.600077283 +0100
@@ -19,16 +19,21 @@
 # SOFTWARE.
 import re
 import requests
+import warnings
+import urllib3
 
 from domi_owned import utility
 
-requests.packages.urllib3.disable_warnings()
+# disable warnings
+urllib3.disable_warnings()
+warnings.filterwarnings(action='ignore', module='requests')
+
 
 # Get Domino version
 def fingerprint(target, header):
 	domino_version = None
 
-	version_files = ['download/filesets/l_LOTUS_SCRIPT.inf', 
+	version_files = ['download/filesets/l_LOTUS_SCRIPT.inf',
 			'download/filesets/n_LOTUS_SCRIPT.inf',
 			'download/filesets/l_SEARCH.inf',
 			'download/filesets/n_SEARCH.inf',
