--- src/domi-owned/domi_owned/quickconsole.py	2015-12-23 08:13:15.300991144 +0100
+++ quickconsole.py	2015-12-23 08:14:12.274015828 +0100
@@ -21,10 +21,13 @@
 import re
 import requests
 import urllib
+import urllib3
+import warnings
 
 from domi_owned import utility
 
-requests.packages.urllib3.disable_warnings()
+urllib3.disable_warnings()
+warnings.filterwarnings(action='ignore', module='requests')
 
 class Interactive(cmd.Cmd, object):
 	"""Interact with Domino Quick Console through web requests"""
