diff --git a/intrace-1.5.old/Makefile b/intrace-1.5/Makefile
index 4d1e28b..dcd6d5b 100644
--- a/intrace-1.5.old/Makefile
+++ b/intrace-1.5/Makefile
@@ -19,7 +19,7 @@
 CC = gcc
 CFLAGS = -fPIC -O3 -g -ggdb -c -std=gnu99 -I. -pedantic \
 		 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
-		 -Wall -Werror -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized -Wcast-align \
+		 -Wall -Werror -Wimplicit -Wunused -Wcomment -Wchar-subscripts -Wuninitialized \
 		 -Wreturn-type -Wpointer-arith -Wbad-function-cast
 
 LD = gcc
