diff --git a/chromium/content/public/browser/BUILD.gn b/chromium/content/public/browser/BUILD.gn
index 7cf82a4aacf..0e0dda2c3ca 100644
--- a/chromium/content/public/browser/BUILD.gn
+++ b/chromium/content/public/browser/BUILD.gn
@@ -469,6 +469,7 @@ jumbo_source_set("browser_sources") {
     "//build:chromeos_buildflags",
     "//cc",
     "//components/viz/host",
+    "//components/spellcheck:buildflags",
     "//content/browser",  # Must not be public_deps!
     "//device/fido",
     "//gpu",
