--- src/gui/kernel/qkde.cpp.orig	2015-05-27 11:42:02.507129332 +0200
+++ src/gui/kernel/qkde.cpp	2015-05-27 11:43:26.182875729 +0200
@@ -63,7 +63,7 @@
         kdeHomePath = QString::fromLocal8Bit(qgetenv("KDEHOME"));
         if (kdeHomePath.isEmpty()) {
             QDir homeDir(QDir::homePath());
-            QString kdeConfDir(QLatin1String("/.kde"));
+            QString kdeConfDir(QLatin1String("/.kde4"));
             if (4 == X11->desktopVersion && homeDir.exists(QLatin1String(".kde4")))
             kdeConfDir = QLatin1String("/.kde4");
             kdeHomePath = QDir::homePath() + kdeConfDir;
