--- Intersect-2.5.orig/Create.py	2014-01-14 15:22:03.295699788 +0000
+++ Intersect-2.5/Create.py	2014-01-14 17:33:49.529303557 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Intersect2 Payload Generation Utility
 # copyright 2012 - ohdae
@@ -17,8 +17,6 @@
 global ModulesDir
 global CustomDir
 global PayloadTemplate
-global currentloc
-global tab_complete
 global BuildLog
 global now
 
@@ -33,11 +31,10 @@
 if tab_complete == True:
     readline.parse_and_bind("tab: complete")
 
-currentloc = os.getcwd()
-ModulesDir = (currentloc+"/src/Modules/local/Standard/")
-CustomDir = (currentloc+"/src/Modules/local/Custom/")
-PayloadTemplate = (currentloc+"/src/Templates/local/stock-template")
-BuildLog = (currentloc+"/Logs/build_log")
+ModulesDir = ("/usr/share/Intersect/src/Modules/local/Standard/")
+CustomDir = ("/usr/share/Intersect/src/Modules/local/Custom/")
+PayloadTemplate = ("/usr/share/Intersect/src/Templates/local/stock-template")
+BuildLog = ("/usr/share/Intersect/Logs/build_log")
 
 logging.basicConfig(filename=BuildLog, level=logging.INFO, format='%(asctime)s %(message)s')
 
@@ -245,7 +242,8 @@
 ** To view a full list of all available commands type :help.
 The command :quit will return you to the main menu.\n""")
 
-      desired_modules = []
+      desired_modules = os.listdir(ModulesDir)
+      desired_modules += os.listdir(CustomDir)
       logging.info("New Build Process Started.")
 
       while 1:
@@ -352,7 +350,7 @@
           )
 
   def globalconfig(self, modules):
-      writeglobals = (currentloc+"/Junk/globals")
+      writeglobals = ("/usr/share/Intersect/Junk/globals")
       globalstemp = open(writeglobals, "a")
 
       globalstemp.write("\ndef globalvars():")
@@ -466,10 +464,10 @@
       if os.path.exists(PayloadTemplate) is True:
           print("Enter a name for your Intersect script. The finished script will be placed in the Scripts directory. Do not include Python file extension.")
           name = raw_input("%s " % (self.header))
-          script = (currentloc+"/Scripts/%s.py" % name)
+          script = ("/usr/share/Intersect/Scripts/%s.py" % name)
 
           if name == "":
-              script = (currentloc+"/Scripts/Intersect.py")
+              script = ("/usr/share/Intersect/Scripts/Intersect.py")
               shutil.copy2(PayloadTemplate, script)
               newpayload = open(script, "a")
               logging.info("Script named: Intersect.py")
@@ -495,7 +493,7 @@
   def CombineFeatures(self, moduleList):
 
 
-      savedglobals = (currentloc+"/Junk/globals")
+      savedglobals = ("/usr/share/Intersect/Junk/globals")
       writeglobals = open(savedglobals, "r")
       for lines in writeglobals.readlines():
           newpayload.write(lines)
@@ -526,7 +524,7 @@
 
       
   def MakeUsage(self, moduleList):  # Clean up this function
-      usage = (currentloc+"/Junk/usage")
+      usage = ("/usr/share/Intersect/Junk/usage")
       descriptions = []
 
       for module in moduleList:
@@ -593,7 +591,7 @@
     
       
       newpayload = open(script, "a")
-      writeopts = (currentloc+"/Junk/writeopts") # give name to getopt temporary file # 
+      writeopts = ("/usr/share/Intersect/Junk/writeopts") # give name to getopt temporary file # 
       newopts = open(writeopts, "a") # open temporary getopt file to append getopt functions into
 
       
