diff -Naur mono-tools-4.2.orig/Makefile.am mono-tools-4.2/Makefile.am
--- mono-tools-4.2.orig/Makefile.am	2015-12-31 22:35:31.000000000 +0100
+++ mono-tools-4.2/Makefile.am	2021-04-21 19:03:35.255389645 +0200
@@ -1,5 +1,5 @@
-SUBDIRS= po docbrowser asn1view create-native-map ilcontrast gui-compare \
-	gendarme mperfmon gsharp webdoc minvoke Mono.Profiler
+SUBDIRS= po asn1view create-native-map ilcontrast gui-compare \
+	mperfmon gsharp minvoke Mono.Profiler
 
 #capture aclocal flags for autoreconf
 ACLOCAL_AMFLAGS=$(ACLOCAL_FLAGS)
diff -Naur mono-tools-4.2.orig/configure.in mono-tools-4.2/configure.in
--- mono-tools-4.2.orig/configure.in	2015-12-31 22:35:31.000000000 +0100
+++ mono-tools-4.2/configure.in	2021-04-21 18:14:30.880246048 +0200
@@ -42,15 +42,12 @@
 AC_SUBST(GACUTIL)
 
 
-PKG_CHECK_MODULES(NUNIT, mono-nunit)
-AC_SUBST(NUNIT_LIBS)
-
 PKG_CHECK_MODULES(MONODOC, monodoc)
 AC_SUBST(MONODOC_LIBS)
 AC_PATH_PROG(MDOC, mdoc, no)
 AC_SUBST(MDOC)
 
-PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0 glade-sharp-2.0)
+PKG_CHECK_MODULES(GTK_SHARP, gtk-sharp-2.0)
 AC_SUBST(GTK_SHARP_LIBS)
 
 CSHARP_FLAGS=""
diff -Naur mono-tools-4.2.orig/gui-compare/Makefile.am mono-tools-4.2/gui-compare/Makefile.am
--- mono-tools-4.2.orig/gui-compare/Makefile.am	2015-12-31 22:35:31.000000000 +0100
+++ mono-tools-4.2/gui-compare/Makefile.am	2021-04-21 18:19:02.091160776 +0200
@@ -51,7 +51,7 @@
 
 EXTRA_DIST = $(gui_compare_sources) $(gui_compare_resources) gui-compare.mds gui-compare.mdp  gui-compare.in
 
-packages = -pkg:gtk-sharp-2.0 -pkg:glib-sharp-2.0 -pkg:glade-sharp-2.0 -r:$(CECIL_ASM)
+packages = -pkg:gtk-sharp-2.0 -pkg:glib-sharp-2.0 -r:$(CECIL_ASM)
 assemblies = -r:System -r:System.Core -r:Mono.Posix -r:System.Xml
 ress= $(foreach res,$(gui_compare_resources), $(addprefix -resource:,$(res)),$(notdir $(res)))
 
