diff -r -U 3 a/btscanner.xml b/btscanner.xml
--- a/btscanner.xml	2004-11-01 17:26:41.000000000 +0530
+++ b/btscanner.xml	2008-04-24 00:43:44.927640836 +0530
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE btscanner SYSTEM "file:///usr/local/etc/btscanner.dtd">
+<!DOCTYPE btscanner SYSTEM "file:///etc/btscanner/btscanner.dtd">
 <btscanner>
 <files>
 	<log>btscanner.log</log>
-	<oui>/usr/local/share/oui.txt</oui>
+	<oui>/usr/share/btscanner/oui.txt</oui>
 	<store>~/bts</store>
 </files>
 
diff -r -U 3 a/configure b/configure
--- a/configure	2008-04-24 00:19:55.321329170 +0530
+++ b/configure	2008-04-24 00:29:38.431296231 +0530
@@ -5310,7 +5310,7 @@
 
 # CFLAGS
 CFLAGS="$CFLAGS -Wall -pthread -Wshadow -Wbad-function-cast \
-  -Wformat -Wimplicit-function-dec -Wparentheses -Wsign-compare \
+  -Wformat -Wparentheses -Wsign-compare \
   -Wstrict-prototypes -Wtrigraphs -Wundef -Wuninitialized \
   -W -Wunused -Wformat-security -Wmissing-braces -Wbad-function-cast \
   -Wcast-qual -falign-functions -falign-labels -falign-loops\
