--- src/flake8-3.7.9/setup.py	2019-10-28 18:30:21.000000000 +0100
+++ setup.py	2019-12-01 10:48:18.583333055 +0100
@@ -19,10 +19,10 @@
     # http://flake8.pycqa.org/en/latest/faq.html#why-does-flake8-use-ranges-for-its-dependencies
     # And in which releases we will update those ranges here:
     # http://flake8.pycqa.org/en/latest/internal/releases.html#releasing-flake8
-    "entrypoints >= 0.3.0, < 0.4.0",
-    "pyflakes >= 2.1.0, < 2.2.0",
-    "pycodestyle >= 2.5.0, < 2.6.0",
-    "mccabe >= 0.6.0, < 0.7.0",
+    "entrypoints",
+    "pyflakes",
+    "pycodestyle",
+    "mccabe",
 ]
 
 extras_require = {
