--- src/birp/py3270wrapper.py	2020-05-14 04:17:00.173331634 +0200
+++ py3270wrapper.py	2020-05-14 04:18:02.153331621 +0200
@@ -65,7 +65,7 @@
 					whine('Read error encountered, assuming host is slow, increasing delay by 1s to: ' + str(self.delay),kind='warn')
 					return False
 		return False
-	
+
 	# Get the current x3270 cursor position
 	def get_pos(self):
 		results = self.exec_command('Query(Cursor)')
@@ -83,7 +83,7 @@
 		s3270App.executable = 's3270'
 elif platform.system() == 'Linux':
 	class WrappedEmulator(EmulatorIntermediate):
-		x3270App.executable = './x3270'
+		x3270App.executable = '/usr/bin/x3270'
 		s3270App.executable = 's3270'
 elif platform.system() == 'Windows':
 	class WrappedEmulator(EmulatorIntermediate):
