diff --git a/httpscreenshot.py b/httpscreenshot.py
index 6794d8d..dc5a5e0 100644
--- a/httpscreenshot.py
+++ b/httpscreenshot.py
@@ -26,6 +26,7 @@ from PIL import Image, ImageDraw, ImageFont
 from pyvirtualdisplay import Display
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.firefox.options import Options as FOptions
 from selenium.webdriver.chrome.service import Service
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
 from webdriver_manager.chrome import ChromeDriverManager
@@ -185,12 +186,10 @@ def setupBrowserProfile(headless, proxy, browserType):
 
                 browser = webdriver.Chrome(service=service, options=coptions)
             else:
-                capabilities = DesiredCapabilities.FIREFOX
-                capabilities['acceptSslCerts'] = True
                 fp = webdriver.FirefoxProfile()
                 fp.set_preference("webdriver.accept.untrusted.certs", True)
                 fp.set_preference("security.enable_java", False)
-                fp.set_preference("webdriver.load.strategy", "fast");
+                fp.set_preference("webdriver.load.strategy", "fast")
                 if (proxy is not None):
                     proxyItems = proxy.split(":")
                     fp.set_preference("network.proxy.socks", proxyItems[0])
@@ -198,12 +197,12 @@ def setupBrowserProfile(headless, proxy, browserType):
                     fp.set_preference("network.proxy.type", 1)
 
                 fireFoxOptions = webdriver.FirefoxOptions()
+                fireFoxOptions.profile = fp
+                fireFoxOptions.set_capability('acceptInsecureCerts', True)
                 if headless:
                     fireFoxOptions.headless = True
 
-                browser = webdriver.Firefox(firefox_profile=fp,
-                                            capabilities=capabilities,
-                                            options=fireFoxOptions)
+                browser = webdriver.Firefox(options=fireFoxOptions)
                 browser.set_window_size(1024, 768)
 
         except Exception as e:
