--- a/SConstruct
+++ b/SConstruct
@@ -44,7 +44,7 @@
 # Documentation
 #  This only needs to be run during the release/packaging process
 man_fixup = "|sed 's/.SH DESCRIPTION/\\n.SH DESCRIPTION/'"
-man_builder = Builder(action='docbook2x-man --to-stdout $SOURCE'
+man_builder = Builder(action='docbook2man --to-stdout $SOURCE'
                       + man_fixup + '| gzip -9 > $TARGET',
                       suffix = '.gz',
                       src_suffix = '.docbook')
@@ -63,1 +63,1 @@
-mandir     = os.environ.get('MANDIR', prefix + 'man')
+mandir     = os.environ.get('MANDIR', prefix + 'share/man')
@@ -79,2 +79,0 @@
-if os.getuid() == 0 and destdir == '':
-   env.AddPostAction(libinstall, 'ldconfig')
