--- SConstruct.old	2012-02-24 23:07:42.759585143 +0100
+++ SConstruct	2012-02-24 23:10:08.082282514 +0100
@@ -34,7 +34,7 @@
 # Documentation
 #  This only needs to be run during the release/packaging process
 man_fixup = "|sed 's/.SH DESCRIPTION/\\n.SH DESCRIPTION/'"
-man_builder = Builder(action='docbook2x-man --to-stdout $SOURCE'
+man_builder = Builder(action='docbook2man --to-stdout $SOURCE'
                       + man_fixup + '| gzip -9 > $TARGET',
                       suffix = '.gz',
                       src_suffix = '.docbook')
@@ -52,14 +52,13 @@
 install_items = [prefix+'bin',
                  prefix+'lib', 
                  prefix+'include/regfi',
-                 prefix+'man']
+                 prefix+'share/man']
 
 env.Install(prefix+'bin', [reglookup, reglookup_recover, 'bin/reglookup-timeline'])
 libinstall = env.Install(prefix+'lib', [libregfi, libregfi_static])
 env.Install(prefix+'include/regfi', Glob('include/*.h'))
-env.Install(prefix+'man/man1', [man_reglookup, man_reglookup_recover,
+env.Install(prefix+'share/man/man1', [man_reglookup, man_reglookup_recover,
                                 man_reglookup_timeline])
-env.AddPostAction(libinstall, 'ldconfig')
 
 if sys.version_info[0] == 2:
    install_items.append('pyregfi2-install.log')
