--- src/Google-Search-API-1.1.14/setup.py	2016-12-27 23:25:20.000000000 +0100
+++ setup.py	2018-08-12 11:36:37.033859056 +0200
@@ -12,9 +12,6 @@
 with open("requirements.txt") as f:
     requirements = [req.strip() for req in f.readlines()]

-with open("test_requirements.txt") as f:
-    test_requirements = [req.strip() for req in f.readlines()]
-
 setup(name='Google-Search-API',
       version='1.1.14',
       url='https://github.com/abenassi/Google-Search-API',
@@ -41,7 +38,5 @@
           'Programming Language :: Python :: 2.7'
       ],
       setup_requires=['nose>=1.0'],
-      test_suite='nose.collector',
-      tests_require=test_requirements
       # **r.requirements
       )
