diff -u --color -Naur sphinxcontrib-napoleon-0.7.orig/sphinxcontrib/napoleon/docstring.py sphinxcontrib-napoleon-0.7/sphinxcontrib/napoleon/docstring.py
--- sphinxcontrib-napoleon-0.7.orig/sphinxcontrib/napoleon/docstring.py	2018-09-23 15:41:55.000000000 +0200
+++ sphinxcontrib-napoleon-0.7/sphinxcontrib/napoleon/docstring.py	2022-01-05 09:29:00.555246571 +0100
@@ -13,7 +13,10 @@
 
 import inspect
 import re
-from collections import Callable
+try:
+    from collections.abc import Callable
+except ImportError:
+    from collections import Callable
 from functools import partial
 
 from six import string_types, u
