--- vinetto
+++ vinetto
@@ -32,7 +32,7 @@
 import sys
 import os
 import StringIO
-import md5
+import hashlib
 import vinreport
 
 from optparse import OptionParser
@@ -164,7 +164,7 @@
     if htmlrep == True:
         report = vinreport.HtRep(tDBfname, outputdir, getencodings(), __version__ + " (r" + \
                                  __revision__[11:-2] +")")
-        md5tDB = md5.new(thumbsDB).hexdigest()
+        md5tDB = hashlib.md5(thumbsDB).hexdigest()
         report.SetFileSection(longueur, md5tDB)
 
 # ----------------------------------------------------------------------------- 
