--- src/CANToolz/setup.py	2016-05-31 13:05:11.674907948 +0200
+++ setup.py	2016-05-31 13:15:51.224776542 +0200
@@ -1,13 +1,6 @@
 from setuptools import setup
-import pip
 import sys
 
-if len(sys.argv) >= 2 and sys.argv[1] == 'install':
-    pip.main(['install', 'pyserial'])
-    pip.main(['install', 'numpy'])
-    pip.main(['install', 'mido'])
-
-
 setup(
     name='CANToolz',
     version='3.3.3',
@@ -25,4 +18,4 @@
     description='Framework and library for black-box analysis and reverse engineering of Controller Area Network (CAN)'
 
 
-)
\ No newline at end of file
+)
