--- src/CANToolz/setup.py	2018-09-04 18:47:58.638291593 +0200
+++ setup.py	2018-09-04 18:48:25.388291590 +0200
@@ -1,17 +1,6 @@
 from setuptools import setup
-import pip
 import sys
 
-if len(sys.argv) >= 2 and sys.argv[1] == 'install':
-    #pip.main(['install', 'neurolab'])
-    pip.main(['install', 'pyserial'])
-    pip.main(['install', 'numpy'])
-    pip.main(['install', 'mido'])
-    #pip.main(['install', 'scipy'])
-    #pip.main(['install', 'pybrain'])
-    pip.main(['install', 'bitstring'])
-
-
 setup(
     name='CANToolz',
     version='3.6.0',
@@ -29,4 +18,4 @@
     description='Framework and library for black-box analysis and reverse engineering of Controller Area Network (CAN)'
 
 
-)
\ No newline at end of file
+)
