diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ad1545..4c83847 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,7 +142,6 @@ add_executable(fridatest ${fridatest_SOURCES})
 qt5_use_modules(fridatest Widgets)
 
 #ADD_SUBDIRECTORY(/usr/src/gtest gtest)
-ADD_SUBDIRECTORY(/usr/src/gmock gmock)
 enable_testing()
 add_test(NAME fridatest COMMAND fridatest ${CMAKE_SOURCE_DIR}/testdata)
 target_link_libraries(fridatest
diff --git a/src/core/InformationManager.cxx b/src/core/InformationManager.cxx
index 604425e..c97beeb 100644
--- a/src/core/InformationManager.cxx
+++ b/src/core/InformationManager.cxx
@@ -7,8 +7,8 @@
 #include "core/events/ChangeCommentEvent.hxx"
 
 #include "qt.hxx"
-#include <quazip/quazip.h>
-#include <quazip/quazipfile.h>
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 #include <QTemporaryFile>
 
