--- Makefile	2018-10-14 15:00:17.296116551 -0500
+++ Makefile2	2018-10-14 15:00:53.759758411 -0500
@@ -96,7 +96,7 @@
 	@echo -n 'Installing scripts into $(DESTDIR)$(SBINDIR)... '
 	@install -d -m 755 '$(DESTDIR)$(SBINDIR)' 2>/dev/null
 	@for i in $(SBINFILES) ; do \
-		install -m 750 "sbin/$$i.patched" "$(DESTDIR)$(SBINDIR)/$$i" ; \
+		install -m 755 "sbin/$$i.patched" "$(DESTDIR)$(SBINDIR)/$$i" ; \
 	done
 	@echo 'ok.'
 
