diff --git a/tests/unit/ec2/test_volume.py b/tests/unit/ec2/test_volume.py
index 81d7f552..6833c763 100644
--- a/tests/unit/ec2/test_volume.py
+++ b/tests/unit/ec2/test_volume.py
@@ -3,6 +3,7 @@ from tests.compat import mock, unittest
 from boto.ec2.snapshot import Snapshot
 from boto.ec2.tag import Tag, TagSet
 from boto.ec2.volume import Volume, AttachmentSet, VolumeAttribute
+from boto.resultset import ResultSet
 
 
 class VolumeTests(unittest.TestCase):
@@ -52,8 +53,7 @@ class VolumeTests(unittest.TestCase):
         self.assertEqual(retval, tag_set)
 
     @mock.patch("boto.ec2.volume.TaggedEC2Object.startElement")
-    @mock.patch("boto.resultset.ResultSet")
-    def test_startElement_with_name_tagSet_calls_ResultSet(self, ResultSet, startElement):
+    def test_startElement_with_name_tagSet_calls_ResultSet(self, startElement):
         startElement.return_value = None
         result_set = mock.Mock(ResultSet([("item", Tag)]))
         volume = Volume()
