diff -aur boto-2.49.0.20190327/requirements.txt boto-2.49.0.20190327.new/requirements.txt
--- boto-2.49.0.20190327/requirements.txt	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/requirements.txt	2022-11-12 17:45:21.295229358 +0100
@@ -7,4 +7,3 @@
 paramiko>=1.10.0
 PyYAML>=3.10
 coverage==3.7.1
-mock==1.0.1
diff -aur boto-2.49.0.20190327/tests/integration/s3/test_bucket.py boto-2.49.0.20190327.new/tests/integration/s3/test_bucket.py
--- boto-2.49.0.20190327/tests/integration/s3/test_bucket.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/integration/s3/test_bucket.py	2022-11-12 17:45:37.645593701 +0100
@@ -26,7 +26,7 @@
 Some unit tests for the S3 Bucket
 """
 
-from mock import patch, Mock
+from unittest.mock import patch, Mock
 import unittest
 import time
 
diff -aur boto-2.49.0.20190327/tests/integration/s3/test_https_cert_validation.py boto-2.49.0.20190327.new/tests/integration/s3/test_https_cert_validation.py
--- boto-2.49.0.20190327/tests/integration/s3/test_https_cert_validation.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/integration/s3/test_https_cert_validation.py	2022-11-12 17:57:21.673791221 +0100
@@ -38,7 +38,7 @@
 import os
 import ssl
 import unittest
-import mock
+from unittest import mock
 
 from nose.plugins.attrib import attr
 
diff -aur boto-2.49.0.20190327/tests/integration/s3/test_multipart.py boto-2.49.0.20190327.new/tests/integration/s3/test_multipart.py
--- boto-2.49.0.20190327/tests/integration/s3/test_multipart.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/integration/s3/test_multipart.py	2022-11-12 17:57:07.306831428 +0100
@@ -37,7 +37,7 @@
 import time
 from boto.compat import StringIO
 
-import mock
+from unittest import mock
 
 import boto
 from boto.s3.connection import S3Connection
diff -aur boto-2.49.0.20190327/tests/unit/auth/test_stsanon.py boto-2.49.0.20190327.new/tests/unit/auth/test_stsanon.py
--- boto-2.49.0.20190327/tests/unit/auth/test_stsanon.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/auth/test_stsanon.py	2022-11-12 17:45:37.652260517 +0100
@@ -20,7 +20,7 @@
 # IN THE SOFTWARE.
 #
 import copy
-from mock import Mock
+from unittest.mock import Mock
 from tests.unit import unittest
 
 from boto.auth import STSAnonHandler
diff -aur boto-2.49.0.20190327/tests/unit/cloudformation/test_connection.py boto-2.49.0.20190327.new/tests/unit/cloudformation/test_connection.py
--- boto-2.49.0.20190327/tests/unit/cloudformation/test_connection.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/cloudformation/test_connection.py	2022-11-12 17:45:37.655593923 +0100
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 import unittest
 from datetime import datetime
-from mock import Mock
+from unittest.mock import Mock
 
 from tests.unit import AWSMockServiceTestCase
 from boto.cloudformation.connection import CloudFormationConnection
diff -aur boto-2.49.0.20190327/tests/unit/cloudsearch/test_document.py boto-2.49.0.20190327.new/tests/unit/cloudsearch/test_document.py
--- boto-2.49.0.20190327/tests/unit/cloudsearch/test_document.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/cloudsearch/test_document.py	2022-11-12 17:45:37.655593923 +0100
@@ -2,7 +2,7 @@
 
 from tests.unit import unittest
 from httpretty import HTTPretty
-from mock import MagicMock
+from unittest.mock import MagicMock
 
 import json
 
diff -aur boto-2.49.0.20190327/tests/unit/cloudsearch2/test_document.py boto-2.49.0.20190327.new/tests/unit/cloudsearch2/test_document.py
--- boto-2.49.0.20190327/tests/unit/cloudsearch2/test_document.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/cloudsearch2/test_document.py	2022-11-12 17:45:37.655593923 +0100
@@ -4,7 +4,7 @@
 
 from tests.unit import unittest, AWSMockServiceTestCase
 from httpretty import HTTPretty
-from mock import MagicMock
+from unittest.mock import MagicMock
 
 import json
 
diff -aur boto-2.49.0.20190327/tests/unit/cloudsearchdomain/test_cloudsearchdomain.py boto-2.49.0.20190327.new/tests/unit/cloudsearchdomain/test_cloudsearchdomain.py
--- boto-2.49.0.20190327/tests/unit/cloudsearchdomain/test_cloudsearchdomain.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/cloudsearchdomain/test_cloudsearchdomain.py	2022-11-12 17:56:32.289449217 +0100
@@ -1,6 +1,6 @@
 #!/usr/bin env python
 import json
-import mock
+from unittest import mock
 from tests.unit import AWSMockServiceTestCase
 from boto.cloudsearch2.domain import Domain
 from boto.cloudsearch2.layer1 import CloudSearchConnection
diff -aur boto-2.49.0.20190327/tests/unit/dynamodb/test_layer2.py boto-2.49.0.20190327.new/tests/unit/dynamodb/test_layer2.py
--- boto-2.49.0.20190327/tests/unit/dynamodb/test_layer2.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/dynamodb/test_layer2.py	2022-11-12 17:45:37.658927332 +0100
@@ -22,7 +22,7 @@
 #
 
 from tests.unit import unittest
-from mock import Mock
+from unittest.mock import Mock
 
 from boto.dynamodb.layer2 import Layer2
 from boto.dynamodb.table import Table, Schema
diff -aur boto-2.49.0.20190327/tests/unit/ec2/test_connection.py boto-2.49.0.20190327.new/tests/unit/ec2/test_connection.py
--- boto-2.49.0.20190327/tests/unit/ec2/test_connection.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/ec2/test_connection.py	2022-11-12 17:45:37.662260738 +0100
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 from datetime import datetime, timedelta
-from mock import MagicMock, Mock
+from unittest.mock import MagicMock, Mock
 from tests.unit import unittest
 from tests.unit import AWSMockServiceTestCase
 
diff -aur boto-2.49.0.20190327/tests/unit/ec2containerservice/test_connection.py boto-2.49.0.20190327.new/tests/unit/ec2containerservice/test_connection.py
--- boto-2.49.0.20190327/tests/unit/ec2containerservice/test_connection.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/ec2containerservice/test_connection.py	2022-11-12 17:45:37.662260738 +0100
@@ -20,7 +20,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 #
-from mock import Mock
+from unittest.mock import Mock
 from tests.unit import unittest
 
 import boto.ec2containerservice
diff -aur boto-2.49.0.20190327/tests/unit/glacier/test_layer2.py boto-2.49.0.20190327.new/tests/unit/glacier/test_layer2.py
--- boto-2.49.0.20190327/tests/unit/glacier/test_layer2.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/glacier/test_layer2.py	2022-11-12 17:45:37.668927553 +0100
@@ -23,7 +23,7 @@
 
 from tests.unit import unittest
 
-from mock import call, Mock, patch, sentinel
+from unittest.mock import call, Mock, patch, sentinel
 
 import codecs
 from boto.glacier.layer1 import Layer1
diff -aur boto-2.49.0.20190327/tests/unit/glacier/test_writer.py boto-2.49.0.20190327.new/tests/unit/glacier/test_writer.py
--- boto-2.49.0.20190327/tests/unit/glacier/test_writer.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/glacier/test_writer.py	2022-11-12 17:45:37.668927553 +0100
@@ -24,7 +24,7 @@
 from boto.compat import StringIO
 
 from tests.unit import unittest
-from mock import (
+from unittest.mock import (
     call,
     Mock,
     sentinel,
diff -aur boto-2.49.0.20190327/tests/unit/mws/test_connection.py boto-2.49.0.20190327.new/tests/unit/mws/test_connection.py
--- boto-2.49.0.20190327/tests/unit/mws/test_connection.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/mws/test_connection.py	2022-11-12 17:45:37.672260962 +0100
@@ -29,7 +29,7 @@
 
 from tests.unit import AWSMockServiceTestCase
 
-from mock import MagicMock
+from unittest.mock import MagicMock
 
 
 class TestMWSConnection(AWSMockServiceTestCase):
diff -aur boto-2.49.0.20190327/tests/unit/s3/test_bucketlistresultset.py boto-2.49.0.20190327.new/tests/unit/s3/test_bucketlistresultset.py
--- boto-2.49.0.20190327/tests/unit/s3/test_bucketlistresultset.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/s3/test_bucketlistresultset.py	2022-11-12 17:45:37.675594369 +0100
@@ -22,7 +22,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
-from mock import patch, Mock
+from unittest.mock import patch, Mock
 import unittest
 
 from boto.s3.bucket import ResultSet
diff -aur boto-2.49.0.20190327/tests/unit/s3/test_bucket.py boto-2.49.0.20190327.new/tests/unit/s3/test_bucket.py
--- boto-2.49.0.20190327/tests/unit/s3/test_bucket.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/s3/test_bucket.py	2022-11-12 17:45:37.675594369 +0100
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-from mock import patch
+from unittest.mock import patch
 import xml.dom.minidom
 
 from tests.unit import unittest
diff -aur boto-2.49.0.20190327/tests/unit/sns/test_connection.py boto-2.49.0.20190327.new/tests/unit/sns/test_connection.py
--- boto-2.49.0.20190327/tests/unit/sns/test_connection.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/sns/test_connection.py	2022-11-12 17:45:37.678927776 +0100
@@ -23,7 +23,7 @@
 import json
 from tests.unit import unittest
 from tests.unit import AWSMockServiceTestCase
-from mock import Mock
+from unittest.mock import Mock
 
 from boto.sns.connection import SNSConnection
 
diff -aur boto-2.49.0.20190327/tests/unit/sqs/test_queue.py boto-2.49.0.20190327.new/tests/unit/sqs/test_queue.py
--- boto-2.49.0.20190327/tests/unit/sqs/test_queue.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/sqs/test_queue.py	2022-11-12 17:45:37.678927776 +0100
@@ -20,7 +20,7 @@
 # IN THE SOFTWARE.
 #
 from tests.unit import unittest
-from mock import Mock
+from unittest.mock import Mock
 
 from boto.sqs.queue import Queue
 
diff -aur boto-2.49.0.20190327/tests/unit/swf/test_layer2_actors.py boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_actors.py
--- boto-2.49.0.20190327/tests/unit/swf/test_layer2_actors.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_actors.py	2022-11-12 17:45:37.678927776 +0100
@@ -1,7 +1,7 @@
 import boto.swf.layer2
 from boto.swf.layer2 import Decider, ActivityWorker
 from tests.unit import unittest
-from mock import Mock
+from unittest.mock import Mock
 
 
 class TestActors(unittest.TestCase):
diff -aur boto-2.49.0.20190327/tests/unit/swf/test_layer2_base.py boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_base.py
--- boto-2.49.0.20190327/tests/unit/swf/test_layer2_base.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_base.py	2022-11-12 17:45:37.678927776 +0100
@@ -1,7 +1,7 @@
 import boto.swf.layer2
 from boto.swf.layer2 import SWFBase
 from tests.unit import unittest
-from mock import Mock
+from unittest.mock import Mock
 
 
 MOCK_DOMAIN = 'Mock'
diff -aur boto-2.49.0.20190327/tests/unit/swf/test_layer2_domain.py boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_domain.py
--- boto-2.49.0.20190327/tests/unit/swf/test_layer2_domain.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_domain.py	2022-11-12 17:45:37.678927776 +0100
@@ -1,7 +1,7 @@
 import boto.swf.layer2
 from boto.swf.layer2 import Domain, ActivityType, WorkflowType, WorkflowExecution
 from tests.unit import unittest
-from mock import Mock
+from unittest.mock import Mock
 
 
 class TestDomain(unittest.TestCase):
diff -aur boto-2.49.0.20190327/tests/unit/swf/test_layer2_types.py boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_types.py
--- boto-2.49.0.20190327/tests/unit/swf/test_layer2_types.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/swf/test_layer2_types.py	2022-11-12 17:45:37.678927776 +0100
@@ -1,7 +1,7 @@
 import boto.swf.layer2
 from boto.swf.layer2 import ActivityType, WorkflowType, WorkflowExecution
 from tests.unit import unittest
-from mock import Mock, ANY
+from unittest.mock import Mock, ANY
 
 
 class TestTypes(unittest.TestCase):
diff -aur boto-2.49.0.20190327/tests/unit/test_endpoints.py boto-2.49.0.20190327.new/tests/unit/test_endpoints.py
--- boto-2.49.0.20190327/tests/unit/test_endpoints.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/test_endpoints.py	2022-11-12 17:50:47.358958026 +0100
@@ -10,7 +10,7 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
-import mock
+import unittest.mock as mock
 import os
 import json
 
diff -aur boto-2.49.0.20190327/tests/unit/test_regioninfo.py boto-2.49.0.20190327.new/tests/unit/test_regioninfo.py
--- boto-2.49.0.20190327/tests/unit/test_regioninfo.py	2019-03-27 02:07:03.000000000 +0100
+++ boto-2.49.0.20190327.new/tests/unit/test_regioninfo.py	2022-11-12 17:50:51.855719636 +0100
@@ -20,7 +20,7 @@
 # IN THE SOFTWARE.
 #
 import os
-import mock
+import unittest.mock as mock
 
 import boto
 from boto.pyami.config import Config
