--- src/vivisect/setup.py	2021-06-21 10:05:11.683333243 +0200
+++ setup.py	2021-06-21 10:07:21.699999902 +0200
@@ -29,11 +29,11 @@
         ]
     },
     install_requires=[
-        'pyasn1>=0.4.5,<0.5.0',
-        'pyasn1-modules>=0.2.4,<0.3.0',
-        'cxxfilt>=0.2.1,<0.3.0',
-        'msgpack>=1.0.0,<1.1.0',
-        'pycparser>=2.20',
+        'pyasn1',
+        'pyasn1-modules',
+        'cxxfilt',
+        'msgpack',
+        'pycparser',
     ],
     extras_require={
         'docs': [
@@ -41,8 +41,8 @@
             'sphinx-rtd-theme>=0.4.2,<1.0.0',
         ],
         'gui': [
-            'pyqt5==5.15.1',
-            'pyqtwebengine==5.15.1',
+            'pyqt5',
+            'pyqtwebengine',
         ]
     },
     classifiers=[
