--- src/nipe/lib/Nipe/Engine/Start.pm	2020-04-08 13:14:37.833333083 +0200
+++ Start.pm	2020-04-08 13:15:38.619999747 +0200
@@ -5,8 +5,8 @@
 use Nipe::Utils::Device;
 
 sub new {
-	my $dnsPort      = "9061";
-	my $transferPort = "9051";
+	my $dnsPort      = "5353";
+	my $transferPort = "9040";
 	my @table        = ("nat", "filter");
 	my $network      = "10.66.0.0/255.255.0.0";
 	my $startTor     = "sudo systemctl start tor";
@@ -18,7 +18,7 @@
 	}
 
 	system ($startTor);
-	
+
 	foreach my $table (@table) {
 		my $target = "ACCEPT";
 
@@ -68,4 +68,4 @@
 	return 1;
 }
 
-1;
\ No newline at end of file
+1;
