--- src/whatportis/setup.py	2020-07-07 01:27:27.299998376 +0200
+++ setup.py	2020-07-07 01:27:46.369998372 +0200
@@ -18,11 +18,11 @@
     packages=['whatportis'],
     include_package_data=True,
     install_requires=[
-        "simplejson==3.17.0",
-        "tinydb==3.15.1",
-        "requests==2.23.0",
-        "prettytable==0.7.2",
-        "click==7.1.2"
+        "simplejson",
+        "tinydb",
+        "requests",
+        "prettytable",
+        "click"
     ],
     extras_require={
         "dev": [
