--- cidr2range-0.9.pl	2014-05-22 19:31:51.197795607 -0400
+++ cidr2range-0.9.pl	2014-05-22 19:32:28.307797212 -0400
@@ -83,7 +83,7 @@
     ($addr, $mask) = split(/\//, $ARGV[0]);
     ($w, $x, $y, $z) = split(/\./, $addr);
     foreach my $octet ($w, $x, $y, $z) {
-        if ((!$octet) || ($octet !~ /^\d+$/) || ($octet < 0) || ($octet > 255)) {
+        if ((!defined $octet) || ($octet !~ /^\d+$/) || ($octet < 0) || ($octet > 255)) {
             fail_validation("Invalid address or netmask. Use cidr2range --help for help.");
         }
     }
