--- tools/linguist/linguist/messagemodel.cpp    2021-06-07 12:18:43.713434493 +0200
+++ tools/linguist/linguist/messagemodel.cpp    2021-06-07 12:24:24.828016548 +0200
@@ -183,7 +183,7 @@
         if (ContextItem *c = one->findContext(oc->context())) {
             for (int j = 0; j < oc->messageCount(); ++j) {
                 MessageItem *m = oc->messageItem(j);
-                if (c->findMessage(m->text(), m->comment()) >= 0)
+                if (c->findMessage(m->text(), m->comment()))
                     ++inBoth;
             }
         }
