diff --git a/modules/systemd/src/main/java/org/elasticsearch/systemd/SystemdPlugin.java b/modules/systemd/src/main/java/org/elasticsearch/systemd/SystemdPlugin.java
index 40cc219cbb4..5b89f469da7 100644
--- a/modules/systemd/src/main/java/org/elasticsearch/systemd/SystemdPlugin.java
+++ b/modules/systemd/src/main/java/org/elasticsearch/systemd/SystemdPlugin.java
@@ -59,11 +59,7 @@ public class SystemdPlugin extends Plugin implements ClusterPlugin {
     }
 
     SystemdPlugin(final boolean assertIsPackageDistribution, final Build.Type buildType, final String esSDNotify) {
-        final boolean isPackageDistribution = buildType == Build.Type.DEB || buildType == Build.Type.RPM;
-        if (assertIsPackageDistribution) {
-            // our build is configured to only include this module in the package distributions
-            assert isPackageDistribution : buildType;
-        }
+        final boolean isPackageDistribution = true;
         if (isPackageDistribution == false) {
             logger.debug("disabling sd_notify as the build type [{}] is not a package distribution", buildType);
             enabled = false;
