diff -urN truecrypt-7.0a-source/Makefile truecrypt-7.0a-source.fixed/Makefile
--- truecrypt-7.0a-source/Makefile	2010-09-05 10:32:10.000000000 -0400
+++ truecrypt-7.0a-source.fixed/Makefile	2011-07-05 21:41:50.296437879 -0400
@@ -120,10 +120,7 @@
 
 export CPU_ARCH ?= unknown
 
-ARCH = $(shell uname -p)
-ifeq "$(ARCH)" "unknown"
-	ARCH = $(shell uname -m)
-endif
+ARCH = $(shell uname -m)
 
 ifneq (,$(filter i386 i486 i586 i686 x86,$(ARCH)))
 	CPU_ARCH = x86
