diff -Naur old/agent/protect-tool.c new/agent/protect-tool.c
--- old/agent/protect-tool.c	2012-12-08 13:53:17.067611957 +1100
+++ new/agent/protect-tool.c	2012-12-08 13:53:28.247633012 +1100
@@ -102,6 +102,7 @@
 static int opt_status_msg;
 static const char *opt_p12_charset;
 static const char *opt_agent_program; 
+static session_env_t opt_session_env;
 
 static char *get_passphrase (int promptno);
 static void release_passphrase (char *pw);
@@ -1040,6 +1041,7 @@
 
   opt_homedir = default_homedir ();
 
+  opt_session_env = session_env_new ();
 
   pargs.argc = &argc;
   pargs.argv = &argv;
@@ -1091,7 +1093,7 @@
                                 opt.verbose,
                                 opt_homedir,
                                 opt_agent_program,
-                                NULL, NULL, NULL);
+                                NULL, NULL, opt_session_env);
 
   if (opt_prompt)
     opt_prompt = percent_plus_unescape (opt_prompt, 0);
