From: hpj@urpla.net

This patch allows PyQt 3.18.1 to build with sip 4.15+. 

--- sip/qt/qlayout.sip~	2009-06-16 10:47:31.000000000 +0200
+++ sip/qt/qlayout.sip	2013-11-19 23:34:01.451326665 +0100
@@ -328,8 +328,8 @@ public:
 	QSizePolicy::ExpandData expanding() const;
 	QSize minimumSize() const;
 	QSize maximumSize() const;
-	void setGeometry(const QRect &) = 0;
-	QLayoutIterator iterator() = 0;
+	virtual void setGeometry(const QRect &) = 0;
+	virtual QLayoutIterator iterator() = 0;
 	bool isEmpty() const;
 
 	int totalHeightForWidth(int) const;

--- sip/qtnetwork/qftp.sip~	2009-06-16 10:47:31.000000000 +0200
+++ sip/qtnetwork/qftp.sip	2013-11-20 11:22:25.710338805 +0100
@@ -177,6 +177,9 @@ protected:
 	virtual void operationRename(QNetworkOperation *);
 	virtual void operationGet(QNetworkOperation *);
 	virtual void operationPut(QNetworkOperation *);
+
+private:
+	virtual bool checkConnection(QNetworkOperation *);
 };
 
 %End
