--- src/pyexfil/pyexfil/__init__.py	2021-06-23 10:28:39.363333608 +0200
+++ __init__.py	2021-06-23 10:29:38.473333609 +0200
@@ -1,7 +1,8 @@
 NAME            = "PyExfil"
 SLOGAN          = "Stress Testing Detection & Creativity"
-VERSION_CODE    = 1.3.3
+VERSION_CODE    = "1.3.3"
 VERSION_NAME    = "Harpax"
+VERSION         = "1.3.3"
 AUTHORS         = ['tisf']
 RELEASE_DATE    = '2020-12-31'
 URL             = 'https://PyExfil.MoriRT.com'
