--- params.h.orig	2006-03-17 08:23:42.000000000 -0500
+++ params.h	2006-03-17 08:50:35.000000000 -0500
@@ -101,8 +101,8 @@
 /*
  * File names.
  */
-#define CFG_FULL_NAME			"$JOHN/john.conf"
-#define CFG_ALT_NAME			"$JOHN/john.ini"
+#define CFG_FULL_NAME			"/etc/john/john.conf"
+#define CFG_ALT_NAME			"/etc/john/john.ini"
 #if JOHN_SYSTEMWIDE
 #define CFG_PRIVATE_FULL_NAME		JOHN_PRIVATE_HOME "/john.conf"
 #define CFG_PRIVATE_ALT_NAME		JOHN_PRIVATE_HOME "/john.ini"
