diff -ur xprobe2-0.3.old/libs-external/USI++/src/ip.cc xprobe2-0.3/libs-external/USI++/src/ip.cc
--- xprobe2-0.3.old/libs-external/USI++/src/ip.cc	2008-12-31 12:24:53.000000000 +0100
+++ xprobe2-0.3/libs-external/USI++/src/ip.cc	2008-12-31 12:24:34.000000000 +0100
@@ -18,6 +18,7 @@
 #include <errno.h>
 #include <new>
 #include <vector>
+#include <stdlib.h>
 
 namespace usipp {
 
Only in xprobe2-0.3/libs-external/USI++/src: libusi++.a
diff -ur xprobe2-0.3.old/libs-external/USI++/src/misc.cc xprobe2-0.3/libs-external/USI++/src/misc.cc
--- xprobe2-0.3.old/libs-external/USI++/src/misc.cc	2008-12-31 12:24:53.000000000 +0100
+++ xprobe2-0.3/libs-external/USI++/src/misc.cc	2008-12-31 12:24:34.000000000 +0100
@@ -4,7 +4,7 @@
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <errno.h>
-
+#include <stdlib.h>
 
 namespace usipp {
 
diff -ur xprobe2-0.3.old/libs-external/USI++/usi++/tcp.h xprobe2-0.3/libs-external/USI++/usi++/tcp.h
--- xprobe2-0.3.old/libs-external/USI++/usi++/tcp.h	2008-12-31 12:24:53.000000000 +0100
+++ xprobe2-0.3/libs-external/USI++/usi++/tcp.h	2008-12-31 12:24:34.000000000 +0100
@@ -15,6 +15,7 @@
 #include "usi-structs.h"
 #include "datalink.h"
 #include "ip.h"
+#include <string.h>
 
 namespace usipp {
 
