--- a/wxPython/distrib/DIRLIST
+++ b/wxPython/distrib/DIRLIST
@@ -146,104 +146,5 @@
 wx/tools/XRCed/plugins
 wx/tools/XRCed/plugins/bitmaps
 wx/tools/XRCed/plugins/src-images
-wx/tools/Editra
-wx/tools/Editra/docs
-wx/tools/Editra/include
-wx/tools/Editra/include/python2.5
-wx/tools/Editra/include/python2.5/darwin
-wx/tools/Editra/include/python2.5/win32
-wx/tools/Editra/locale
-wx/tools/Editra/locale/ca_ES@valencia
-wx/tools/Editra/locale/ca_ES@valencia/LC_MESSAGES
-wx/tools/Editra/locale/cs_CZ
-wx/tools/Editra/locale/cs_CZ/LC_MESSAGES
-wx/tools/Editra/locale/da_DK
-wx/tools/Editra/locale/da_DK/LC_MESSAGES
-wx/tools/Editra/locale/de_DE
-wx/tools/Editra/locale/de_DE/LC_MESSAGES
-wx/tools/Editra/locale/en_US
-wx/tools/Editra/locale/en_US/LC_MESSAGES
-wx/tools/Editra/locale/es_ES
-wx/tools/Editra/locale/es_ES/LC_MESSAGES
-wx/tools/Editra/locale/fr_FR
-wx/tools/Editra/locale/fr_FR/LC_MESSAGES
-wx/tools/Editra/locale/gl_ES
-wx/tools/Editra/locale/gl_ES/LC_MESSAGES
-wx/tools/Editra/locale/it_IT
-wx/tools/Editra/locale/it_IT/LC_MESSAGES
-wx/tools/Editra/locale/hr_HR
-wx/tools/Editra/locale/hr_HR/LC_MESSAGES
-wx/tools/Editra/locale/hu_HU
-wx/tools/Editra/locale/hu_HU/LC_MESSAGES
-wx/tools/Editra/locale/ja_JP
-wx/tools/Editra/locale/ja_JP/LC_MESSAGES
-wx/tools/Editra/locale/lv_LV
-wx/tools/Editra/locale/lv_LV/LC_MESSAGES
-wx/tools/Editra/locale/nl_NL
-wx/tools/Editra/locale/nl_NL/LC_MESSAGES
-wx/tools/Editra/locale/nn_NO
-wx/tools/Editra/locale/nn_NO/LC_MESSAGES
-wx/tools/Editra/locale/pl_PL
-wx/tools/Editra/locale/pl_PL/LC_MESSAGES
-wx/tools/Editra/locale/pt_BR
-wx/tools/Editra/locale/pt_BR/LC_MESSAGES
-wx/tools/Editra/locale/ro_RO
-wx/tools/Editra/locale/ro_RO/LC_MESSAGES
-wx/tools/Editra/locale/ru_RU
-wx/tools/Editra/locale/ru_RU/LC_MESSAGES
-wx/tools/Editra/locale/sk_SK
-wx/tools/Editra/locale/sk_SK/LC_MESSAGES
-wx/tools/Editra/locale/sl_SI
-wx/tools/Editra/locale/sl_SI/LC_MESSAGES
-wx/tools/Editra/locale/sr_RS
-wx/tools/Editra/locale/sr_RS/LC_MESSAGES
-wx/tools/Editra/locale/sv_SE
-wx/tools/Editra/locale/sv_SE/LC_MESSAGES
-wx/tools/Editra/locale/tr_TR
-wx/tools/Editra/locale/tr_TR/LC_MESSAGES
-wx/tools/Editra/locale/uk_UA
-wx/tools/Editra/locale/uk_UA/LC_MESSAGES
-wx/tools/Editra/locale/zh_CN
-wx/tools/Editra/locale/zh_CN/LC_MESSAGES
-wx/tools/Editra/locale/zh_TW
-wx/tools/Editra/locale/zh_TW/LC_MESSAGES
-wx/tools/Editra/pixmaps
-wx/tools/Editra/pixmaps/theme
-wx/tools/Editra/pixmaps/theme/Default
-wx/tools/Editra/pixmaps/theme/Tango
-wx/tools/Editra/pixmaps/theme/Tango/menu
-wx/tools/Editra/pixmaps/theme/Tango/mime
-wx/tools/Editra/pixmaps/theme/Tango/other
-wx/tools/Editra/pixmaps/theme/Tango/toolbar
-wx/tools/Editra/pixmaps/theme/Tango/other
-wx/tools/Editra/plugins
-wx/tools/Editra/plugins/codebrowser
-wx/tools/Editra/plugins/codebrowser/codebrowser
-wx/tools/Editra/plugins/codebrowser/codebrowser/gentag
-wx/tools/Editra/plugins/filebrowser
-wx/tools/Editra/plugins/filebrowser/filebrowser
-wx/tools/Editra/plugins/hello
-wx/tools/Editra/plugins/hello/hello
-wx/tools/Editra/plugins/Launch
-wx/tools/Editra/plugins/Launch/launch
-wx/tools/Editra/plugins/pyshell
-wx/tools/Editra/plugins/pyshell/pyshell
-wx/tools/Editra/scripts
-wx/tools/Editra/scripts/i18n
-wx/tools/Editra/src
-wx/tools/Editra/src/autocomp
-wx/tools/Editra/src/ebmlib
-wx/tools/Editra/src/eclib
-wx/tools/Editra/src/extern
-wx/tools/Editra/src/extern/aui
-wx/tools/Editra/src/extern/pygments
-wx/tools/Editra/src/extern/pygments/filters
-wx/tools/Editra/src/extern/pygments/formatters
-wx/tools/Editra/src/extern/pygments/lexers
-wx/tools/Editra/src/extern/pygments/styles
-wx/tools/Editra/src/syntax
-wx/tools/Editra/styles
-wx/tools/Editra/templates
-wx/tools/Editra/tests/syntax
 
 wxversion
--- a/wxPython/setup.py
+++ b/wxPython/setup.py
@@ -895,13 +895,6 @@
                     'wx.tools',
                     'wx.tools.XRCed',
                     'wx.tools.XRCed.plugins',
-                    'wx.tools.Editra',
-                    'wx.tools.Editra.src',
-                    'wx.tools.Editra.src.autocomp',
-                    'wx.tools.Editra.src.eclib',
-                    'wx.tools.Editra.src.ebmlib',
-                    'wx.tools.Editra.src.extern',
-                    'wx.tools.Editra.src.syntax',
                     ]
 
 
@@ -919,7 +912,6 @@
                opj('scripts/pywrap'),
                opj('scripts/pywxrc'),
                opj('scripts/xrced'),
-               opj('scripts/editra'),
                ]
     if os.name == 'nt':
         SCRIPTS.append( opj('scripts/genaxmodule') ) 
@@ -934,16 +926,6 @@
 DATA_FILES += find_data_files('wx/tools/XRCed/plugins', '*.crx')
 DATA_FILES += find_data_files('wx/tools/XRCed/plugins/bitmaps', '*.png')
 
-DATA_FILES += find_data_files('wx/tools/Editra/docs', '*.txt')
-DATA_FILES += find_data_files('wx/tools/Editra/locale', '*.mo')
-DATA_FILES += find_data_files('wx/tools/Editra/pixmaps',
-                              '*.png', '*.icns', '*.ico', 'README', 'AUTHORS', 'COPYING')
-DATA_FILES += find_data_files('wx/tools/Editra/plugins', '*.egg')
-DATA_FILES += find_data_files('wx/tools/Editra/src', 'README')
-DATA_FILES += find_data_files('wx/tools/Editra/styles', '*.ess')
-DATA_FILES += find_data_files('wx/tools/Editra/tests/syntax', '*')
-DATA_FILES += find_data_files('wx/tools/Editra', '[A-Z]*', recursive=False)
-
 
 ## import pprint
 ## pprint.pprint(DATA_FILES)
@@ -993,7 +975,6 @@
                                   'pyshell = wx.py.PyShell:main',
                                   'pywrap = wx.py.PyWrap:main',
                                   'helpviewer = wx.tools.helpviewer:main',
-                                  'editra = wx.tools.Editra.launcher:main',
                                   'xrced = wx.tools.XRCed.xrced:main',
                                   ], 
             },
