--- src/sqlninja-0.2.999-alpha1/sqlninja	2013-02-03 23:25:10.000000000 +0100
+++ sqlninja	2018-09-08 18:49:15.296666556 +0200
@@ -1,11 +1,11 @@
-#!/usr/bin/env perl 
+#!/usr/bin/env perl
 
 # sqlninja - SQL injection and takeover tool
 # Copyright (C) 2006-2013
 # http://sqlninja.sourceforge.net
 # icesurfer <r00t@northernfortress.net>
 # nico <nico@leidecker.info>
-# 
+#
 # Sqlninja is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation, either version 3 of the License, or
@@ -37,6 +37,8 @@
 # icesurfer: we should check only if $conf->session == 1, but the checks are done after the conf file is parsed
 use DBI;
 
+unshift @INC, "/usr/share/sqlninja/";
+
 require "lib/utils.pl";
 require "lib/session.pl";
 require "lib/fingerprint.pl";
@@ -69,7 +71,7 @@
 $conf->{'proxyport'} = "8080";
 $conf->{'httprequest'} = ""; # This substitutes a bunch of variables of previous versions
 $conf->{'method'}; # GET or POST
-$conf->{'vhost'} = ""; 
+$conf->{'vhost'} = "";
 $conf->{'postline'};        #######
 $conf->{'httpversion'} = 0;
 $conf->{'filterconf'} = "";
@@ -111,7 +113,7 @@
 $conf->{'language_map_adaptive'} = 1;
 $conf->{'blindtime'} = 5;
 $conf->{'dnschars'} = 25; # Chars to convert to the hostname in DNS-data-extraction
-			   # Does not include counter (currently 1 char). 
+			   # Does not include counter (currently 1 char).
 			   # NOTE: 2x$conf->{'dnschars'}+len(counter)+1 < 64 (RFC)
 $conf->{'dnscountermodulo'} = 1000000;
 $conf->{'store_session'} = 1;
@@ -303,7 +305,7 @@
 		exit(0);
 	}
 
-	# Children either signal when they are done via socket 
+	# Children either signal when they are done via socket
 	# or they are killed by the parent
 	$SIG{CHLD} = 'IGNORE';
 
@@ -320,7 +322,7 @@
 				print "    Assuming cleartext connection\n";
 				$conf->{'ssl'} = 0;
 			}
-		}	      
+		}
 	} elsif ($conf->{'ssl'} eq "yes") {
 		if ($conf->{'verbose'} == 1) {
 			print "[v] Using SSL connection\n";
@@ -397,6 +399,6 @@
 sub banner()
 {
 	print("Sqlninja rel. ".$conf->{'RELEASE'}." <http://sqlninja.sf.net>\n");
-	print("(C) 2006-2013 icesurfer & nico\n");	
+	print("(C) 2006-2013 icesurfer & nico\n");
 }
 
