diff -rupN '--exclude=.git' a/hdmi_ddc.py b/hdmi_ddc.py
--- a/hdmi_ddc.py	2014-05-09 23:09:17.081249725 +0100
+++ b/hdmi_ddc.py	2014-05-09 23:10:29.644584218 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 #  hdmi_ddc.py - HDMI DDC (I2C) definitions
 # 
@@ -8,6 +8,8 @@
 #  This code is copyright (c) Aperture Labs Ltd., 2013, All rights reserved.
 
 # import hdcp-genkey for HDCP decoding
+import os,sys
+sys.path.append('/usr/share/hdcp-genkey')
 try:
         from generate_key import read_key_file, gen_sink_key, gen_source_key, output_human_readable
 except:
@@ -20,7 +22,7 @@ except:
         exit(True)
 
 try:
-        master= open('hdcp-master.txt','r')
+        master= open('/usr/share/hdcp-genkey/master-key.txt','r')
 except:
         try:
                 master= open('master-key.txt','r')
diff -rupN '--exclude=.git' a/hdmi-sniff.py b/hdmi-sniff.py
--- a/hdmi-sniff.py	2014-05-09 23:09:17.081249725 +0100
+++ b/hdmi-sniff.py	2014-05-09 23:10:41.694584412 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2
 
 #  hdmi-sniff.py - sniff HDMI DDC (I2C) traffic with a GPHHT or Bus Pirate
 # 
