--- src/peepdf-0.4.2/setup.py	2019-12-07 13:36:14.806666476 +0100
+++ setup.py	2019-12-07 13:36:09.156666475 +0100
@@ -7,11 +7,11 @@
     license="GNU GPLv3",
     url="http://eternal-todo.com",
     install_requires=[
-        "jsbeautifier==1.6.2",
-        "colorama==0.3.7",
-        "future>=0.16.0",
-        "Pillow==3.2.0",
-        "pythonaes==1.0",
+        "jsbeautifier",
+        "colorama",
+        "future",
+        "Pillow",
+        "pythonaes",
     ],
     entry_points={
         "console_scripts": [
