--- src/requests-2.22.0/requests/certs.py	2019-05-16 16:18:16.000000000 +0200
+++ certs.py	2019-12-02 21:27:35.056666370 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
 """
@@ -12,7 +12,9 @@
 environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
-from certifi import where
+
+def where():
+  return '/etc/ssl/certs/ca-certificates.crt'
 
 if __name__ == '__main__':
     print(where())
