--- src/flare-floss/setup.py	2021-12-14 13:46:32.293329767 +0100
+++ setup.py	2021-12-14 13:47:08.246663099 +0100
@@ -7,13 +7,13 @@
 import setuptools
 
 requirements = [
-    "tabulate==0.8.9",
-    "vivisect==1.0.5",
-    "viv-utils[flirt]==0.6.9",
-    "pydantic==1.8.2",
-    "tqdm==4.62.3",
-    "networkx==2.5.1",
-    "halo==0.0.31",
+    "tabulate",
+    "vivisect",
+    "viv-utils[flirt]",
+    "pydantic",
+    "tqdm",
+    "networkx",
+    "halo",
 ]
 
 # this sets __version__
