diff --git a/autosploit.py b/autosploit.py
index 93f5130..5079960 100644
--- a/autosploit.py
+++ b/autosploit.py
@@ -362,13 +362,13 @@ if __name__=="__main__":
 	print "[" + t.green("+") + "]Initializing AutoSploit..."
 	print "[" + t.green("+") + "]One moment please while we check the Postgresql and Apache services...\n"
 
-	postgresql = cmdline("sudo service postgresql status | grep active")
+	postgresql = cmdline("sudo systemctl status postgresql | grep active")
 	if "Active: inactive" in postgresql:
 		print "\n[" + t.red("!") + "]Warning. Heuristics indicate Postgresql Service is offline"
 
 		start_pst = raw_input("\n[" + t.magenta("?") + "]Start Postgresql Service? [Y]es/[N]o: ").lower()
 		if start_pst == 'y':
-			os.system("sudo service postgresql start")
+			os.system("sudo systemctl start postgresql")
 		
 			print "[" + t.green("+") + "]Postgresql Service Started..."
 			time.sleep(1.5)
@@ -380,18 +380,18 @@ if __name__=="__main__":
 			sys.exit(0)
 		else:
 			print "\n[" + t.red("!") + "]Unhandled Option. Defaulting to starting the service."
-			os.system("sudo service postgresql start")
+			os.system("sudo systemctl start postgresql")
 		
 			print "[" + t.green("+") + "]Postgresql Service Started..."
 			time.sleep(1.5)
 
-	apache = cmdline("service apache2 status | grep active")
+	apache = cmdline("sudo systemctl status httpd | grep active")
 	if "Active: inactive" in apache:
 		print "\n[" + t.red("!") + "]Warning. Heuristics indicate Apache Service is offline"
 		
 		start_ap = raw_input("\n[" + t.magenta("?") + "]Start Apache Service? [Y]es/[N]o: ").lower()
 		if start_ap == 'y':
-			os.system("sudo service apache2 start")
+			os.system("sudo systemctl start httpd")
 		
 			print "[" + t.green("+") + "]Apache2 Service Started..."
 			time.sleep(1.5)
@@ -403,7 +403,7 @@ if __name__=="__main__":
 			sys.exit(0)
 		else:
 			print "\n[" + t.red("!") + "]Unhandled Option. Defaulting to starting the service."
-			os.system("sudo service apache2 start")
+			os.system("sudo systemctl start httpd")
 		
 			print "[" + t.green("+") + "]Apache2 Service Started..."
 			time.sleep(1.5)
