--- a/src/Makefile.am	2010-04-16 00:08:01.180977550 +0200
+++ b/src/Makefile.am	2010-04-16 00:08:48.675727493 +0200
@@ -190,7 +190,9 @@
 ucsniff_LDADD += interfaces/gtk/libec_gtk.a
 endif
 
-ucsniff_LDFLAGS = -export-dynamic @EC_LIBS@
+ucsniff_LDADD += @EC_LIBS@
+
+ucsniff_LDFLAGS = -export-dynamic
 
 CLEANFILES = *.log *.out core* missing/*.o *~

--- a/configure.in    2010-04-16 00:16:52.119980238 +0200
+++ b/configure.in    2010-04-16 00:18:51.895736536 +0200
@@ -29,6 +29,11 @@
 dnl Force automake 1.7 or better
 dnl AM_INIT_AUTOMAKE(1.7.0)

+dnl fixes
+AC_PROG_CXX
+
+AC_PROG_LIBTOOL
+
 dnl Do not recompile configure Makefile et similia
 AM_MAINTAINER_MODE

