--- Makefile.orig	2014-10-08 20:43:19.853022732 -0400
+++ Makefile	2014-10-08 21:03:26.379687926 -0400
@@ -14,11 +14,11 @@
 	install -d -m 755 $(DESTDIR)/var/lib/mana-toolkit/sslsplit
 	install -d -m 755 $(DESTDIR)/etc/mana-toolkit/
 	install -d -m 755 $(DESTDIR)/etc/stunnel/
-	install -d -m 755 $(DESTDIR)/etc/apache2/sites-available/
+	install -d -m 755 $(DESTDIR)/etc/httpd/conf/extra/
 	# Install configuration files
 	install -m 644 run-mana/conf/* $(DESTDIR)/etc/mana-toolkit/
 	install -m 644 crackapd/crackapd.conf $(DESTDIR)/etc/mana-toolkit/
-	install -m 644 apache/etc/apache2/sites-available/* $(DESTDIR)/etc/apache2/sites-available/
+	install -m 644 apache/etc/apache2/sites-available/* $(DESTDIR)/etc/httpd/conf/extra/
 	# Install the stunnel configuration we want
 	install -m 644 apache/etc/stunnel/stunnel.conf $(DESTDIR)/etc/stunnel/mana-toolkit.conf
 	# Install the hostapd binary
@@ -39,13 +39,3 @@
 	chmod 755 $(DESTDIR)/usr/share/mana-toolkit/sslstrip-hsts/sslstrip.py \
 	          $(DESTDIR)/usr/share/mana-toolkit/sslstrip-hsts/dns2proxy.py
 	install -m 755 run-mana/*.sh $(DESTDIR)/usr/share/mana-toolkit/run-mana
-	# Dynamic configuration (if not fake install)
-	if [ "$(DESTDIR)" = "" ]; then \
-	    if [ -e /etc/default/stunnel4 ]; then \
-	        sed -i -e 's/^ENABLED=.*/ENABLED=1/' /etc/default/stunnel4; \
-	    fi; \
-	    a2enmod rewrite || true; \
-	    for conf in apache/etc/apache2/sites-available/*; do \
-	        a2ensite `basename $$conf` || true; \
-	    done; \
-	fi
