diff -Naur Firewalk-bad/src/firewalk.c Firewalk/src/firewalk.c
--- Firewalk-bad/src/firewalk.c	2006-07-12 11:36:48.000000000 -0400
+++ Firewalk/src/firewalk.c	2006-07-12 11:34:39.000000000 -0400
@@ -190,6 +190,7 @@
                         return (FW_SERIOUS_ERROR);
                     default:
                         /* empty */
+		    break;
                 }
             }
         }
diff -Naur Firewalk-bad/config.sub Firewalk/config.sub
--- Firewalk-bad/config.sub	2011-03-31 14:59:23.000000000 +0200
+++ Firewalk/config.sub	2011-03-31 14:59:50.000000000 +0200
@@ -170,7 +170,7 @@
 		exit 1
 		;;
 	# Recognize the basic CPU types with company name.
-	vax-* | tahoe-* | i[3456]86-* | i860-* | m68k-* | m68000-* | m88k-* \
+	vax-* | tahoe-* | i[3456]86-* | i860-* | m68k-* | m68000-* | m88k-* | x86_64-* \
 	      | sparc-* | ns32k-* | fx80-* | arm-* | c[123]* \
 	      | mips-* | pyramid-* | tron-* | a29k-* | romp-* | rs6000-* | power-* \
 	      | none-* | 580-* | cray2-* | h8300-* | i960-* | xmp-* | ymp-* \
