--- include/ec_file.h	2010-04-16 00:35:14.093755063 +0200
+++ include/ec_file.h	2010-04-16 00:34:25.133751660 +0200
@@ -11,7 +11,7 @@
 #define MAC_FINGERPRINTS   "etter.finger.mac"
 #define TCP_FINGERPRINTS   "etter.finger.os"
 #define SERVICES_NAMES     "etter.services"
-#define ETTER_CONF         "etter.conf"
+#define ETTER_CONF         "ucsniff.conf"
 #define ETTER_DNS          "etter.dns"
 #define ETTER_FIELDS       "etter.fields"
 #define CERT_FILE          "etter.ssl.crt"
