--- a/decode.c	2025-12-13 20:09:11.722359701 +0100
+++ b/decode.c	2025-12-13 20:12:03.585628765 +0100
@@ -43,9 +43,9 @@
 	return (in_bits > 0);
 }
 
-main(
-	int			argc,
-	char			**argv)
+int main(
+	int argc,
+	char **argv)
 {
 	short			sample;
 	unsigned char		code;
@@ -53,7 +53,7 @@
 	struct g72x_state	state;
 	int			out_coding;
 	int			out_size;
-	int			(*dec_routine)();
+	int			(*dec_routine)(int, int, struct g72x_state *);
 	int			dec_bits;
 	g726_state_t *dec_state = NULL;
 	int g726_out_coding;
@@ -185,4 +185,5 @@
 		} while (len == BLOCKSIZE);
 	}
 	fclose(stdout);
+	return 0;
 }
--- a/g72x.h	2025-12-13 20:09:11.722359701 +0100
+++ b/g72x.h	2025-12-13 20:13:11.076944305 +0100
@@ -107,9 +107,9 @@
 	 int		y,	
 	short		*table,	
 	int		size);
-extern int reconstruct(int,int,int);void
+extern int reconstruct(int, int, int);
 
-extern update(
+extern void update(
 	int		code_size,	
 	int		y,		
 	int		wi,		
