--- src/stegoVeritas-1.8/setup.py	2020-07-06 17:54:55.000000000 +0200
+++ setup.py	2020-09-10 16:31:34.339999339 +0200
@@ -35,7 +35,7 @@
     ],
     keywords='steg stego steganography stegoveritas',
     packages=find_packages(exclude=['contrib', 'docs', 'tests', 'dist']),
-    install_requires=['pillow', 'numpy', 'python-magic', 'prettytable', 'exifread', 'python-xmp-toolkit', 'stegoveritas-binwalk', 'pypng', 'apng', 'stegoveritas-pfp', 'distro'],
+    install_requires=['pillow', 'numpy', 'python-magic', 'prettytable', 'exifread', 'python-xmp-toolkit', 'binwalk', 'pypng', 'apng', 'stegoveritas-pfp', 'distro'],
     extras_require={
         'dev': ['ipython','twine','pytest','python-coveralls','coverage','pytest-cov','pytest-xdist','sphinxcontrib-napoleon', 'sphinx_rtd_theme','sphinx-autodoc-typehints', 'pyOpenSSL'],
     },
