--- configure~	2009-10-07 16:09:14.426222346 +0600
+++ configure	2009-10-07 16:09:46.862888069 +0600
@@ -11089,12 +11089,7 @@
 fi
 
 
-	if test "$SYSTEM_OS" = "MACOS"
-	then
-		SSL_LIB="crypto"
-	else
-		SSL_LIB="ssl"
-	fi
+SSL_LIB="crypto"
 
 
 as_ac_Lib=`$as_echo "ac_cv_lib_${SSL_LIB}''_PEM_read_PrivateKey" | $as_tr_sh`
--- src/plugins/enc-file.c~	2009-10-07 16:13:47.192862347 +0600
+++ src/plugins/enc-file.c	2009-10-07 16:14:03.606195842 +0600
@@ -33,12 +33,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <locale.h>
-
-#ifdef LINUX
-	#include <gpgme/gpgme.h>
-#else
-	#include <gpgme.h>
-#endif
+#include <gpgme.h>
 
 #include "../plugin.h"
 #include "../utils.h"
