--- Makefile.orig	2011-02-20 22:28:43.451060686 -0500
+++ Makefile	2011-02-20 22:29:09.784344746 -0500
@@ -24,7 +24,7 @@ DISTFILES = AUTHORS ChangeLog hive.c LIC
 	samdump2.1 samdump2.h bkhive.c COPYING hive.h list.c\
 	main.c README samdump2.c
 DESTDIR   = 
-PREFIX    = /usr/local/
+PREFIX    = /usr/
 BINDIR    = $(PREFIX)/bin
 MANDIR    = $(PREFIX)/share/man/man1
 INSTALL   = $(shell which install)
@@ -49,7 +49,7 @@ clean:
 	@echo ""
 
 # Install target
-install: build infos
+install: build
 	@echo "Creating directories..."
 	$(INSTALL) -d -m 755 -o root -g root $(DESTDIR)$(BINDIR)
 	$(INSTALL) -d -m 755 -o root -g root $(DESTDIR)$(MANDIR)
