--- src/intelmq/setup.py	2021-07-28 21:01:02.420000785 +0200
+++ setup.py	2021-07-28 21:07:19.686667448 +0200
@@ -12,13 +12,13 @@
 from setuptools import find_packages, setup
 
 REQUIRES = [
-    'dnspython>=1.11.1',
-    'psutil>=1.2.1',
-    'python-dateutil>=2.5',
-    'python-termstyle>=0.1.10',
-    'pytz>=2012c',
-    'redis>=2.10',
-    'requests>=2.2.0',
+    'dnspython',
+    'psutil',
+    'python-dateutil',
+    'python-termstyle',
+    'pytz',
+    'redis',
+    'requests',
     'ruamel.yaml',
 ]
 
