--- src/pyminifier-2.1/setup.py	2014-05-24 21:11:30.000000000 +0200
+++ setup.py	2023-09-01 09:30:48.423331863 +0200
@@ -2,26 +2,13 @@
 import pyminifier
 from setuptools import setup
 from distutils.command.install import INSTALL_SCHEMES
+from distutils.command.build_py import build_py
 
 for scheme in INSTALL_SCHEMES.values():
     scheme['data'] = scheme['purelib']
 
 extra = {}
 
-if sys.version_info.major == 2:
-    from distutils.command.build_py import build_py
-elif sys.version_info.major == 3:
-    extra['use_2to3'] = True # Automatically convert to Python 3; love it!
-    try:
-        from distutils.command.build_py import build_py_2to3 as build_py
-    except ImportError:
-        print("Python 3.X support requires the 2to3 tool.")
-        print(
-            "It normally comes with Python 3.X but (apparenty) not on your "
-            "distribution.\nPlease find out what package you need to get 2to3"
-            "and install it.")
-        sys.exit(1)
-
 cmdclass = {'build_py': build_py}
 
 setup(
