--- wifite2.py	2012-05-26 05:10:34.000000000 +0100
+++ wifite.py	2014-05-29 16:32:36.847207103 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # -*- coding: utf-8 -*-
 
@@ -77,6 +77,7 @@
 
 import urllib # Check for new versions from the repo
 
+import fcntl, socket, struct
 
 ################################
 # GLOBAL VARIABLES IN ALL CAPS #
@@ -1518,15 +1519,11 @@
 	"""
 		Returns MAC address of "iface".
 	"""
-	proc = Popen(['ifconfig', iface], stdout=PIPE, stderr=DN)
-	proc.wait()
-	mac = ''
-	first_line = proc.communicate()[0].split('\n')[0]
-	for word in first_line.split(' '):
-		if word != '': mac = word
-	if mac.find('-') != -1: mac = mac.replace('-', ':')
-	if len(mac) > 17: mac = mac[0:17]
-	return mac
+        # Author: vishnubob
+        # http://code.activestate.com/recipes/439094-get-the-ip-address-associated-with-a-network-inter/
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        info = fcntl.ioctl(s.fileno(), 0x8927,  struct.pack('256s', iface[:15]))
+        return ''.join(['%02x:' % ord(char) for char in info[18:24]])[:-1]
 
 
 def generate_random_mac(old_mac):
