diff -ruB luksipc-0.01/luksipc.c src/luksipc-0.01/luksipc.c
--- luksipc-0.01/luksipc.c	2011-10-14 10:29:46.000000000 -0700
+++ src/luksipc-0.01/luksipc.c	2013-10-17 08:45:00.442435585 -0700
@@ -261,7 +261,7 @@
 		};
 		logmsg(LLVL_INFO, "Performing dm-crypt status lookup\n");
 		result = execGetReturnCode(arguments);
-		if (result != 1) {
+		if (result != 4) {
 			logmsg(LLVL_ERROR, "LUKS device 'luksipc' already open, aborting.\n");
 			exit(EXIT_FAILURE);
 		}
