diff -au0r neo4j-community-3.0.6-orig/bin/neo4j-admin neo4j-community-3.0.6-mod/bin/neo4j-admin
--- neo4j-community-3.0.6-orig/bin/neo4j-admin	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.0.6-mod/bin/neo4j-admin	2016-09-21 21:48:09.366135928 -0300
@@ -21,0 +22,3 @@
+NEO4J_CONF=/etc/neo4j
+NEO4J_HOME=/usr/share/neo4j
+
diff -au0r neo4j-community-3.0.6-orig/bin/neo4j-import neo4j-community-3.0.6-mod/bin/neo4j-import
--- neo4j-community-3.0.6-orig/bin/neo4j-import	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.0.6-mod/bin/neo4j-import	2016-09-21 21:48:09.366135928 -0300
@@ -23,0 +24,3 @@
+NEO4J_CONF=/etc/neo4j
+NEO4J_HOME=/usr/share/neo4j
+
diff -au0r neo4j-community-3.0.6-orig/bin/neo4j-shell neo4j-community-3.0.6-mod/bin/neo4j-shell
--- neo4j-community-3.0.6-orig/bin/neo4j-shell	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.0.6-mod/bin/neo4j-shell	2016-09-21 21:48:09.366135928 -0300
@@ -23,0 +24,3 @@
+NEO4J_CONF=/etc/neo4j
+NEO4J_HOME=/usr/share/neo4j
+
