diff -au0r neo4j-community-orig/bin/neo4j-admin neo4j-community-mod/bin/neo4j-admin
--- neo4j-community-orig/bin/neo4j-admin
+++ neo4j-community-mod/bin/neo4j-admin
@@ -3,0 +4,2 @@
+export NEO4J_CONF=/etc/neo4j
+
diff -au0r neo4j-community-orig/bin/cypher-shell neo4j-community-mod/bin/cypher-shell
--- neo4j-community-orig/bin/cypher-shell
+++ neo4j-community-mod/bin/cypher-shell
@@ -75,1 +75,1 @@
-  APP_HOME="$(cd "$(dirname "$0")" && pwd)"
+  APP_HOME="$(cd "$(dirname "$(readlink -f "$0")")" && pwd)"
