diff -au0r neo4j-community-3.4.10-orig/bin/neo4j-admin neo4j-community-3.4.10-mod/bin/neo4j-admin
--- neo4j-community-3.4.10-orig/bin/neo4j-admin	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.4.10-mod/bin/neo4j-admin	2016-09-21 21:48:09.366135928 -0300
@@ -3,0 +4,2 @@
+export NEO4J_CONF=/etc/neo4j
+
diff -au0r neo4j-community-3.4.10-orig/bin/neo4j-import neo4j-community-3.4.10-mod/bin/neo4j-import
--- neo4j-community-3.4.10-orig/bin/neo4j-import	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.4.10-mod/bin/neo4j-import	2016-09-21 21:48:09.366135928 -0300
@@ -3,0 +4,2 @@
+export NEO4J_CONF=/etc/neo4j
+
