diff -au0r neo4j-community-3.0.6-orig/conf/neo4j.conf neo4j-community-3.0.6-mod/conf/neo4j.conf
--- neo4j-community-3.0.6-orig/conf/neo4j.conf	2016-09-14 11:15:28.000000000 -0300
+++ neo4j-community-3.0.6-mod/conf/neo4j.conf	2016-09-21 21:49:06.659469158 -0300
@@ -9,4 +9,6 @@
-#dbms.directories.data=data
-#dbms.directories.plugins=plugins
-#dbms.directories.certificates=certificates
-#dbms.directories.logs=logs
+dbms.directories.data=/var/lib/neo4j/data
+dbms.directories.plugins=/usr/share/java/neo4j/plugins
+dbms.directories.lib=/usr/share/java/neo4j
+dbms.directories.logs=/var/log/neo4j
+dbms.directories.run=/run/neo4j
+dbms.directories.certificates=/var/lib/neo4j/certificates
