--- Makefile.orig	2021-02-07 23:06:32.422175783 +1000
+++ Makefile	2021-02-08 07:12:27.882160164 +1000
@@ -20,13 +20,7 @@
 
 all: chipsec
 
-check_kernel_dir:
-	@if [ ! -d $(KSRC) ]; then \
-		echo "Unable to find the Linux source tree."; \
-		exit 1; \
-	fi
-
-chipsec: check_kernel_dir clean 
+chipsec: clean 
 	nasm -f $(elf-size) -o $(asm-path)/cpu.o $(asm-path)/cpu.asm
 	touch $(asm-path)/.cpu.o.cmd
 	@if [ `grep -c 'efi_call' /proc/kallsyms` != "0" ]; then \
@@ -35,7 +29,6 @@
 	    make -C $(KSRC) M=$(CURDIR) modules; \
 	fi
 
-clean: check_kernel_dir 
+clean: 
 	make -C $(KSRC) M=$(CURDIR) clean
 	rm -f ${asm-path}/cpu.o
-
