--- src/twint/twint/cli.py	2022-03-27 18:05:24.619998957 +0200
+++ cli.py	2022-03-27 18:06:51.486665619 +0200
@@ -6,7 +6,7 @@
 https://github.com/twintproject/twint/wiki
 
 Licensed under MIT License
-Copyright (c) 2018 The Twint Project  
+Copyright (c) 2018 The Twint Project
 '''
 import sys
 import os
@@ -332,10 +332,6 @@
 
 def run_as_command():
     version = ".".join(str(v) for v in sys.version_info[:2])
-    if float(version) < 3.6:
-        print("[-] TWINT requires Python version 3.6+.")
-        sys.exit(0)
-
     main()
 
 
