--- src/NoSQLMap/nosqlmap.py	2017-08-26 11:29:01.386666669 +0200
+++ nosqlmap.py	2017-08-26 11:27:47.143333325 +0200
@@ -1,4 +1,5 @@
 #!/usr/bin/python
+#-*- coding: utf-8 -*-
 # NoSQLMap Copyright 2012-2017 NoSQLMap Development team
 # See the file 'doc/COPYING' for copying permission
 
@@ -55,11 +56,11 @@
     mmSelect = True
     while mmSelect:
         os.system('clear')
-        print " _  _     ___  ___  _    __  __           "
-        print "| \| |___/ __|/ _ \| |  |  \/  |__ _ _ __ "
-        print "| .` / _ \__ \ (_) | |__| |\/| / _` | '_ \"
-        print "|_|\_\___/___/\__\_\____|_|  |_\__,_| .__/"
-        print " v0.7 codingo@protonmail.com        |_|   "
+        print " _  _     ___  ___  _    __  __           "
+        print "| \| |___/ __|/ _ \| |  |  \/  |__ _ _ __ "
+        print "| .` / _ \__ \ (_) | |__| |\/| / _` | '_ \\"
+        print "|_|\_\___/___/\__\_\____|_|  |_\__,_| .__/"
+        print " v0.7 codingo@protonmail.com        |_|   "
         print "\n"
         print "1-Set options"
         print "2-NoSQL DB Access Attacks"
