--- src/scapy-http-1.8/setup.py	2016-01-24 06:10:40.000000000 +0100
+++ setup.py	2018-09-06 02:42:35.904957351 +0200
@@ -6,23 +6,6 @@
 VERSION = '1.8'
 
 
-def install_into_scapy(a):
-    print('Installing the HTTP layer extension into Scapy...')
-    import scapy
-    target_path = os.path.join(
-        os.path.dirname(scapy.__file__),
-        'layers'
-    )
-    source_path = os.path.join(
-        os.path.dirname(__file__),
-        'scapy_http/http.py'
-    )
-    shutil.copy2(source_path, target_path)
-    print('done!')
-
-
-install.sub_commands.append(('install_into_scapy', install_into_scapy))
-
 setup(
     name="scapy-http",
     packages=['scapy_http'],
