--- pyclutter-1.3.2.orig/clutter/clutter-base-types.defs	2010-10-25 05:00:14.000000000 +0600
+++ pyclutter-1.3.2/clutter/clutter-base-types.defs	2012-05-04 07:00:28.556157053 +0600
@@ -29,13 +29,6 @@
   (vtable "ClutterAnimatableIface")
 )
 
-(define-interface StageWindow
-  (in-module "Clutter")
-  (c-name "ClutterStageWindow")
-  (gtype-id "CLUTTER_TYPE_STAGE_WINDOW")
-  (vtable "ClutterStageWindowIface")
-)
-
 ; boxed types...
 
 (define-boxed Color
@@ -357,17 +350,6 @@
   )
 )
 
-(define-enum ShaderError
-  (in-module "Clutter")
-  (c-name "ClutterShaderError")
-  (gtype-id "CLUTTER_TYPE_SHADER_ERROR")
-  (values
-    '("no-asm" "CLUTTER_SHADER_ERROR_NO_ASM")
-    '("no-glsl" "CLUTTER_SHADER_ERROR_NO_GLSL")
-    '("compile" "CLUTTER_SHADER_ERROR_COMPILE")
-  )
-)
-
 (define-enum ScriptError
   (in-module "Clutter")
   (c-name "ClutterScriptError")
