diff -ru socketpool-0.2.3.old/setup.py socketpool-0.2.3/setup.py
--- socketpool-0.2.3.old/setup.py	2012-01-26 21:57:45.947850603 +0100
+++ socketpool-0.2.3/setup.py	2012-01-26 21:58:04.771081964 +0100
@@ -23,8 +23,8 @@
     long_description = f.read()
 
 DATA_FILES = [
-        ('socketpool', ["LICENSE", "MANIFEST.in", "NOTICE", "README.rst",
-                        "THANKS", "UNLICENSE"])
+        ('share/doc/socketpool', ["LICENSE", "MANIFEST.in", "NOTICE",
+                                  "README.rst", "THANKS", "UNLICENSE"])
         ]
 
 
