--- src/quicksand-lite/libqs.h	2019-12-24 12:17:02.496666649 +0100
+++ libqs.h	2019-12-24 12:17:32.803333315 +0100
@@ -4,7 +4,7 @@
  *
  * File libqs.h   Sep 12 2017
  * Original source code available from https://github.com/tylabs/quicksand_lite
- * 
+ *
  * Decode and look in streams of Office Documents, RTF, MIME MSO.
  * XOR Database attack up to 256 byte keys to find embedded exe's.
  * Lite version - doesn't include cryptanalysis module and latest Office CVEs
@@ -12,7 +12,7 @@
  *
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
  * Commercial licensing is available for the full version.
  */
@@ -91,7 +91,7 @@
     const char *identifier;
     struct qs_file *parent;
     struct qs_file *qs_root;
-    
+
     int rel;
 };
 
@@ -109,24 +109,24 @@
     struct qs_file *parent;
     struct qs_file *next; //linked list
     struct qs_file *child; //linked list
-    
+
     struct qs_hit *hits; //linked list
     int count; //number of hits
-    
-    
+
+
     const char *md5;
     const char *sha1;
     const char *sha256;
     const char *sha512;
     const char *head;
-    
+
     long unsigned data_len;
-    
+
     const unsigned char *data;
-    
+
     int malware_score;
-    
-    
+
+
 };
 
 
@@ -167,5 +167,3 @@
 
 void quicksandDropObjects(struct qs_file *, struct qs_file **);
 
-#endif /* defined(____quicksand__) */
-
