--- src/Google-Search-API-1.1.14/setup.py	2019-04-02 14:44:44.000000000 +0200
+++ setup.py	2019-04-20 01:26:15.559999757 +0200
@@ -9,12 +9,6 @@
 
 # from requirements import r
 
-with open("requirements.txt") as f:
-    requirements = [req.strip() for req in f.readlines()]
-
-with open("test_requirements.txt") as f:
-    test_requirements = [req.strip() for req in f.readlines()]
-
 setup(name='Google-Search-API',
       version='1.1.14',
       url='https://github.com/abenassi/Google-Search-API',
@@ -31,7 +25,6 @@
       ],
       package_dir={'google': 'google'},
       include_package_data=True,
-      install_requires=requirements,
       keywords="google search images api",
       classifiers=[
           'Development Status :: 3 - Alpha',
@@ -42,8 +35,6 @@
           'Programming Language :: Python :: 3.6',
           'Programming Language :: Python :: 3.7',
       ],
-      setup_requires=['nose>=1.0'],
-      test_suite='nose.collector',
-      tests_require=test_requirements
+      setup_requires=['nose'],
       # **r.requirements
       )
