--- src/usbrip/setup.py	2020-01-25 15:07:37.089999696 +0100
+++ setup.py	2020-01-25 15:07:53.346666364 +0100
@@ -52,10 +52,10 @@
 			else:
 				deps.append(dep)
 
-		resolve(wheel, tools_dir)
+		#resolve(wheel, tools_dir)
 
-		for dep in deps:
-			resolve(dep, tools_dir)
+		#for dep in deps:
+			#resolve(dep, tools_dir)
 
 
 class CleanCommand(Command):
@@ -115,7 +115,7 @@
 
 keywords = 'forensics cybersecurity infosec usb-history usb-devices'
 
-resolve('wheel')
+#resolve('wheel')
 
 setup(
 	name='usbrip',
