--- src/pathlib-1.0.1/pathlib.py	2014-09-03 17:25:37.000000000 +0200
+++ pathlib.py	2023-09-01 22:42:09.349998358 +0200
@@ -7,7 +7,6 @@
 import re
 import sys
 import time
-from collections import Sequence
 from contextlib import contextmanager
 from errno import EINVAL, ENOENT
 from operator import attrgetter
@@ -534,7 +533,7 @@
 # Public API
 #
 
-class _PathParents(Sequence):
+class _PathParents():
     """This object provides sequence-like access to the logical ancestors
     of a path.  Don't try to construct it yourself."""
     __slots__ = ('_pathcls', '_drv', '_root', '_parts')
