--- src/trevorproxy-1.0.6/trevorproxy/cli.py	2023-06-20 16:58:58.912897800 +0200
+++ cli.py	2024-11-28 01:17:15.363337185 +0100
@@ -12,10 +12,10 @@
 package_path = Path(__file__).resolve().parent
 sys.path.append(str(package_path))
 
-import lib.logger
-from lib import util
-from lib import logger
-from lib.errors import *
+from .lib import logger
+from .lib import util
+from .lib import logger
+from .lib.errors import *
 
 log = logging.getLogger("trevorproxy.cli")
 
