--- a/src/galleta.c	2025-12-13 20:34:52.290481814 +0100
+++ b/src/galleta.c	2025-12-13 20:36:30.042547028 +0100
@@ -27,7 +27,8 @@
  * SUCH DAMAGE.
  */
 
-
+#include <stdlib.h>
+#include <string.h>
 #include <sys/types.h>
 #include <sys/uio.h>
 #include <unistd.h>
@@ -130,7 +131,8 @@
   while ((opt = getopt( argc, argv, "d:")) != -1) {
     switch(opt) {
       case 'd':
-        strncpy( delim, optarg, 10 );
+        strncpy(delim, optarg, sizeof(delim) - 1);
+        delim[sizeof(delim) - 1] = '\0';
         break;
 
       default:
@@ -260,4 +262,5 @@
   }
 
   close (cookie_file);
+  return 0;
 }
