diff --git a/Makefile-Linux b/Makefile-Linux
index 0e4d012..a67adbd 100644
--- a/Makefile-Linux
+++ b/Makefile-Linux
@@ -53,7 +53,7 @@ install-themes:
 
 post-install:
 	@echo ""
-	@if [ "$(aircrack_prefix)" != "/usr/local " ]; then sed -i 's/AIRPREFIX=\"\/usr\/local\"/AIRPREFIX=\"$(aircrack_prefix)\"/g' $(etcdir)/airoscript-ng.conf &>/dev/null;fi
+	@if [ "$(aircrack_prefix)" != "/usr/local " ]; then sed -i 's:AIRPREFIX=\"\/usr\/local\":AIRPREFIX=\"$(aircrack_prefix)\":g' $(etcdir)/airoscript-ng.conf;fi
 
 install-docs: install-main-docs install-manpages
 
@@ -71,12 +71,6 @@ install-manpages:
 #		                    locale                                   #
 ###############################################################################
 generate-locale:
-	@for i in $(srcdir)/locale/* ; do \
-		if [ -d $$i ] && [ $$i != "." ] && [ $$i != ".." ]; then \
-			make -s -C $$i &> /dev/null ; \
-		fi ; \
-	done
-
 
 install-locale: generate-locale
 	@echo "Installing locales"
