diff --git a/hack_library/hack_library.c b/hack_library/hack_library.c
index 419a27e..9dd7d52 100644
--- a/hack_library/hack_library.c
+++ b/hack_library/hack_library.c
@@ -26,6 +26,11 @@
 //
 //-------------------------------------------------------------------------------
 
+#include <sys/time.h>
+#include <ctype.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 #include "hack_library.h"
 
 //-------------------------------------------------------------------------------
