--- honggfuzz-0.5/Makefile	2015-02-25 20:59:27.000000000 +0100
+++ honggfuzz-0.5/Makefile	2015-03-01 20:26:05.892041306 +0100
@@ -20,8 +20,8 @@

 CC ?= gcc
 CFLAGS += -c -std=c11 -I. -I/usr/local/include -I/usr/include \
-	-D_GNU_SOURCE \
-	-Wall -Wextra -Wno-initializer-overrides -Wno-override-init -Wno-unknown-warning-option -Werror
+	-D_GNU_SOURCE -DPACKAGE -DPACKAGE_VERSION \
+	-Wall -Wextra  -Wno-override-init  -Werror

 LD = $(CC)
 LDFLAGS += -lm -lpthread -L/usr/local/include -L/usr/include
@@ -43,7 +43,8 @@
 	ifeq ("$(wildcard /usr/include/libunwind-ptrace.h)","")
 		WARN_LIBRARY += "libunwind-dev/libunwind8-dev "
 	endif
-	LDFLAGS += -lunwind-ptrace -lunwind-generic -lbfd -lopcodes
+	LDFLAGS += -lunwind-ptrace -lunwind-generic -lbfd -lopcodes -lm -liberty \
+						 -ldl -lz
 	ARCH_SRCS := $(wildcard linux/*.c)
 	ARCH = LINUX
 endif
