diff -ur gsd-1.2.1.orig/src/omp_utilities.h gsd-1.2.1/src/omp_utilities.h
--- gsd-1.2.1.orig/src/omp_utilities.h	2011-12-09 15:31:24.000000000 +0100
+++ gsd-1.2.1/src/omp_utilities.h	2012-04-12 22:53:59.386444437 +0200
@@ -76,7 +76,7 @@
       };
     typedef int omp_type;
 
-    enum omp_entity_command
+    enum omp_entity_command_t
       {
         OMP_GET = 30,
         OMP_CREATE = 31,
@@ -89,7 +89,7 @@
       };
     typedef int omp_entity_command;
 
-    enum omp_task_command
+    enum omp_task_command_t
       {
         START = 40,
         STOP = 41,
