--- src/yawast-0.11.0/setup.py	2020-01-02 04:01:28.000000000 +0100
+++ setup.py	2023-09-01 11:08:27.323331781 +0200
@@ -144,7 +144,6 @@
     executables=[Executable("bin/yawast", base=None)],
     packages=find_packages(exclude=["tests"]),
     entry_points={"console_scripts": ["yawast = yawast.__main__:main"]},
-    install_requires=get_install_reqs(),
     include_package_data=True,
     package_data={"yawast": ["resources/*"]},
     zip_safe=False,
