--- a/bootloader/wscript
+++ b/bootloader/wscript
@@ -518,7 +518,7 @@
             # manipulation functions.
             if ctx.options.debug:
                 ctx.env.append_value('CFLAGS', '-U_FORTIFY_SOURCE')
-            elif ctx.env.CC_NAME == 'gcc':
+            elif ctx.env.CC_NAME == 'gcc' and '-Wp,-D_FORTIFY_SOURCE=3' not in ctx.env['CFLAGS']:
                 # Undefine this macro if already defined by default to avoid "macro redefinition" errors.
                 ctx.env.append_value('CFLAGS', '-U_FORTIFY_SOURCE')
