--- src/honggfuzz-0.6/Makefile	2015-07-31 06:23:54.000000000 +0200
+++ Makefile	2015-11-08 14:20:54.877820066 +0100
@@ -20,12 +20,12 @@
 
 CC ?= gcc
 CFLAGS += -std=c11 -I. -I/usr/local/include -I/usr/include \
-	-D_GNU_SOURCE \
-	-Wall -Wextra -Wno-initializer-overrides -Wno-override-init -Wno-unknown-warning-option -Werror \
-	-funroll-loops -O2
+	-D_GNU_SOURCE -DPACKAGE -DPACKAGE_VERSION \
+	-Wall -Wextra  -Wno-override-init  -Werror -funroll-loops -O2
 
 LD = $(CC)
-LDFLAGS += -lm -lpthread -L/usr/local/include -L/usr/include
+LDFLAGS += -lunwind-ptrace -lunwind-generic -lbfd -lopcodes -lm -lpthread \
+					 -liberty -ldl -lz
 
 SRCS = honggfuzz.c log.c files.c fuzz.c report.c mangle.c util.c
 
@@ -138,7 +138,7 @@
 
 depend:
 	makedepend -Y. -Y* -- $(SRCS)
-	
+
 .PHONY:android
 android:
 	ndk-build NDK_PROJECT_PATH=. APP_BUILD_SCRIPT=./android/Android.mk APP_PLATFORM=android-21
