--- core_file.py.orig	2013-11-22 07:59:07.794840410 -0500
+++ core_file.py	2013-11-22 08:01:12.961509489 -0500
@@ -25,7 +25,7 @@
 ## ################################################################# ##
 
 # This variable is overwritten by configuration script
-CFG_FILE = 'P:/webshag/dev/1.1/config/webshag.conf'
+CFG_FILE = '/usr/share/webshag/config/webshag.conf'
 
 FILE_ENCODING = u'utf-8'
 CFG_SECTION_FILE = u'core_file'
