--- src/intrace-intrace-1.6/Makefile	2017-08-15 22:29:53.000000000 +0200
+++ Makefile	2025-11-30 22:10:31.215498473 +0100
@@ -18,7 +18,7 @@
 CC ?= gcc
 COMMON_CFLAGS = -fPIC -O3 -g -c -std=gnu11  -I. \
                  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE \
-                 -Wall -Werror -Wextra
+                 -Wall -Werror -Wextra -Wno-address-of-packed-member
 
 COMMON_LDFLAGS = -fPIE -lpthread
 
