From 3bbc46967d35cd7e05677018bdc9d552efaeea3e Mon Sep 17 00:00:00 2001
From: Ben Finney <ben+python@benfinney.id.au>
Date: Mar 08 2023 07:02:47 +0000
Subject: Use explicitly-maintained ‘packaging.version’ for version string parsing.


---

diff --git a/test_version.py b/test_version.py
index c43cd36..d6095e4 100644
--- a/test_version.py
+++ b/test_version.py
@@ -20,10 +20,10 @@ import unittest.mock
 import docutils
 import docutils.nodes
 import docutils.writers
+from packaging.version import InvalidVersion
 import setuptools
 import setuptools.command
 import setuptools.dist
-from setuptools.extern.packaging.version import InvalidVersion
 import testscenarios
 import testtools
 
diff --git a/version.py b/version.py
index 20d3ef3..2d7d048 100644
--- a/version.py
+++ b/version.py
@@ -31,12 +31,12 @@ import re
 import sys
 import textwrap
 
+import packaging.version
 import setuptools
 import setuptools.command.build
 import setuptools.command.build_py
 import setuptools.command.egg_info
 import setuptools.dist
-import setuptools.extern.packaging.version
 
 
 def ensure_class_bases_begin_with(namespace, class_name, base_class):
@@ -186,7 +186,7 @@ class ChangeLogEntry:
             # A valid non-version value.
             return None
 
-        valid_version = setuptools.extern.packaging.version.Version(value)
+        valid_version = packaging.version.Version(value)
 
     @classmethod
     def validate_maintainer(cls, value):

