--- src/Net-Pcap-0.18/stubs.inc	2006-10-16 00:00:58.000000000 +0200
+++ stubs.inc	2019-06-13 01:20:07.259999867 +0200
@@ -2,10 +2,10 @@
  * stubs.inc - compatibility definitions to make this module compile in every cases
  *
  * Copyright (C) 2005, 2006 Sebastien Aperghis-Tramoni. All rights reserved.
- * 
- * This program is free software; you can redistribute it and/or modify 
+ *
+ * This program is free software; you can redistribute it and/or modify
  * it under the same terms as Perl itself.
- * 
+ *
  */
 
 /* Macros and constants */
@@ -354,11 +354,11 @@
 #ifdef _MSC_VER
 #pragma message( "Warning: the function pcap_open() is not available" )
 #endif
-struct pcap_rmtauth {
+/*struct pcap_rmtauth {
     int type;
     char *username;
     char *password;
-};
+};*/
 
 pcap_t * pcap_open(const char *source, int snaplen, int flags, int read_timeout, struct pcap_rmtauth *auth, char *err);
 pcap_t * pcap_open(const char *source, int snaplen, int flags, int read_timeout, struct pcap_rmtauth *auth, char *err) {
@@ -516,10 +516,10 @@
 #ifdef _MSC_VER
 #pragma message( "Warning: the function pcap_setsampling() is not available" )
 #endif
-struct pcap_samp {
+/*struct pcap_samp {
     int method;
     int value;
-};
+};*/
 
 struct pcap_samp *pcap_setsampling(pcap_t *p);
 struct pcap_samp *pcap_setsampling(pcap_t *p) {
