diff --git a/common/usb_cdc.c b/common/usb_cdc.c
index c6dec376..425ed148 100644
--- a/common/usb_cdc.c
+++ b/common/usb_cdc.c
@@ -157,28 +157,28 @@ static const char cfgDescriptor[] = {
 };
 
 // Microsoft OS Extended Configuration Compatible ID Descriptor
-static const char OSFeatureDescriptor[] = {
-		// Descriptor Length 40bytes (0x28)
-		0x28, 0x00, 0x00, 0x00,
-		// Version ('1.0')
-		0x00, 0x01,
-		// Compatibility ID Descriptor Index  0x0004
-		0x04, 0x00,
-		// Number of sections. 0x1
-		0x01,
-		// Reserved (7bytes)
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-		// Interface Number #0
-		0x00,
-		//reserved 
-		0x01,
-		// Compatible ID  ('WINUSB\0\0')  (8bytes)
-		0x57, 0x49, 0x4E, 0x55, 0x53, 0x42, 0x00, 0x00,
-		// Sub-Compatible ID (8byte)
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-		// Reserved (6bytes)
-		0x00, 0x00, 0x00, 0x00, 0x00, 0x00
-};
+/** static const char OSFeatureDescriptor[] = { */
+		/** // Descriptor Length 40bytes (0x28) */
+		/** 0x28, 0x00, 0x00, 0x00, */
+		/** // Version ('1.0') */
+		/** 0x00, 0x01, */
+		/** // Compatibility ID Descriptor Index  0x0004 */
+		/** 0x04, 0x00, */
+		/** // Number of sections. 0x1 */
+		/** 0x01, */
+		/** // Reserved (7bytes) */
+		/** 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, */
+		/** // Interface Number #0 */
+		/** 0x00, */
+		/** //reserved  */
+		/** 0x01, */
+		/** // Compatible ID  ('WINUSB\0\0')  (8bytes) */
+		/** 0x57, 0x49, 0x4E, 0x55, 0x53, 0x42, 0x00, 0x00, */
+		/** // Sub-Compatible ID (8byte) */
+		/** 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, */
+		/** // Reserved (6bytes) */
+		/** 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 */
+/** }; */
 
 // Microsoft Extended Properties Feature Descriptor
 const g_sOSProperties OSPropertyDescriptor = {
@@ -261,19 +261,19 @@ static const char StrSerialNumber[] = {
  	'8', 0x00
 };
 
-static const char StrOSDescriptor[] = {
-    18,			// length
-	0x03,		// Type is string
-    'M', 0x00,
-	'S', 0x00,
-	'F', 0x00,
-	'T', 0x00,
-	'1', 0x00,
-	'0', 0x00,
-	'0', 0x00,
-    MS_VENDOR_CODE, 		// Vendor code
-    0x00
-};
+/** static const char StrOSDescriptor[] = { */
+    /** 18,			// length */
+	/** 0x03,		// Type is string */
+    /** 'M', 0x00, */
+	/** 'S', 0x00, */
+	/** 'F', 0x00, */
+	/** 'T', 0x00, */
+	/** '1', 0x00, */
+	/** '0', 0x00, */
+	/** '0', 0x00, */
+    /** MS_VENDOR_CODE, 		// Vendor code */
+    /** 0x00 */
+/** }; */
 
 static const char* const pStrings[] = {
     StrDescLanguageCodes,
