diff -Naur oat.orig/opwg.sh oat/opwg.sh
--- oat.orig/opwg.sh	2002-06-11 09:32:00.000000000 +0000
+++ oat/opwg.sh	2008-02-01 01:40:56.000000000 +0000
@@ -2,7 +2,7 @@
 #
 # Wrapper for OraclePWGuess by patrik.karlsson@ixsecurity.com
 #
-
+cd /usr/share/oat/
 JAVA=java
 JDBC=classes111.zip
-$JAVA -cp .:$JDBC:ork.jar ork.OraclePwGuess $*
\ No newline at end of file
+$JAVA -cp .:$JDBC:ork.jar ork.OraclePwGuess $*
diff -Naur oat.orig/oquery.sh oat/oquery.sh
--- oat.orig/oquery.sh	2002-06-11 09:32:00.000000000 +0000
+++ oat/oquery.sh	2008-02-01 01:41:01.000000000 +0000
@@ -2,7 +2,7 @@
 #
 # Wrapper for OracleQuery by patrik@cqure.net
 #
-
+cd /usr/lib/oat/
 JAVA=java
 JDBC=classes111.zip
 $JAVA -cp .:$JDBC:ork.jar ork.OracleQuery $*
diff -Naur oat.orig/osd.sh oat/osd.sh
--- oat.orig/osd.sh	2002-06-11 09:32:00.000000000 +0000
+++ oat/osd.sh	2008-02-01 01:41:07.000000000 +0000
@@ -2,7 +2,7 @@
 #
 # Wrapper for OracleSamDump by patrik.karlsson@ixsecurity.com
 #
-
+cd /usr/share/oat/
 JAVA=java
 JDBC=classes111.zip
-$JAVA -cp .:$JDBC:ork.jar ork.OracleSamDump $*
\ No newline at end of file
+$JAVA -cp .:$JDBC:ork.jar ork.OracleSamDump $*
diff -Naur oat.orig/ose.sh oat/ose.sh
--- oat.orig/ose.sh	2002-06-11 09:32:00.000000000 +0000
+++ oat/ose.sh	2008-02-01 01:41:12.000000000 +0000
@@ -2,7 +2,7 @@
 #
 # Wrapper for OracleSysExec by patrik.karlsson@ixsecurity.com
 #
-
+cd /usr/share/oat/
 JAVA=java
 JDBC=classes111.zip
-$JAVA -cp .:$JDBC:ork.jar ork.OracleSysExec $*
\ No newline at end of file
+$JAVA -cp .:$JDBC:ork.jar ork.OracleSysExec $*
diff -Naur oat.orig/otnsctl.sh oat/otnsctl.sh
--- oat.orig/otnsctl.sh	2002-06-11 09:32:00.000000000 +0000
+++ oat/otnsctl.sh	2008-02-01 01:41:17.000000000 +0000
@@ -2,7 +2,7 @@
 #
 # Wrapper for OracleTNSCtrl by patrik.karlsson@ixsecurity.com
 #
-
+cd /usr/share/oat/
 JAVA=java
 JDBC=classes111.zip
-$JAVA -cp .:$JDBC:ork.jar ork.OracleTNSCtrl $*
\ No newline at end of file
+$JAVA -cp .:$JDBC:ork.jar ork.OracleTNSCtrl $*
