diff --git a/traxss/core/scanner.py b/traxss/core/scanner.py
index 4487087..80f3548 100644
--- a/traxss/core/scanner.py
+++ b/traxss/core/scanner.py
@@ -125,7 +125,7 @@ class Scanner:
         print(blue('[*] Running XSS Scan [*]'))
         options = webdriver.ChromeOptions()
         options.add_argument('--headless')
-        self.driver = webdriver.Chrome(chrome_options=options)
+        self.driver = webdriver.Chrome(options=options)
         query_window, html_window = self.setup_windows()
         webelement_list = WebDriverWait(self.driver, 10).until(expected_conditions.presence_of_all_elements_located((By.XPATH, "//input | //textarea | //button")))
         for payload in self.payloads:
