diff -upr recommonmark-0.7.1.orig/recommonmark/transform.py recommonmark-0.7.1/recommonmark/transform.py
--- recommonmark-0.7.1.orig/recommonmark/transform.py	2019-08-09 21:38:35.000000000 +0300
+++ recommonmark-0.7.1/recommonmark/transform.py	2022-05-31 13:35:09.247049278 +0300
@@ -28,6 +28,8 @@ class AutoStructify(transforms.Transform
             self.config.update(new_cfg)
         except AttributeError:
             pass
+        # provide fallback like https://repo.or.cz/docutils.git/commit/e3bd0958bac8
+        self.document.settings.setdefault('tab_width', 8)

         # Deprecation notices
         # TODO move this check to an extension pattern, and only call once

