--- a/gdk/glue/windowmanager.c	2025-10-26 03:15:22.499055212 +0100
+++ b/gdk/glue/windowmanager.c	2025-10-26 03:16:13.842846273 +0100
@@ -110,8 +110,8 @@
 	list = g_malloc (*count * sizeof (gpointer));
 	/* Put all of the windows into a GList to return */
 	for (i = 0; i < *count; i ++) {
-		list [i] = data [i];
-		g_message ("WinID: %d", list [i]);
+		list[i] = GSIZE_TO_POINTER ((gsize) data[i]);
+		g_message ("WinID: %p", list[i]);
 	}
 
 	g_free (data);
