diff -ruN a/cewl.rb b/cewl.rb
--- a/cewl.rb	2013-03-10 18:37:24.327471418 +0100
+++ b/cewl.rb	2013-03-10 18:42:32.460806353 +0100
@@ -67,7 +67,7 @@
 require 'spider'
 require 'nokogiri'
 require 'http_configuration'
-require './cewl_lib'
+require 'cewl/cewl_lib'
 
 # Doing this so I can override the allowed? fuction which normally checks
 # the robots.txt file
diff -ruN a/fab.rb b/fab.rb
--- a/fab.rb	2013-03-10 18:48:59.897475034 +0100
+++ b/fab.rb	2013-03-10 18:51:36.244142511 +0100
@@ -29,7 +29,7 @@
 
 require "rubygems"
 require 'getoptlong'
-require "./cewl_lib.rb"
+require "cewl/cewl_lib"
 
 opts = GetoptLong.new(
 	[ '--help', '-h', GetoptLong::NO_ARGUMENT ],
