--- setup.py	2019-03-07 01:58:02.660442056 +0000
+++ setup2.py	2019-03-07 02:14:39.997808979 +0000
@@ -1,4 +1,4 @@
-# !/usr/bin/python
+# !/usr/bin/python2
 # -*- coding: utf-8 -*-
 #
 ##################################################################################
@@ -116,12 +116,12 @@
         "BeautifulSoup",
         "requests",
         "python-emailahoy",
-        "pyexcel==0.2.1",
-        "pyexcel_ods==0.1.1",
-        "pyexcel_xls==0.1.0",
-        "pyexcel_xlsx==0.1.0",
-        "pyexcel_io==0.1.0",
-        "pyexcel_text==0.2.0",
+        "pyexcel",
+        "pyexcel_ods",
+        "pyexcel_xls",
+        "pyexcel_xlsx",
+        "pyexcel_io",
+        "pyexcel_text",
         "tweepy",
         "networkx",
         "decorator",
@@ -131,8 +131,7 @@
         #"py3dns",
         #"robobrowser",
         "tabulate",
-        "oauthlib>=1.0.0",
-        "importlib",
+        "oauthlib",
         #"inspect",
         #"pkgutil",
         # Adding dependencies to avoid the InsecurePlatformWarning when calling Twitter API dealing with SSL: <http://stackoverflow.com/a/29202163>. Other options would require the user to upgrade to Python 2.7.9.
