--- src/osrframework/setup.py	2020-05-02 21:14:01.919998197 +0200
+++ setup.py	2020-05-02 21:14:34.619998197 +0200
@@ -46,8 +46,6 @@
 NEW_VERSION = osrframework.__version__
 
 print("[*] OSRFramework > Reading requirements...")
-with open("requirements.txt") as file:
-    requirements = file.read().splitlines()
 
 # Depending on the place in which the project is going to be upgraded
 def read(fname):
@@ -107,7 +105,6 @@
     packages=find_packages(),
     long_description=read('README.md'),
     long_description_content_type="text/markdown",
-    install_requires=requirements,
 )
 
 ############################
