--- src/pyjfuzz/setup.py	2025-12-04 21:56:17.625000000 +0100
+++ setup.py	2025-12-04 21:56:36.772869650 +0100
@@ -44,9 +44,11 @@
     return data
 
 def install_gramfuzz():
-    cur_dir = os.getcwd()
-    os.chdir("gramfuzz")
-    subprocess.Popen(["python" ,os.getcwd()+os.sep+"setup.py", "install"]).wait()
+    cur_dir = os.path.abspath(os.getcwd())
+    gramfuzz_dir = os.path.join(cur_dir, 'gramfuzz')
+    os.chdir(gramfuzz_dir)
+    subprocess.Popen([ "python2", os.path.join(gramfuzz_dir, "setup.py"),
+                      "install"]).wait()
     os.chdir(cur_dir)
 
 install_gramfuzz()
