diff -urB ruby-1.8.7-p352/ext/dl/depend ruby-1.8.7-p352.patched//ext/dl/depend
--- ruby-1.8.7-p352/ext/dl/depend	2007-02-12 23:01:19.000000000 +0000
+++ ruby-1.8.7-p352.patched//ext/dl/depend	2011-07-26 18:25:12.512525291 +0100
@@ -32,15 +32,15 @@
 
 call.func: $(srcdir)/mkcall.rb ./dlconfig.rb
 	@echo "Generating call.func"
-	@$(RUBY) $(srcdir)/mkcall.rb > $@
+	@$(RUBY) $(srcdir)/mkcall.rb | cat > $@
 
 callback.func: $(srcdir)/mkcallback.rb ./dlconfig.rb
 	@echo "Generating callback.func"
-	@$(RUBY) $(srcdir)/mkcallback.rb > $@
+	@$(RUBY) $(srcdir)/mkcallback.rb | cat > $@
 
 cbtable.func: $(srcdir)/mkcbtable.rb ./dlconfig.rb
 	@echo "Generating cbtable.func"
-	@$(RUBY) $(srcdir)/mkcbtable.rb > $@
+	@$(RUBY) $(srcdir)/mkcbtable.rb | cat > $@
 
 debug:
 	$(MAKE) CPPFLAGS="$(CPPFLAGS) -DDEBUG"
