diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e8b854..c202920 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -350,7 +350,8 @@ add_executable(main_exe
     ${VISUALIZATION_SHADERS}
 )
 
-qt5_use_modules(veles_base Core Gui Widgets Network)
+find_package( Qt5 REQUIRED COMPONENTS Core Gui Widgets Network )
+target_link_libraries(veles_base Qt5::Core Qt5::Gui Qt5::Widgets Qt5::Network)
 
 target_link_libraries(main_exe veles_base Qt5::Widgets ${ZLIB_LIBRARIES} ${ADDITIONAL_LINK_LIBRARIES})
 set_target_properties(main_exe PROPERTIES OUTPUT_NAME "veles")
