--- qt.old/qscintilla.pro	2013-12-18 14:36:25.662223433 +0100
+++ qt/qscintilla.pro	2013-12-18 15:16:18.375254045 +0100
@@ -25,9 +25,8 @@
 
 TEMPLATE = lib
 TARGET = qscintilla
-DESTDIR = $(QTDIR)/lib
 CONFIG += qt warn_off release dll thread
-INCLUDEPATH = . ../include ../src
+INCLUDEPATH = . ../include ../src /usr/include/qt3
 DEFINES = QEXTSCINTILLA_MAKE_DLL QT SCI_LEXER
 
 HEADERS = \
@@ -211,10 +210,10 @@
 	qscintilla_ru.ts
 
 
-header.path = $(QTDIR)/include
+header.path = /usr/include/qt3
 header.files = qextscintilla*.h
 
-trans.path = $(QTDIR)/translations
+trans.path = /usr/share/qt3/translations
 trans.files = qscintilla_*.qm
 
 INSTALLS += header trans
