diff -rupN moloch-0.9.2.orig/capture/Makefile moloch-0.9.2/capture/Makefile
--- moloch-0.9.2.orig/capture/Makefile	2013-12-23 09:27:11.739458108 -0500
+++ moloch-0.9.2/capture/Makefile	2013-12-23 09:27:36.796123983 -0500
@@ -21,7 +21,7 @@ LIB_OTHER     = -LNONE/lib -lnids \
 	        thirdparty/http_parser.o \
 	        thirdparty/js0n.o \
 	        thirdparty/patricia.o \
-		-ldl -lpthread
+		-ldl -lpthread -lgmodule-2.0
 
 C_FILES         = main.c db.c nids.c yara.c http.c config.c detect.c plugins.c field.c trie.c
 O_FILES         = $(C_FILES:.c=.o)
--- moloch-0.9.2.orig/capture/main.c	2013-11-14 14:23:59.000000000 +0000
+++ moloch-0.9.2/capture/main.c	2014-01-13 22:38:12.304500368 +0000
@@ -77,7 +77,7 @@
     }
 
     if (!config.configFile)
-        config.configFile = g_strdup("/data/moloch/etc/config.ini");
+        config.configFile = g_strdup("/etc/moloch/config.ini");
 
     if (showVersion) {
         printf("moloch-capture %s session size=%zd\n", PACKAGE_VERSION, sizeof(MolochSession_t));
