--- src/pymeta/pymeta.py	2018-10-03 12:50:22.106666705 +0200
+++ pymeta.py	2018-10-03 12:50:11.360000023 +0200
@@ -109,7 +109,7 @@
             tmp_head = []
             tmp_data = []
             # Extract metadata in tab-delimited list to split output
-            meta = getoutput("exiftool -t '{}'".format(file_dir + f)).splitlines()
+            meta = getoutput("/usr/bin/vendor_perl/exiftool -t '{}'".format(file_dir + f)).splitlines()
             for m in meta:
                 s = m.split("\t")
                 tmp_head.append(s[0])
@@ -130,7 +130,7 @@
         print("[*] Displaying all unique metadata, use -csv for more")
         found = []
         for f in os.listdir(file_dir):
-            meta = getoutput("exiftool '{}'".format(file_dir+f)).splitlines()
+            meta = getoutput("/usr/bin/vendor_perl/exiftool '{}'".format(file_dir+f)).splitlines()
             for m in meta:
                 n = m.split(":")
                 if "Author" in n[0] or "Creator" == n[0].strip() or "Producer" == n[0].strip():
@@ -159,7 +159,7 @@
 def main(args):
     #Verify Exiftool installed before starting
     try:
-        exif_version = getoutput("exiftool -ver")
+        exif_version = getoutput("/usr/bin/vendor_perl/exiftool -ver")
         float(exif_version)
     except:
         print("\n[!] Exiftool not found, run the setup script to verify\n[*] Closing...\n")
@@ -226,4 +226,4 @@
         main(args)
     except KeyboardInterrupt:
         print("\n[!] Keyboard Interrupt Caught...\n\n")
-        exit(0)
\ No newline at end of file
+        exit(0)
