--- src/domlink/domLink.py	2019-05-26 12:16:22.539999854 +0200
+++ domLink.py	2019-05-26 12:14:43.079999854 +0200
@@ -57,14 +57,14 @@
     for record in ['registrant', 'administrative', 'technical']:
         record = '{}_contact'.format(record)
         email = content.get(record, {}).get('email_address', '').lower()
-	if ( not content.get(record, {}).get('company_name', '') == 'REDACTED FOR PRIVACY'):
-            company = content.get(record, {}).get('company_name', '')
-        if email:
-            logging.debug('domain: adding email {}'.format(email))
-            emails[email] = True
-        if company:
-            logging.debug('domain: adding email {}'.format(email))
-            companies[company] = True
+    if ( not content.get(record, {}).get('company_name', '') == 'REDACTED FOR PRIVACY'):
+      company = content.get(record, {}).get('company_name', '')
+      if email:
+          logging.debug('domain: adding email {}'.format(email))
+          emails[email] = True
+      if company:
+          logging.debug('domain: adding email {}'.format(email))
+          companies[company] = True
     return {'emails': emails, 'companies': companies, 'domains': domains}
 
 
