diff -Naur -x '*~' videosnarf-0.63/configure.in videosnarf-0.63-fix-linking/configure.in
--- videosnarf-0.63/configure.in	2010-05-15 19:40:07.000000000 +0200
+++ videosnarf-0.63-fix-linking/configure.in	2012-02-29 19:33:27.516906867 +0100
@@ -31,4 +31,6 @@
 	;;
 esac
 
+AC_CHECK_LIB(pcap, pcap_compile, , AC_MSG_ERROR(pcap libary required))
+
 AC_OUTPUT(Makefile src/Makefile)
diff -Naur -x '*~' videosnarf-0.63/src/Makefile.am videosnarf-0.63-fix-linking/src/Makefile.am
--- videosnarf-0.63/src/Makefile.am	2010-05-15 19:40:07.000000000 +0200
+++ videosnarf-0.63-fix-linking/src/Makefile.am	2012-02-29 19:33:42.027725455 +0100
@@ -10,8 +10,6 @@
 
 #videosnarf_LDFLAGS = -lpcap -L../codec-lib -lG729a -lG7231 -lG726
 
-videosnarf_LDFLAGS = -lpcap
-
 if ARCH_X32
 
 videosnarf_LDADD = ../codec-lib/libG729a.a \
