--- a/gsad-22.12.0/src/CMakeLists.txt
+++ b/gsad-22.12.0/src/CMakeLists.txt
@@ -34,18 +34,7 @@
 pkg_check_modules (BROTLI libbrotlienc)
 
 message (STATUS "Looking for libgcrypt...")
-find_library (LIBGCRYPT gcrypt)
-if (NOT LIBGCRYPT)
-  message (SEND_ERROR "The libgcrypt library is required.")
-else (NOT LIBGCRYPT)
-  message (STATUS "Looking for libgcrypt... ${LIBGCRYPT}")
-  execute_process (COMMAND libgcrypt-config --libs
-    OUTPUT_VARIABLE LIBGCRYPT_LDFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND libgcrypt-config --cflags
-    OUTPUT_VARIABLE LIBGCRYPT_CFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif (NOT LIBGCRYPT)
+pkg_check_modules (LIBGCRYPT REQUIRED libgcrypt)
 
 if (NOT LIBMICROHTTPD_FOUND OR NOT LIBXML_FOUND OR NOT GLIB_FOUND OR
     (GTHREAD_REQUIRED AND NOT GTHREAD_FOUND) OR NOT

--- a/gsad-22.12.0/src/CMakeLists.txt
+++ b/gsad-22.12.0/src/CMakeLists.txt
@@ -39,13 +39,13 @@
 if (NOT LIBMICROHTTPD_FOUND OR NOT LIBXML_FOUND OR NOT GLIB_FOUND OR
     (GTHREAD_REQUIRED AND NOT GTHREAD_FOUND) OR NOT
     LIBGVM_GMP_FOUND OR NOT GNUTLS_FOUND OR NOT
-    LIBGCRYPT OR NOT ZLIB_FOUND)
+    ZLIB_FOUND)
   message (FATAL_ERROR "One or more required libraries was not found "
     "(see message above), please install the missing "
     "libraries and run cmake again.")
 endif (NOT LIBMICROHTTPD_FOUND OR NOT LIBXML_FOUND OR NOT GLIB_FOUND OR
   (GTHREAD_REQUIRED AND NOT GTHREAD_FOUND) OR NOT
-  LIBGVM_GMP_FOUND OR NOT GNUTLS_FOUND OR NOT LIBGCRYPT OR NOT ZLIB_FOUND)
+  LIBGVM_GMP_FOUND OR NOT GNUTLS_FOUND OR NOT ZLIB_FOUND)
 
 ## Program
 
