--- src/oletools-0.60/setup.py	2021-06-02 16:30:56.000000000 +0200
+++ setup.py	2022-03-27 16:47:58.686665692 +0200
@@ -319,14 +319,14 @@
         test_suite="tests",
         # scripts=scripts,
         install_requires=[
-            "pyparsing>=2.1.0,<3",  # changed from 2.2.0 to 2.1.0 for issue #481
-            "olefile>=0.46",
+            "pyparsing",  # changed from 2.2.0 to 2.1.0 for issue #481
+            "olefile",
             "easygui",
             'colorclass',
             # msoffcrypto-tool is not installable on PyPy+Windows (see issue #473),
             # so we only require it if the platform is not Windows or not PyPy:
             'msoffcrypto-tool; platform_python_implementation!="PyPy" or (python_version>="3" and platform_system!="Windows" and platform_system!="Darwin")',
-            'pcodedmp>=1.2.5',
+            'pcodedmp',
         ],
         extras_require={
             # Optional packages - to be installed with pip install -U oletools[full]
