--- src/rmd_getzpixmap.c	2008-12-13 17:04:10.000000000 +0100
+++ src/rmd_getzpixmap.c	2009-11-11 11:30:18.000000000 +0100
@@ -32,7 +32,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xlibint.h>
-#include <X11/extensions/shmstr.h>
+#include <X11/extensions/shmproto.h>
 #include <X11/extensions/XShm.h>
 
 
--- src/rmd_update_image.c	2008-12-13 17:20:49.000000000 +0100
+++ src/rmd_update_image.c	2009-11-11 11:30:34.000000000 +0100
@@ -32,7 +32,7 @@
 #include "rmd_types.h"
 
 #include <X11/Xlibint.h>
-#include <X11/extensions/shmstr.h>
+#include <X11/extensions/shmproto.h>
 #include <X11/extensions/XShm.h>
