diff -rud z3-z3-4.4.1/src/util/debug.cpp z3-z3-4.4.1-patched/src/util/debug.cpp
--- z3-z3-4.4.1/src/util/debug.cpp	2015-10-05 13:07:19.000000000 +0100
+++ z3-z3-4.4.1-patched/src/util/debug.cpp	2016-08-15 11:48:14.718440841 +0100
@@ -76,7 +76,7 @@
     for (;;) {
         std::cerr << "(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB\n";
         char result;
-        bool ok = (std::cin >> result);
+        bool ok = static_cast<bool>(std::cin >> result);
         if (!ok) exit(ERR_INTERNAL_FATAL); // happens if std::cin is eof or unattached.
         switch(result) {
         case 'C':
diff -rud z3-z3-4.4.1/src/util/mpz.cpp z3-z3-4.4.1-patched/src/util/mpz.cpp
--- z3-z3-4.4.1/src/util/mpz.cpp	2015-10-05 13:07:19.000000000 +0100
+++ z3-z3-4.4.1-patched/src/util/mpz.cpp	2016-08-15 12:02:35.178737211 +0100
@@ -134,7 +134,7 @@
 #endif
     
     mpz one(1);
-    set(m_two64, UINT64_MAX);
+    set(m_two64, static_cast<uint64>(UINT64_MAX));
     add(m_two64, one, m_two64);
 }
 
