From: Stephane Glondu <steph@glondu.net>
Date: Fri, 29 Jul 2011 19:46:45 +0200
Subject: Install missing camomileLibrary.a

Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index e1fc579..45dc0ac 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -406,7 +406,7 @@ install-with-ocamlfind:
 	files= &&\
 	if [ -f camomileLibrary.cmx ]; then files=camomileLibrary.cmx; fi && \
 	if [ -f camomileLibrary.cma ]; then files="camomileLibrary.cma $$files"; fi && \
-	if [ -f camomileLibrary.cmxa ]; then files="camomileLibrary.cmxa $$files"; fi && \
+	if [ -f camomileLibrary.cmxa ]; then files="camomileLibrary.cmxa camomileLibrary.a $$files"; fi && \
 	if [ -f camomileLibraryDefault.cmx ]; then files="camomileLibraryDefault.cmx $$files"; fi && \
 	if [ -f camomileLibraryDyn.cmx ]; then files="camomileLibraryDyn.cmx $$files"; fi && \
 	if [ -f camomile.cma ]; then files="camomile.cma $$files"; fi&& \
-- 
