--- Samba/source/lib/tls/tls.c-orig	2012-09-24 09:57:44.000000000 -0600
+++ Samba/source/lib/tls/tls.c	2012-09-24 09:58:04.000000000 -0600
@@ -505,7 +505,6 @@
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
-	gnutls_transport_set_lowat(tls->session, 0);
 
 	tls->plain_chars = plain_chars;
 	if (plain_chars) {
@@ -584,7 +583,6 @@
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
-	gnutls_transport_set_lowat(tls->session, 0);
 	tls->tls_detect = False;
 
 	tls->output_pending  = False;
