--- qt-x11-free-3.3.5/tools/assistant/lib/lib.pro.orig	2005-10-21 10:09:12.000000000 -0200
+++ qt-x11-free-3.3.5/tools/assistant/lib/lib.pro	2005-10-21 10:09:34.000000000 -0200
@@ -3,8 +3,7 @@
 VERSION		= 1.0
 
 CONFIG		+= qt warn_on release
-CONFIG		+= staticlib
-CONFIG		-= dll
+CONFIG		+= dll
 
 SOURCES		= qassistantclient.cpp
 HEADERS         += $$QT_SOURCE_TREE/include/qassistantclient.h
*** qt-x11-free-qt-copy-3.3.5/tools/designer/editor/editor.pro	2003-12-20 22:48:51.000000000 -0200
--- qt-x11-free-qt-copy-3.3.5/tools/designer/editor/editor.pro.new	2005-12-06 18:47:04.000000000 -0200
***************
*** 1,6 ****
  TEMPLATE	= lib
! CONFIG		+= qt warn_on staticlib
! CONFIG		-= dll
  HEADERS		= editor.h  \
  		  parenmatcher.h  \
  		  completion.h \
--- 1,6 ----
  TEMPLATE	= lib
! CONFIG		+= qt warn_on
! CONFIG		+= dll
  HEADERS		= editor.h  \
  		  parenmatcher.h  \
  		  completion.h \
*** qt-x11-free-qt-copy-3.3.5/tools/designer/designer/designer.pro	2004-04-29 19:31:32.000000000 -0300
--- qt-x11-free-qt-copy-3.3.5/tools/designer/designer/designer.pro.new	2005-12-06 18:46:29.000000000 -0200
***************
*** 1,7 ****
  TEMPLATE	= lib
  
! CONFIG		+= qt warn_on staticlib qmake_cache
! CONFIG 		-= dll
  !force_static:!win32:contains(QT_PRODUCT,qt-internal) {
     CONFIG          -= staticlib
     CONFIG          += dll
--- 1,7 ----
  TEMPLATE	= lib
  
! CONFIG		+= qt warn_on qmake_cache
! CONFIG 		+= dll
  !force_static:!win32:contains(QT_PRODUCT,qt-internal) {
     CONFIG          -= staticlib
     CONFIG          += dll
