--- configure.in	2009-10-28 17:49:11.886060763 +0100
+++ configure.in	2009-10-28 17:50:42.226711119 +0100
@@ -598,6 +598,10 @@
 #
 if test "x$enable_wireshark" = "xyes"; then
 	AM_PATH_GTK_2_0(2.4.0, CFLAGS="$CFLAGS $GTK_CFLAGS", GTK_OK=no, gthread)
+
+    PKG_CHECK_MODULES(GTKHEX, [ gtkhex >= 1.0.0 ])
+    AC_SUBST(GTKHEX_CFLAGS)
+    AC_SUBST(GTKHEX_LIBS)
 else
 	GTK_OK=no
 fi
--- Makefile.am	2009-10-28 17:52:41.009200229 +0100
+++ Makefile.am	2009-10-28 17:52:04.332517449 +0100
@@ -338,7 +338,9 @@
 	@LIBGCRYPT_LIBS@		\
 	@LIBGNUTLS_LIBS@		\
 	@LIBSMI_LDFLAGS@		\
-	@PORTAUDIO_LIBS@
+	@PORTAUDIO_LIBS@        \
+	$(GTKHEX_LIBS)          \
+	-lorcon
 wireshark_CFLAGS = $(AM_CLEAN_CFLAGS) $(py_dissectors_dir)
 
 if ENABLE_STATIC
--- gtk/Makefile.am	2009-10-28 17:53:23.379172192 +0100
+++ gtk/Makefile.am	2009-10-28 17:54:01.982534556 +0100
@@ -50,7 +50,7 @@
 libui_a_DEPENDENCIES = 
 
 # Common headers
-AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/wiretap $(LIBGCRYPT_CFLAGS) $(LIBGNUTLS_CFLAGS) $(PORTAUDIO_INCLUDES)
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/wiretap $(LIBGCRYPT_CFLAGS) $(LIBGNUTLS_CFLAGS) $(PORTAUDIO_INCLUDES) $(GTKHEX_CFLAGS)
 
 
 doxygen:
