--- src/inundator-0.5/inundator.pl	2019-11-16 11:55:43.509998103 +0100
+++ inundator.pl	2019-11-16 11:57:52.739998100 +0100
@@ -45,14 +45,14 @@
 
 ## version info
 my $VERSION = "0.5";
- 
+
 ## local variables
 my $target;	# user-supplied target
 my $rules;	# user-supplied rules dir
 my $thrnum;	# number of threads to spawn
 my $usethreads;	# can we use threads?
 
-## global variables 
+## global variables
 our $delay;      # usleep() delay
 our $proxy_addr; # addr of proxy server
 our $proxy_port; # proxy server port
@@ -174,7 +174,7 @@
 
 	## grab all attacks for the selected port
         my @targeted = ();
-        @targeted = @{$attacks[$rand_port]} if defined(@{$attacks[$rand_port]});
+        @targeted = @{$attacks[$rand_port]} if @{$attacks[$rand_port]};
 
 	## grab all the attacks for any port
 	my @any_port = @{$attacks[0]};
