--- nsat/configure.in.orig	2003-01-05 22:47:13.000000000 +0100
+++ nsat/configure.in	2009-01-15 07:56:35.000000000 +0100
@@ -15,7 +15,6 @@
 else
     AC_MSG_RESULT(crappy :P)
 fi
- LEETCC=$ac_cv_prog_gcc;
 AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
@@ -45,11 +44,11 @@
 if test $LEETCC = yes; then
     CCOPTS='-Wall -O6 -funroll-loops -ansi -fPIC -DLINUX'
 else
-    CCOPTS='-DLINUX'
+    CCOPTS="${CFLAGS} -DLINUX"
 fi
-    CFLAGS="$CCOPTS"
-    SMBCFLAGS="-Wall -DLINUX -DNETGROUP -DALLOW_CHANGE_PASSWORD -DFAST_SHARE_MODES -DNO_ASMSIGNALH -DGLIBC2"
+    SMBCFLAGS="${CCOPTS} -Wall -DLINUX -DNETGROUP -DALLOW_CHANGE_PASSWORD -DFAST_SHARE_MODES -DNO_ASMSIGNALH -DGLIBC2"
     SMBLFLAGS="-lnsl -lcrypt"
+    CFLAGS="${CCOPTS} -Wall -ansi -fPIC -DLINUX"
  ;;
  *Darwin* | *Mac* | *MAC*)
   AC_MSG_RESULT(found MAC OS X)
