diff -Naur nsat.old/src/smb/ADMsmb.c nsat/src/smb/ADMsmb.c
--- nsat.old/src/smb/ADMsmb.c	2006-03-31 01:51:26.000000000 -0500
+++ nsat/src/smb/ADMsmb.c	2006-03-31 01:52:27.000000000 -0500
@@ -123,7 +123,7 @@
     {
       usleep (100000);
 
-      if (recvfrom (socket_client, buffer2, sizeof (buffer2), 0, (struct sockaddr *) &sin_dst, &(int) longueur) != -1)
+      if (recvfrom (socket_client, buffer2, sizeof (buffer2), 0, (struct sockaddr *) &sin_dst, (int *) &longueur) != -1)
 	{
 
 	  if (nmb2->rep_num != 0)
