diff --git a/Gemfile b/Gemfile
index d2bfc80d1..30da52f06 100644
--- a/Gemfile
+++ b/Gemfile
@@ -23,4 +23,8 @@ group :prof do
     gem 'memory_profiler'
 end
 
+group :fufufu do
+    gem 'puma', '2.14.0', github: 'lsfxz/puma-2.14.0', branch: 'master'
+end
+
 gemspec
diff --git a/arachni.gemspec b/arachni.gemspec
index 4704128e8..0646edea7 100644
--- a/arachni.gemspec
+++ b/arachni.gemspec
@@ -49,8 +49,8 @@ Gem::Specification.new do |s|
     # latest one.
     s.add_dependency 'bundler'
 
-    s.add_dependency 'concurrent-ruby',     '1.0.2'
-    s.add_dependency 'concurrent-ruby-ext', '1.0.2'
+    s.add_dependency 'concurrent-ruby',     '1.0.4'
+    s.add_dependency 'concurrent-ruby-ext', '1.0.4'
 
     # For compressing/decompressing system state archives.
     s.add_dependency 'rubyzip',             '1.2.1'
@@ -64,16 +64,16 @@ Gem::Specification.new do |s|
     s.add_dependency 'childprocess',        '0.5.3'
 
     # RPC serialization.
-    s.add_dependency 'msgpack',             '0.7.0'
+    s.add_dependency 'msgpack',             '~> 1.0.0'
 
     if RUBY_PLATFORM != 'java'
         # Optimized JSON.
-        s.add_dependency 'oj',              '2.15.0'
+        s.add_dependency 'oj',              '~> 2.18.0'
         s.add_dependency 'oj_mimic_json',   '1.0.1'
     end
 
     # Web server
-    s.add_dependency 'puma',                '2.14.0'
+    # s.add_dependency 'puma',                '2.14.0'
 
     # REST API
     s.add_dependency 'sinatra',             '1.4.6'
