diff -aur arachni_pristine/arachni.gemspec arachni/arachni.gemspec
--- arachni_pristine/arachni.gemspec	2017-01-14 15:51:35.515330816 +0100
+++ arachni/arachni.gemspec	2017-01-14 15:44:18.516329716 +0100
@@ -50,8 +50,8 @@
     # latest one.
     s.add_dependency 'bundler'
 
-    s.add_dependency 'concurrent-ruby',     '1.0.2'
-    s.add_dependency 'concurrent-ruby-ext', '1.0.2'
+    s.add_dependency 'concurrent-ruby',     '1.0.4'
+    s.add_dependency 'concurrent-ruby-ext', '1.0.4'
 
     # For compressing/decompressing system state archives.
     s.add_dependency 'rubyzip',             '1.1.6'
@@ -65,11 +65,11 @@
     s.add_dependency 'childprocess',        '0.5.3'
 
     # RPC serialization.
-    s.add_dependency 'msgpack',             '0.7.0'
+    s.add_dependency 'msgpack',             '~> 1.0.0'
 
     if RUBY_PLATFORM != 'java'
         # Optimized JSON.
-        s.add_dependency 'oj',              '2.15.0'
+        s.add_dependency 'oj',              '~> 2.18.0'
         s.add_dependency 'oj_mimic_json',   '1.0.1'
     end
 
@@ -106,7 +106,8 @@
     # Browser support for DOM/JS/AJAX analysis stuff.
     # Lock webdriver, newer versions has issues.
     s.add_dependency 'selenium-webdriver',  '3.0.1'
-    s.add_dependency 'watir-webdriver',     '0.8.0'
+    s.add_dependency 'watir',               '~> 6.1.0'
+    # s.add_dependency 'watir-webdriver',     '~> 0.9.0'
 
     # Markdown to HTML conversion, used by the HTML report for component
     # descriptions.
diff -aur arachni_pristine/lib/arachni/browser.rb arachni/lib/arachni/browser.rb
--- arachni_pristine/lib/arachni/browser.rb	2017-01-14 15:51:35.530330816 +0100
+++ arachni/lib/arachni/browser.rb	2017-01-14 15:44:18.517329716 +0100
@@ -7,7 +7,7 @@
 =end
 
 require 'childprocess'
-require 'watir-webdriver'
+require 'watir'
 require_relative 'selenium/webdriver/element'
 require_relative 'processes/manager'
 require_relative 'browser/element_locator'
