--- src/gdbgui/setup.py	2018-05-13 23:03:30.135599456 +0200
+++ setup.py	2018-05-13 23:03:56.502266118 +0200
@@ -13,12 +13,12 @@
 
 EXCLUDE_FROM_PACKAGES = []
 REQUIRED = [
-    'Flask>=0.12.2,<1.0',  # to run server
-    'Flask-Compress>=1.4.0',  # to compress flask responses
-    'Flask-SocketIO>=2.9.3',  # for websockets
-    'gevent>=1.2.2',  # for websockets (preferred)
-    'pygdbmi>=0.8.2.0',  # to parse gdb output
-    'Pygments>=2.2.0',  # for syntax highlighting
+    'Flask',  # to run server
+    'Flask-Compress',  # to compress flask responses
+    'Flask-SocketIO',  # for websockets
+    'gevent',  # for websockets (preferred)
+    'pygdbmi',  # to parse gdb output
+    'Pygments',  # for syntax highlighting
 ]
 
 README = io.open(os.path.join(CURDIR, 'README.rst'), 'r', encoding="utf-8").read()
