--- src/gdbgui/setup.py	2020-12-04 22:16:33.609998662 +0100
+++ setup.py	2020-12-04 22:17:37.083331990 +0100
@@ -59,15 +59,15 @@
     },
     zip_safe=False,
     install_requires=[
-        "Flask>=0.12.2, <1.0",  # http server
-        "Flask-Compress>=1.4.0, <2.0",  # to compress flask responses
-        "Flask-SocketIO>=2.9, <3.0",  # websocket server
-        "gevent>=1.2.2, <2.0",  # websocket handling
-        "gevent-websocket>=0.10.1, <0.11",  # also websocket
-        "eventlet>=0.25.0, <0.26",  # also websocket
-        "pygdbmi>=0.10.0.0b0, <0.11",  # parse gdb output
-        "Pygments>=2.2.0, <3.0",  # syntax highlighting
-        "greenlet==0.4.16",
+        "Flask",  # http server
+        "Flask-Compress",  # to compress flask responses
+        "Flask-SocketIO",  # websocket server
+        "gevent",  # websocket handling
+        "gevent-websocket",  # also websocket
+        "eventlet",  # also websocket
+        "pygdbmi",  # parse gdb output
+        "Pygments",  # syntax highlighting
+        "greenlet",
     ],
     classifiers=[
         "Intended Audience :: Developers",
