--- src/HammerTime-http-0.8/setup.py	2021-03-17 20:21:27.000000000 +0100
+++ setup.py	2022-09-03 08:30:34.769999861 +0200
@@ -21,9 +21,9 @@
                 'hammertime.rules',
                 'hammertime.utils'],
       install_requires=[
-          'aiohttp>=3.7.3,<4.0.0',
-          'easyinject==0.3',
-          'aiodns>=1.1.1,<3.0.0'
+          'aiohttp',
+          'easyinject',
+          'aiodns'
       ],
       extras_require={
           'simhash':  ['simhash==2.0.0'],
