diff -Nur winexe-1.00.orig/source4/lib/tls/tls.c winexe-1.00/source4/lib/tls/tls.c
--- winexe-1.00.orig/source4/lib/tls/tls.c	2011-06-16 08:56:58.000000000 +1000
+++ winexe-1.00/source4/lib/tls/tls.c	2011-09-21 10:20:40.052225642 +1000
@@ -505,7 +505,9 @@
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
 	gnutls_transport_set_lowat(tls->session, 0);
+#endif
 
 	tls->plain_chars = plain_chars;
 	if (plain_chars) {
@@ -574,7 +576,9 @@
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
 	gnutls_transport_set_lowat(tls->session, 0);
+#endif
 	tls->tls_detect = false;
 
 	tls->output_pending  = false;
