diff -Nur winexe-1.00.orig/pidl/idl.yp winexe-1.00/pidl/idl.yp
--- winexe-1.00.orig/pidl/idl.yp	2011-06-16 08:56:58.000000000 +1000
+++ winexe-1.00/pidl/idl.yp	2013-05-26 00:30:32.097102856 +1000
@@ -585,7 +585,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{FILE} = $2;
 				goto again;
diff -Nur winexe-1.00.orig/pidl/lib/Parse/Pidl/IDL.pm winexe-1.00/pidl/lib/Parse/Pidl/IDL.pm
--- winexe-1.00.orig/pidl/lib/Parse/Pidl/IDL.pm	2011-06-16 08:56:58.000000000 +1000
+++ winexe-1.00/pidl/lib/Parse/Pidl/IDL.pm	2013-05-26 00:30:02.863770217 +1000
@@ -2545,7 +2545,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{FILE} = $2;
 				goto again;
