diff -Nur winexe-1.00.orig/pidl/idl.yp winexe-1.00/pidl/idl.yp
--- winexe-1.00.orig/pidl/idl.yp	2011-06-16 08:56:58.000000000 +1000
+++ winexe-1.00/pidl/idl.yp	2013-05-26 00:30:32.097102856 +1000
@@ -585,7 +585,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{FILE} = $2;
 				goto again;
diff -Nur winexe-1.00.orig/pidl/lib/Parse/Pidl/IDL.pm winexe-1.00/pidl/lib/Parse/Pidl/IDL.pm
--- winexe-1.00.orig/pidl/lib/Parse/Pidl/IDL.pm	2011-06-16 08:56:58.000000000 +1000
+++ winexe-1.00/pidl/lib/Parse/Pidl/IDL.pm	2013-05-26 00:30:02.863770217 +1000
@@ -2545,7 +2545,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{FILE} = $2;
 				goto again;
diff -Nur winexe-1.00.orig/pidl/lib/Parse/Pidl/ODL.pm winexe-1.00/pidl/lib/Parse/Pidl/ODL.pm
--- winexe-1.00.orig/pidl/lib/Parse/Pidl/ODL.pm	2011-06-16 00:56:58.000000000 +0200
+++ winexe-1.00/pidl/lib/Parse/Pidl/ODL.pm	2015-07-16 17:22:09.547485504 +0200
@@ -70,7 +70,7 @@
 					next;
 				}
 				my $podl = Parse::Pidl::IDL::parse_file($idl_path, $opt_incdirs);
-				if (defined(@$podl)) {
+				if (@$podl) {
 					require Parse::Pidl::Typelist;
 					my $basename = basename($idl_path, ".idl");
 
diff -Nur winexe-1.00.orig/pidl/pidl winexe-1.00/pidl/pidl
--- winexe-1.00.orig/pidl/pidl	2011-06-16 00:56:58.000000000 +0200
+++ winexe-1.00/pidl/pidl	2015-07-16 17:22:19.477380979 +0200
@@ -605,7 +605,7 @@
 		require Parse::Pidl::IDL;
 
 		$pidl = Parse::Pidl::IDL::parse_file($idl_file, \@opt_incdirs);
-		defined @$pidl || die "Failed to parse $idl_file";
+		@$pidl || die "Failed to parse $idl_file";
 	}
 
 	require Parse::Pidl::Typelist;

