diff -ur binwalk-1.3.0.orig/src/binwalk/update.py binwalk-1.3.0/src/binwalk/update.py
--- binwalk-1.3.0.orig/src/binwalk/update.py	2013-12-12 16:11:50.000000000 +0100
+++ binwalk-1.3.0/src/binwalk/update.py	2014-02-04 13:03:12.012589869 +0100
@@ -37,8 +37,8 @@
 		self.update_bincast()
 		self.update_binarch()
 		self.update_extract()
-		self.update_zlib()
-		self.update_compressd()
+#		self.update_zlib()
+#		self.update_compressd()
 
 	def _do_update_from_svn(self, prefix, fname):
 		'''
diff -ur binwalk-1.3.0.orig/src/setup.py binwalk-1.3.0/src/setup.py
--- binwalk-1.3.0.orig/src/setup.py	2013-12-12 16:11:50.000000000 +0100
+++ binwalk-1.3.0/src/setup.py	2014-02-04 13:02:55.589255872 +0100
@@ -65,33 +65,33 @@
 	warning(msg, prompt=True)
 
 # Build / install C compression libraries
-c_lib_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "C")
-c_lib_makefile = os.path.join(c_lib_dir, "Makefile")
-
-working_directory = os.getcwd()
-os.chdir(c_lib_dir)
-status = 0
-
-if not os.path.exists(c_lib_makefile):
-	status |= os.system("./configure")
-
-status |= os.system("make")
-
-if status != 0:
-	msg = ["Build warning: failed to build compression libraries.",
-		"Some plugins will not work without these libraries."
-	]
-	
-	warning(msg, prompt=True)
-elif "install" in sys.argv:
-		if os.system("make install") != 0:
-			msg = ["Install warning: failed to install compression libraries.",
-				"Some plugins will not work without these libraries."
-			]
-
-			warning(msg, prompt=True)
-		
-os.chdir(working_directory)
+#c_lib_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "C")
+#c_lib_makefile = os.path.join(c_lib_dir, "Makefile")
+#
+#working_directory = os.getcwd()
+#os.chdir(c_lib_dir)
+#status = 0
+#
+#if not os.path.exists(c_lib_makefile):
+#	status |= os.system("./configure")
+#
+#status |= os.system("make")
+#
+#if status != 0:
+#	msg = ["Build warning: failed to build compression libraries.",
+#		"Some plugins will not work without these libraries."
+#	]
+#	
+#	warning(msg, prompt=True)
+#elif "install" in sys.argv:
+#		if os.system("make install") != 0:
+#			msg = ["Install warning: failed to install compression libraries.",
+#				"Some plugins will not work without these libraries."
+#			]
+#
+#			warning(msg, prompt=True)
+#		
+#os.chdir(working_directory)
 
 # Generate a new magic file from the files in the magic directory
 print("generating binwalk magic file")
