--- Makefile.orig	2014-08-29 13:19:03.872987035 -0400
+++ Makefile	2014-08-29 17:19:38.969019430 -0400
@@ -3,6 +3,7 @@
 export SONAME=-soname
 export SOEXT=so
 export prefix=/usr
+export DESTDIR=""
 export exec_prefix=${prefix}
 export LIBDIR=${exec_prefix}/lib
 export INSTALL_OPTIONS=-m644
@@ -22,7 +23,7 @@
 ifeq ($(strip $(prefix)),)
 	PREFIX=""
 else
-	PREFIX="--prefix=$(prefix)"
+	PREFIX="--prefix=$(DESTDIR)$(prefix)"
 endif

 .PHONY: all install build deps clean uninstall
@@ -30,7 +31,7 @@
 all: build

 install: build
-	$(PYTHON) ./setup.py install $(PREFIX)
+	$(PYTHON) ./setup.py install $(PREFIX) --optimize=1

 build:
 	if [ "$(BUILD_C_LIBS)" -eq "1" ]; then make -C $(SRC_C_DIR); fi
