--- deps.sh.orig	2014-05-26 13:23:08.445233248 +0100
+++ deps.sh	2014-05-26 12:54:25.335178904 +0100
@@ -52,6 +52,22 @@ function fmk
 }
 
 
+function archlinux
+{
+        if [ ! -f /usr/bin/lha ]; then
+            wget https://aur.archlinux.org/packages/lh/lha/lha.tar.gz
+            cd lha
+            makepkg -c
+            cd ..
+        fi
+
+        if [ ! -f /usr/local/bin/uncompress ]; then
+            wget https://aur.archlinux.org/packages/nc/ncompress/ncompress.tar.gz
+            cd ncompress
+            makepkg -c
+            cd ..
+        fi
+}
 function debian
 {
 	# The appropriate unrar package goes under different names in Debian vs Ubuntu
@@ -113,7 +129,6 @@ case $DISTRO in
 	aptosid)
 		debian
 		;;
-
 	redhat)
 		redhat
 		;;
@@ -126,6 +141,9 @@ case $DISTRO in
 	centos)
 		redhat
 		;;
+        arch)
+                archlinux
+                ;;
 	*)
 		echo ""
 		echo "This system is not recognized by easy install! You may need to install dependent packages manually."
