--- Makefile	2014-08-31 00:17:59.700016566 +0200
+++ Makefile.new	2014-08-31 00:21:12.158973926 +0200
@@ -11,10 +11,11 @@
 export BUILD_FUZZY=1
 export BUILD_PYQTGRAPH=1
 export PYLIBDIR="./binwalk/libs"
+export DESTDIR=""
 
 BUILD_C_LIBS=1
 BUILD_BUNDLES=1
-PYTHON=python
+PYTHON=python2
 
 SRC_C_DIR="./src/C"
 SRC_BUNDLES_DIR="./src/bundles"
@@ -22,7 +23,7 @@
 ifeq ($(strip $(prefix)),)
 	PREFIX=""
 else
-	PREFIX="--prefix=$(prefix)"
+	PREFIX="--prefix=$(DESTDIR)$(prefix)"
 endif
 
 .PHONY: all install build deps clean uninstall
@@ -30,7 +31,7 @@
 all: build
 
 install: build
-	$(PYTHON) ./setup.py install $(PREFIX)
+	$(PYTHON) ./setup.py install $(PREFIX) --optimize=1
 
 build:
 	if [ "$(BUILD_C_LIBS)" -eq "1" ]; then make -C $(SRC_C_DIR); fi
