--- setup.py.old	2013-04-05 13:38:53.107076231 +0200
+++ setup.py	2013-04-05 13:39:38.167076141 +0200
@@ -5,41 +5,7 @@
 
 WIDTH = 115
 
-# Check for pre-requisite modules
-print "checking pre-requisites"
-try:
-	import magic
-	try:
-		magic.MAGIC_NO_CHECK_TEXT
-	except Exception, e:
-		print "\n", "*" * WIDTH
-		print "Pre-requisite failure:", str(e)
-		print "It looks like you have an old or incompatible magic module installed."
-		print "Please install the official python-magic module, or download and install it from source: ftp://ftp.astron.com/pub/file/"
-		print "*" * WIDTH, "\n"
-		sys.exit(1)
-except Exception, e:
-	print "\n", "*" * WIDTH
-	print "Pre-requisite failure:", str(e)
-	print "Please install the python-magic module, or download and install it from source: ftp://ftp.astron.com/pub/file/"
-	print "*" * WIDTH, "\n"
-	sys.exit(1)
-
-try:
-	import matplotlib.pyplot
-except Exception, e:
-	print "\n", "*" * WIDTH
-	print "Pre-requisite check warning:", str(e)
-	print "To take advantage of this tool's entropy plotting capabilities, please install the python-matplotlib module."
-	print "*" * WIDTH, "\n"
-	
-	if raw_input('Continue installation without this module (Y/n)? ').lower().startswith('n'):
-		print 'Quitting...\n'
-		sys.exit(1)
-		
-
 # Generate a new magic file from the files in the magic directory
-print "generating binwalk magic file"
 magic_files = listdir("magic")
 magic_files.sort()
 fd = open("binwalk/magic/binwalk", "wb")
