diff -rupN b/httpd.conf a/httpd.conf
--- b/httpd.conf	2014-05-02 20:01:19.733444439 +0100
+++ a/httpd.conf	2014-05-02 20:01:14.678361445 +0100
@@ -150,7 +150,7 @@ LoadModule lbmethod_byrequests_module mo
 LoadModule lbmethod_bytraffic_module modules/mod_lbmethod_bytraffic.so
 LoadModule lbmethod_bybusyness_module modules/mod_lbmethod_bybusyness.so
 LoadModule lbmethod_heartbeat_module modules/mod_lbmethod_heartbeat.so
-LoadModule mpm_event_module modules/mod_mpm_event.so
+LoadModule mpm_prefork_module modules/mod_mpm_prefork.so
 LoadModule unixd_module modules/mod_unixd.so
 #LoadModule heartbeat_module modules/mod_heartbeat.so
 #LoadModule heartmonitor_module modules/mod_heartmonitor.so
@@ -167,6 +167,7 @@ LoadModule autoindex_module modules/mod_
 #LoadModule vhost_alias_module modules/mod_vhost_alias.so
 LoadModule negotiation_module modules/mod_negotiation.so
 LoadModule dir_module modules/mod_dir.so
+LoadModule php5_module modules/libphp5.so
 #LoadModule imagemap_module modules/mod_imagemap.so
 #LoadModule actions_module modules/mod_actions.so
 #LoadModule speling_module modules/mod_speling.so
@@ -265,6 +266,8 @@ DocumentRoot "/srv/http"
     # Controls who can get stuff from this server.
     #
     Require all granted
+
+    php_admin_value open_basedir none
 </Directory>
 
 #
@@ -504,6 +507,7 @@ Include conf/extra/httpd-default.conf
 Include conf/extra/proxy-html.conf
 </IfModule>
 
+Include conf/extra/php5_module.conf
 # Secure (SSL/TLS) connections
 #Include conf/extra/httpd-ssl.conf
 #
