From d7bd0b7ac3c36b8aacacb14baa84c8a1ee17f43e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C3=ABl=20Zasso?= <targos@protonmail.com>
Date: Wed, 19 Oct 2022 18:30:13 +0200
Subject: [PATCH] fixup! deps: update ICU to 72.1

fix tests

diff --git a/test/parallel/test-intl.js b/test/parallel/test-intl.js
index 7d1742f2c7d1..df391bd4de3d 100644
--- a/test/parallel/test-intl.js
+++ b/test/parallel/test-intl.js
@@ -97,7 +97,7 @@ if (!common.hasIntl) {
   // Test format
   {
     const localeString = date0.toLocaleString(['en'], optsGMT);
-    assert.strictEqual(localeString, '1/1/1970, 12:00:00 AM');
+    assert.strictEqual(localeString, '1/1/1970, 12:00:00 AM');
   }
   // number format
   {
