diff -Naur vnc-4_1_3-unixsrc/common/network/TcpSocket.cxx vnc-4_1_3-unixsrc-new/common/network/TcpSocket.cxx
--- common/network/TcpSocket.cxx	2008-10-16 11:16:20.000000000 -0400
+++ common/network/TcpSocket.cxx	2009-02-26 00:46:38.000000000 -0500
@@ -35,6 +35,7 @@
 #include <string.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <cstdlib>
 #endif
 
 #include <network/TcpSocket.h>
diff -Naur vnc-4_1_3-unixsrc/unix/tx/TXImage.cxx vnc-4_1_3-unixsrc-new/unix/tx/TXImage.cxx
--- unix/tx/TXImage.cxx	2008-10-16 11:16:21.000000000 -0400
+++ unix/tx/TXImage.cxx	2009-02-26 00:47:53.000000000 -0500
@@ -33,6 +33,7 @@
 #include <rfb/LogWriter.h>
 #include "TXWindow.h"
 #include "TXImage.h"
+#include <cstdlib>
 
 using namespace rfb;
 
diff -Naur vnc-4_1_3-unixsrc/unix/x0vncserver/Image.cxx vnc-4_1_3-unixsrc-new/unix/x0vncserver/Image.cxx
--- unix/x0vncserver/Image.cxx	2008-10-16 11:16:21.000000000 -0400
+++ unix/x0vncserver/Image.cxx	2009-02-26 00:47:29.000000000 -0500
@@ -29,6 +29,7 @@
 #include <X11/extensions/XShm.h>
 #include "Image.h"
 #include <list>
+#include <cstdlib>
 
 class ImageCleanup {
 public:
