--- Makefile.orig	2015-02-11 20:38:37.533248594 -0500
+++ Makefile	2015-02-11 20:40:00.349915218 -0500
@@ -13,13 +13,13 @@
 	rm -rf $(OBJDIR) src/
 	rm -rf prebuilt/win64/capstone.dll
 	rm -rf prebuilt/win32/capstone.dll
-	python setup.py build -b $(OBJDIR) install
+	python2 setup.py install --root= --optimize=1
 
 install3:
 	rm -rf $(OBJDIR) src/
 	rm -rf prebuilt/win64/capstone.dll
 	rm -rf prebuilt/win32/capstone.dll
-	python3 setup.py build -b $(OBJDIR) install
+	python3 setup.py install --root= --optimize=1
 
 # NOTE: Newer cython can be installed by: sudo pip install --upgrade cython
 install_cython:
