Index: google-mock-1.6.0/scripts/generator/gmock_gen.py
===================================================================
--- google-mock-1.6.0.orig/scripts/generator/gmock_gen.py	2011-07-01 02:41:39.819517320 +0200
+++ google-mock-1.6.0/scripts/generator/gmock_gen.py	2011-07-01 02:43:26.396045796 +0200
@@ -23,7 +23,7 @@
 
 if __name__ == '__main__':
   # Add the directory of this script to the path so we can import gmock_class.
-  sys.path.append(os.path.dirname(__file__))
+  sys.path.append(os.path.join(os.path.dirname(os.path.dirname(__file__)), "share", "gmock"))
 
   from cpp import gmock_class
   # Fix the docstring in case they require the usage.
