diff -urN gmock-1.7.0.orig/gtest/Makefile.am gmock-1.7.0/gtest/Makefile.am
--- gmock-1.7.0.orig/gtest/Makefile.am	2013-09-23 22:34:27.649060735 -0700
+++ gmock-1.7.0/gtest/Makefile.am	2013-09-23 22:39:16.097069357 -0700
@@ -299,8 +299,6 @@
 
 install-exec-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
 
 install-data-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
diff -urN gmock-1.7.0.orig/gtest/Makefile.in gmock-1.7.0/gtest/Makefile.in
--- gmock-1.7.0.orig/gtest/Makefile.in	2013-09-23 22:34:27.654060735 -0700
+++ gmock-1.7.0/gtest/Makefile.in	2013-09-23 22:41:35.593073526 -0700
@@ -1349,11 +1349,9 @@
 
 install-exec-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
 
 install-data-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Test into your build system."
-	false
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -urN gmock-1.7.0.orig/Makefile.am gmock-1.7.0/Makefile.am
--- gmock-1.7.0.orig/Makefile.am	2013-09-23 22:34:27.644060735 -0700
+++ gmock-1.7.0/Makefile.am	2013-09-23 22:42:00.442074269 -0700
@@ -209,8 +209,6 @@
 
 install-exec-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Mock into your build system."
-	false
 
 install-data-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Mock into your build system."
-	false
diff -urN gmock-1.7.0.orig/Makefile.in gmock-1.7.0/Makefile.in
--- gmock-1.7.0.orig/Makefile.in	2013-09-23 22:34:27.663060735 -0700
+++ gmock-1.7.0/Makefile.in	2013-09-23 22:42:12.904074641 -0700
@@ -1346,11 +1346,9 @@
 
 install-exec-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Mock into your build system."
-	false
 
 install-data-local:
 	echo "'make install' is dangerous and not supported. Instead, see README for how to integrate Google Mock into your build system."
-	false
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
