From c7d5ff81313acddf582faa5f7c576ebce95a4076 Mon Sep 17 00:00:00 2001
From: Caleb Fenton <calebjfenton@gmail.com>
Date: Mon, 3 Jun 2019 11:17:49 -0700
Subject: [PATCH] Properly set DEX string size

---
 libyara/modules/dex.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libyara/modules/dex.c b/libyara/modules/dex.c
index f28f7986..1e8a6af4 100644
--- a/libyara/modules/dex.c
+++ b/libyara/modules/dex.c
@@ -775,8 +775,8 @@ void dex_parse(
         "string_ids[%i].offset", i);
 
     set_integer(
-        yr_le32toh(string_id_item->string_data_offset), dex->object,
-        "string_ids[%i].size", value);
+        value, dex->object,
+        "string_ids[%i].size", i);
 
     set_sized_string(
         (const char*) ((dex->data + yr_le32toh(string_id_item->string_data_offset) + 1)),
