--- src/mstring.cpp	2010-02-01 12:02:19.972249219 +0100
+++ src/mstring.cpp	2010-02-01 12:03:33.158999955 +0100
@@ -546,7 +546,7 @@
 	cout << "begin + to_st.leng : " << (void*) (begin + to_st.leng) << endl;
 	cout << "begin + to_st.leng : " << (begin + to_st.leng) << endl;
 */	
-	int mv = leng - (int)begin + (int)stripped - to_st.leng + 1;
+	unsigned long mv = leng - (unsigned long)begin + (unsigned long)stripped - to_st.leng + 1;
 //	cout << "mv : " << mv << endl;
 	memmove ( begin, begin + to_st.leng , mv );
 	begin = strstr ( stripped, to_st.sp );
