--- configure.orig	2002-04-15 10:41:43.000000000 -0400
+++ configure	2013-06-06 17:33:25.340670726 -0400
@@ -2182,7 +2182,7 @@
 #include <stddef.h>
 #endif
 EOF
-if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
+if (eval "$ac_cpp -E conftest.$ac_ext") 3>&5 |
   egrep "(^|[^a-zA-Z_0-9])off_t[^a-zA-Z_0-9]" >/dev/null 2>&1; then
   rm -rf conftest*
   ac_cv_type_off_t=yes
@@ -2215,7 +2215,7 @@
 #include <stddef.h>
 #endif
 EOF
-if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
+if (eval "$ac_cpp -E conftest.$ac_ext") 2>&5 |
   egrep "(^|[^a-zA-Z_0-9])size_t[^a-zA-Z_0-9]" >/dev/null 2>&1; then
   rm -rf conftest*
   ac_cv_type_size_t=yes
