diff -uNr foremost-1.4.orig/config.c foremost-1.4/config.c
--- foremost-1.4.orig/config.c	2006-08-31 17:50:47.000000000 +0100
+++ foremost-1.4/config.c	2006-08-31 17:51:09.000000000 +0100
@@ -288,7 +288,7 @@
 #ifdef __WIN32
 		set_config_file(s, "/Program Files/foremost/foremost.conf");
 #else
-		set_config_file(s, "/usr/local/etc/foremost.conf");
+		set_config_file(s, "/etc/foremost.conf");
 #endif
 		if ((f = fopen(get_config_file(s), "r")) == NULL)
 			{
