From ba77d861f2ec15ba74f5bdeaf2d0941995653454 Mon Sep 17 00:00:00 2001
From: pad <yoann.padioleau@gmail.com>
Date: Thu, 11 Aug 2016 13:01:33 -0700
Subject: [PATCH] * lang_php/matcher/Makefile: remove -warn-error should fix
 issue #147

---
 lang_php/matcher/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lang_php/matcher/Makefile b/lang_php/matcher/Makefile
index 056b4fb..1b1375e 100644
--- a/lang_php/matcher/Makefile
+++ b/lang_php/matcher/Makefile
@@ -31,7 +31,7 @@ INCLUDEDIRS= $(TOP)/commons $(TOP)/commons/ocollection \
 # Generic variables
 ##############################################################################
 #BUG in ocamlc, false positive on unused value
-WARNING_FLAGS=-w +A-4-29-6-45-32 -warn-error +a
+#WARNING_FLAGS=-w +A-4-29-6-45-32
 -include $(TOP)/Makefile.common
 
 ##############################################################################
-- 
2.9.2

