diff -urb pfff-0.29.sic/lang_bytecode/parsing/test_parsing_bytecode.ml pfff-0.29/lang_bytecode/parsing/test_parsing_bytecode.ml
--- pfff-0.29.sic/lang_bytecode/parsing/test_parsing_bytecode.ml	2015-03-03 19:52:16.000000000 +0100
+++ pfff-0.29/lang_bytecode/parsing/test_parsing_bytecode.ml	2016-11-15 17:20:37.833563042 +0100
@@ -11,8 +11,6 @@
  *)
 open Common
 
-module PI = Parse_info
-
 (*****************************************************************************)
 (* Subsystem testing *)
 (*****************************************************************************)
