--- src/social-engineer-toolkit-7.3.13/setoolkit	2016-09-15 13:33:14.446666667 -0400
+++ setoolkit	2016-09-15 15:09:57.043333331 -0400
@@ -191,10 +191,10 @@
     while True:
         core.show_banner(define_version, '1')
         show_main_menu = core.create_menu(text.main_text, text.main_menu)
-    
-            # special case of list item 99
-            print('\n  99) Exit the Social-Engineer Toolkit\n')
-    
+
+        # special case of list item 99
+        print('\n  99) Exit the Social-Engineer Toolkit\n')
+
         # main core menu
         main_menu_choice = (input(core.setprompt("0", "")))
 
