--- src/semgrep-0.32.0/setup.py	2020-11-19 06:26:52.000000000 +0100
+++ setup.py	2020-11-20 00:49:52.323333207 +0100
@@ -142,15 +142,15 @@
     long_description_content_type="text/markdown",
     url="https://github.com/returntocorp/semgrep",
     install_requires=[
-        "attrs>=19.3.0",
-        "colorama>=0.4.3",
-        "junit_xml==1.9",
-        "requests>=2.22.0",
+        "attrs",
+        "colorama",
+        "junit_xml",
+        "requests",
         # exact version because of unstable API
-        "ruamel.yaml==0.16.10",
-        "tqdm>=4.46.1",
-        "packaging>=20.4",
-        "jsonschema~=3.2.0",
+        "ruamel.yaml",
+        "tqdm",
+        "packaging",
+        "jsonschema",
     ],
     entry_points={"console_scripts": ["semgrep=semgrep.__main__:main"]},
     packages=setuptools.find_packages(),
