--- src/semgrep-0.29.0/setup.py	2020-08-05 19:21:56.000000000 +0200
+++ setup.py	2020-08-09 13:21:20.323331341 +0200
@@ -121,7 +121,7 @@
     author="Return To Corporation",
     author_email="support@r2c.dev",
     description="Lightweight static analysis for many languages. Find bug variants with patterns that look like source code.",
-    cmdclass={"install": PostInstallCommand, "bdist_wheel": bdist_wheel},
+    #cmdclass={"install": PostInstallCommand, "bdist_wheel": bdist_wheel},
     long_description=long_description,
     long_description_content_type="text/markdown",
     url="https://github.com/returntocorp/semgrep",
