--- src/sslyze-2.1.3/setup.py	2019-06-13 01:06:45.000000000 +0200
+++ setup.py	2019-06-22 18:07:40.116666240 +0200
@@ -79,10 +79,10 @@

     # Dependencies
     install_requires=[
-        'nassl>=2.2.0,<2.3.0',
-        'cryptography==2.5',
-        'tls-parser>=1.2.0,<1.3.0',
-        "dataclasses ; python_version<'3.7'",  # To remove when we drop support for Python 3.6
+        'nassl',
+        'cryptography',
+        'tls-parser',
+        "dataclasses ; python_version<'3.7'",  # To remove when we drop support for Python 3.6
     ],

     # cx_freeze info for Windows builds with Python embedded
