--- a/include/firewalk.h	2025-09-30 14:45:07.758089641 +0200
+++ b/include/firewalk.h	2025-09-30 14:47:11.809796085 +0200
@@ -248,7 +248,7 @@
 int                                 /* 1 on success -1 or failure */
 catch_sig(
     int,                            /* signal to catch */
-    void (*)()                      /* new signal handler */
+    void (*)(int)                   /* new signal handler */
     );
 
 /* handles SIGINT from user */
--- a/src/signal.c	2025-09-30 14:50:59.333580062 +0200
+++ b/src/signal.c	2025-09-30 14:50:15.667074006 +0200
@@ -40,7 +40,7 @@
 extern int loop;
 
 int
-catch_sig(int signo, void (*handler)())
+catch_sig(int signo, void (*handler)(int))
 {
     struct sigaction action;
 
