--- src/morpheus/settings	2017-01-20 17:02:54.456666669 -0500
+++ settings	2017-01-20 17:05:47.843333331 -0500
@@ -15,9 +15,9 @@
 
 ### prompt displays (morpheus) available
 # chose your own 'personalized prompt'.
-# -▷   -▶   '>'  ☭   ✿   ☯   ☠   ☆   ➽  ♛  ⊶ 
+# -▷   -▶   '>'  ☭   ✿   ☯   ☠   ☆   ➽  ♛  ⊶
 ------------------
-PROMPT_DISPLAY=--▷ 
+PROMPT_DISPLAY=>
 ------------------
 
 
@@ -37,7 +37,7 @@
 ### ENABLE IPv6 ARP POISON? (etter.conf)
 # by default morpheus lunch arp poison againt IPv6 protocol
 # IF your facing a "token (///) Error" then change the next value
-# to "DISABLED" to force morpheus framework to run arp againts IPv4. 
+# to "DISABLED" to force morpheus framework to run arp againts IPv4.
 # accepted values are: ACTIVE or DISABLED
 ----------------
 USE_IPV6=ACTIVE
@@ -49,10 +49,10 @@
 ### REDIRECT DOMAINS (etter.dns)
 # by default morpheus will redirect all .com prefixed
 # domains and another one (user choice) the next field
-# allow you to config the 2º domain name to be redirected. 
-# accepted values are: pt | org | ir | etc..   
+# allow you to config the 2º domain name to be redirected.
+# accepted values are: pt | org | ir | etc..
 ---------
-PREFIX=pt
+PREFIX=org
 ---------
 
 
@@ -84,9 +84,9 @@
 # just change the next values to point
 # to your dsniff packet install folders.
 ----------------------------
-TCP_KILL=/usr/sbin/tcpkill
-URL_SNARF=/usr/sbin/urlsnarf
-MSG_SNARF=/usr/sbin/msgsnarf
+TCP_KILL=/usr/bin/tcpkill
+URL_SNARF=/usr/bin/urlsnarf
+MSG_SNARF=/usr/bin/msgsnarf
 DRI_NET=/usr/bin/driftnet
 ----------------------------
 
