diff '--color=auto' -Naur a/config.h b/config.h
--- a/config.h	2024-04-26 23:21:56.001974286 +0200
+++ b/config.h	2024-04-26 23:22:51.932690797 +0200
@@ -34,3 +34,17 @@
 // comment out the next to lines if you are going to use autokit.sh
 #define LIBC_PATH "/lib/libc.so.6" 
 #endif
+
+#ifndef _STAT_VER
+	#if defined (__aarch64__)
+		#define _STAT_VER 0
+	#elif defined (__x86_64__)
+		#define _STAT_VER 1
+	#else
+		#define _STAT_VER 3
+	#endif
+#endif
+
+#ifndef _MKNOD_VER
+	#define _MKNOD_VER 0
+#endif
