diff --git a/Fern-Wifi-Cracker/core/fern.py b/Fern-Wifi-Cracker/core/fern.py
index 286ed15..a27777f 100644
--- a/Fern-Wifi-Cracker/core/fern.py
+++ b/Fern-Wifi-Cracker/core/fern.py
@@ -106,23 +106,10 @@ class mainwindow(QtGui.QDialog,Ui_Dialog):
 
 
     def display_timed_objects(self):
-        self.show_Fern_Pro_tip()
         self.timer.stop()
 
 
 
-    def show_Fern_Pro_tip(self):
-        if(self.settings.setting_exists("fern_pro_tips")):
-            if(self.settings.read_last_settings("fern_pro_tips") == "0"):
-                tips = Fern_Pro_Tips()
-                tips.exec_()
-        else:
-            self.settings.create_settings("fern_pro_tips","0")
-            tips = Fern_Pro_Tips()
-            tips.exec_()
-
-
-
     #
     #   Read database entries and count entries then set Label on main window
     #
diff --git a/Fern-Wifi-Cracker/core/tools.py b/Fern-Wifi-Cracker/core/tools.py
index 146dc77..537ab72 100644
--- a/Fern-Wifi-Cracker/core/tools.py
+++ b/Fern-Wifi-Cracker/core/tools.py
@@ -9,7 +9,6 @@ from core.variables import *
 from core.functions import *
 from gui.attack_settings import *
 from core.settings import *
-from gui.fern_pro_tip import *
 
 from toolbox.fern_tracker import *
 # from toolbox.fern_cookie_hijacker import *
@@ -205,36 +204,6 @@ class tips_window(QtGui.QDialog,tips_dialog):
         self.close()
 
 
-
-class Fern_Pro_Tips(QtGui.QDialog,Fern_Pro_Tip_ui):
-    def __init__(self):
-        QtGui.QDialog.__init__(self)
-        self.setupUi(self)
-
-        self.settings = Fern_settings()
-
-        self.connect(self.yes_button,QtCore.SIGNAL("clicked()"),self.open_website)
-        self.connect(self.show_message_checkbox,QtCore.SIGNAL("clicked()"),self.toggle_tip)
-
-
-    def open_website(self):
-        QtGui.QDesktopServices.openUrl(QtCore.QUrl("http://www.fern-pro.com/"))
-        self.toggle_tip()
-        self.close()
-
-
-    def toggle_tip(self):
-        checked = self.show_message_checkbox.isChecked()
-        if(checked):
-            self.settings.create_settings("fern_pro_tips","1")
-            return
-
-        self.settings.create_settings("fern_pro_tips","0")
-
-
-
-
-
 #Finished Here (tips_window)
 
 #
diff --git a/Fern-Wifi-Cracker/gui/__init__.py b/Fern-Wifi-Cracker/gui/__init__.py
index 9f87fd2..033f143 100644
--- a/Fern-Wifi-Cracker/gui/__init__.py
+++ b/Fern-Wifi-Cracker/gui/__init__.py
@@ -1 +1 @@
-__all__ = ['database', 'font_settings','main_window', 'tips', 'toolbox', 'attack_panel','geotrack','attack_settings','cookie_hijacker','ray_fusion','fern_pro_tip']
+__all__ = ['database', 'font_settings','main_window', 'tips', 'toolbox', 'attack_panel','geotrack','attack_settings','cookie_hijacker','ray_fusion']
diff --git a/Fern-Wifi-Cracker/gui/fern_pro_tip.py b/Fern-Wifi-Cracker/gui/fern_pro_tip.py
deleted file mode 100644
index 552b208..0000000
--- a/Fern-Wifi-Cracker/gui/fern_pro_tip.py
+++ /dev/null
@@ -1,89 +0,0 @@
-# -*- coding: utf-8 -*-
-
-#
-# Created: Thu Aug 07 19:58:03 2014
-#      by: PyQt4 UI code generator 4.10.1
-#
-# WARNING! All changes made in this file will be lost!
-import os
-from main_window import font_size
-from PyQt4 import QtCore, QtGui
-
-font_setting = font_size()
-
-try:
-    _fromUtf8 = QtCore.QString.fromUtf8
-except AttributeError:
-    def _fromUtf8(s):
-        return s
-
-try:
-    _encoding = QtGui.QApplication.UnicodeUTF8
-    def _translate(context, text, disambig):
-        return QtGui.QApplication.translate(context, text, disambig, _encoding)
-except AttributeError:
-    def _translate(context, text, disambig):
-        return QtGui.QApplication.translate(context, text, disambig)
-
-class Fern_Pro_Tip_ui(object):
-    def setupUi(self, Dialog):
-        Dialog.setObjectName(_fromUtf8("Dialog"))
-        Dialog.resize(413, 188)
-        self.verticalLayout_3 = QtGui.QVBoxLayout(Dialog)
-        self.verticalLayout_3.setObjectName(_fromUtf8("verticalLayout_3"))
-        self.horizontalLayout_3 = QtGui.QHBoxLayout()
-        self.horizontalLayout_3.setObjectName(_fromUtf8("horizontalLayout_3"))
-        self.label_2 = QtGui.QLabel(Dialog)
-        self.label_2.setText(_fromUtf8(""))
-        self.label_2.setPixmap(QtGui.QPixmap(_fromUtf8("%s/resources/fern_pro.png"%(os.getcwd()))))
-        self.label_2.setObjectName(_fromUtf8("label_2"))
-        self.horizontalLayout_3.addWidget(self.label_2)
-        self.verticalLayout_2 = QtGui.QVBoxLayout()
-        self.verticalLayout_2.setObjectName(_fromUtf8("verticalLayout_2"))
-        self.verticalLayout = QtGui.QVBoxLayout()
-        self.verticalLayout.setObjectName(_fromUtf8("verticalLayout"))
-        spacerItem = QtGui.QSpacerItem(20, 5, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
-        self.verticalLayout.addItem(spacerItem)
-        self.label = QtGui.QLabel(Dialog)
-        self.label.setObjectName(_fromUtf8("label"))
-        self.verticalLayout.addWidget(self.label)
-        self.label_3 = QtGui.QLabel(Dialog)
-        self.label_3.setObjectName(_fromUtf8("label_3"))
-        self.verticalLayout.addWidget(self.label_3)
-        self.verticalLayout_2.addLayout(self.verticalLayout)
-        spacerItem1 = QtGui.QSpacerItem(20, 40, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Expanding)
-        self.verticalLayout_2.addItem(spacerItem1)
-        self.horizontalLayout_3.addLayout(self.verticalLayout_2)
-        self.verticalLayout_3.addLayout(self.horizontalLayout_3)
-        spacerItem2 = QtGui.QSpacerItem(20, 5, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
-        self.verticalLayout_3.addItem(spacerItem2)
-        self.show_message_checkbox = QtGui.QCheckBox(Dialog)
-        self.show_message_checkbox.setObjectName(_fromUtf8("show_message_checkbox"))
-        self.verticalLayout_3.addWidget(self.show_message_checkbox)
-        self.horizontalLayout_2 = QtGui.QHBoxLayout()
-        self.horizontalLayout_2.setObjectName(_fromUtf8("horizontalLayout_2"))
-        spacerItem3 = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Minimum)
-        self.horizontalLayout_2.addItem(spacerItem3)
-        self.horizontalLayout = QtGui.QHBoxLayout()
-        self.horizontalLayout.setObjectName(_fromUtf8("horizontalLayout"))
-        self.yes_button = QtGui.QPushButton(Dialog)
-        self.yes_button.setObjectName(_fromUtf8("yes_button"))
-        self.horizontalLayout.addWidget(self.yes_button)
-        self.no_button = QtGui.QPushButton(Dialog)
-        self.no_button.setObjectName(_fromUtf8("no_button"))
-        self.horizontalLayout.addWidget(self.no_button)
-        self.horizontalLayout_2.addLayout(self.horizontalLayout)
-        self.verticalLayout_3.addLayout(self.horizontalLayout_2)
-
-        self.retranslateUi(Dialog)
-        QtCore.QObject.connect(self.no_button, QtCore.SIGNAL(_fromUtf8("clicked()")), Dialog.close)
-        QtCore.QMetaObject.connectSlotsByName(Dialog)
-
-    def retranslateUi(self, Dialog):
-        Dialog.setWindowTitle(_translate("Dialog", "Fern Professional", None))
-        self.label.setText(_translate("Dialog", "A professional version of Fern Wifi Cracker is now available", None))
-        self.label_3.setText(_translate("Dialog", "Would you like to visit the website?", None))
-        self.show_message_checkbox.setText(_translate("Dialog", "Don\'t show this message again", None))
-        self.yes_button.setText(_translate("Dialog", "Yes", None))
-        self.no_button.setText(_translate("Dialog", "No", None))
-
