 buildscripts/moduleconfig.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/buildscripts/moduleconfig.py b/buildscripts/moduleconfig.py
index b4d0bba0490..03541fab940 100644
--- a/buildscripts/moduleconfig.py
+++ b/buildscripts/moduleconfig.py
@@ -27,7 +27,6 @@ MongoDB SConscript files do.
 __all__ = ('discover_modules', 'discover_module_directories', 'configure_modules',
            'register_module_test')  # pylint: disable=undefined-all-variable
 
-import imp
 import inspect
 import os
 
@@ -71,8 +70,6 @@ def discover_modules(module_root, allowed_modules):
             print("adding module: %s" % (name))
             fp = open(build_py, "r")
             try:
-                module = imp.load_module("module_" + name, fp, build_py,
-                                         (".py", "r", imp.PY_SOURCE))
                 if getattr(module, "name", None) is None:
                     module.name = name
                 found_modules.append(module)
