diff --git a/buildscripts/scons.py b/buildscripts/scons.py
index 534fca32..c38f64df 100755
--- a/buildscripts/scons.py
+++ b/buildscripts/scons.py
@@ -19,13 +19,13 @@ SITE_TOOLS_DIR = os.path.join(MONGODB_ROOT, 'site_scons')
 sys.path = [SCONS_DIR, SITE_TOOLS_DIR] + sys.path
 
 # pylint: disable=C0413
-from mongo.pip_requirements import verify_requirements, MissingRequirements
+#from mongo.pip_requirements import verify_requirements, MissingRequirements
 
-try:
-    verify_requirements('etc/pip/compile-requirements.txt')
-except MissingRequirements as ex:
-    print(ex)
-    sys.exit(1)
+#try:
+#    verify_requirements('etc/pip/compile-requirements.txt')
+#except MissingRequirements as ex:
+#    print(ex)
+#    sys.exit(1)
 
 try:
     import SCons.Script
