diff --git a/SConstruct b/SConstruct
index 89c044ab..0484e475 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3511,17 +3511,11 @@ def doConfigure(myenv):
             "BOOST_LOG_NO_SHORTHAND_NAMES",
             "BOOST_LOG_USE_NATIVE_SYSLOG",
             "BOOST_LOG_WITHOUT_THREAD_ATTR",
+            "BOOST_LOG_DYN_LINK",
             "ABSL_FORCE_ALIGNED_ACCESS",
         ]
     )
 
-    if link_model.startswith("dynamic") and not link_model == 'dynamic-sdk':
-        conf.env.AppendUnique(
-            CPPDEFINES=[
-                "BOOST_LOG_DYN_LINK",
-            ]
-    )
-
     if use_system_version_of_library("boost"):
         if not conf.CheckCXXHeader( "boost/filesystem/operations.hpp" ):
             myenv.ConfError("can't find boost headers")
