diff --git a/src/mongo/db/repl/tenant_file_cloner.h b/src/mongo/db/repl/tenant_file_cloner.h
index def0952..0625e46 100644
--- a/src/mongo/db/repl/tenant_file_cloner.h
+++ b/src/mongo/db/repl/tenant_file_cloner.h
@@ -32,6 +32,7 @@
 #include <boost/filesystem.hpp>
 #include <memory>
 #include <vector>
+#include <fstream>
 
 #include "mongo/db/repl/base_cloner.h"
 #include "mongo/db/repl/task_runner.h"
