diff -ruN ncpfs-2.2.6/contrib/pam/pam_ncp_auth.c ncpfs-2.2.6-fixed/contrib/pam/pam_ncp_auth.c
--- ncpfs-2.2.6/contrib/pam/pam_ncp_auth.c	2005-01-27 18:35:59.000000000 +0100
+++ ncpfs-2.2.6-fixed/contrib/pam/pam_ncp_auth.c	2007-05-11 21:38:05.143474750 +0200
@@ -257,6 +257,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <syslog.h>
 #include <unistd.h>
 #include <pwd.h>
 #include <grp.h>
diff -ruN ncpfs-2.2.6/sutil/ncpm_common.c ncpfs-2.2.6-fixed/sutil/ncpm_common.c
--- ncpfs-2.2.6/sutil/ncpm_common.c	2005-01-27 18:35:59.000000000 +0100
+++ ncpfs-2.2.6-fixed/sutil/ncpm_common.c	2007-05-11 21:38:29.609003750 +0200
@@ -82,6 +82,7 @@
 
 #include "ncpm_common.h"
 
+#include <stddef.h>
 #include <stdio.h>
 #include <string.h>
 #include <signal.h>
