--- a/Libnet-1.0.2a/include/libnet.h	2001-02-07 04:49:25.000000000 +0530
+++ b/Libnet-1.0.2a/include/libnet.h	2007-06-24 16:54:09.000000000 +0530
@@ -84,9 +84,7 @@
 #define LIBNET_VERSION  "1.0.2a"
 
 #if (!LIBNET_LIL_ENDIAN && !LIBNET_BIG_ENDIAN)
-#error "byte order has not been specified, you'll
-need to #define either LIBNET_LIL_ENDIAN or LIBNET_BIG_ENDIAN.  See the
-documentation regarding the libnet-config script."
+#error "byte order has not been specified, you'll need to #define either LIBNET_LIL_ENDIAN or LIBNET_BIG_ENDIAN.  See the documentation regarding the libnet-config script."
 #endif
 
 #endif  /* __LIBNET_H */
