--- src/commix/src/utils/settings.py	2015-04-29 12:11:20.274018574 +0200
+++ settings.py	2015-04-29 12:12:26.766991684 +0200
@@ -10,7 +10,7 @@
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.
- 
+
  For more see the file 'doc/COPYING' for copying permission.
 """
 
@@ -18,6 +18,8 @@
 import os
 import urllib
 
+from os.path import expanduser
+
 """
  The global variables.
 """
@@ -28,7 +30,7 @@
 AUTHOR  = "Anastasios Stasinopoulos"
 VERSION = "v0.1b"
 YEAR    = "2015"
-TWITTER = "@ancst" 
+TWITTER = "@ancst"
 
 # Inject Tag
 INJECT_TAG = "INJECT_HERE"
@@ -39,14 +41,14 @@
     COMMIT_ID = "-" + f.readline()[0:7]
 else:
     COMMIT_ID = ""
-    
+
 # Output Directory
-OUTPUT_DIR = ".output/"
+OUTPUT_DIR = expanduser('~') + '/commix-output/'
 dir = os.path.dirname(OUTPUT_DIR)
 try:
     os.stat(OUTPUT_DIR)
 except:
-    os.mkdir(OUTPUT_DIR)       
+    os.mkdir(OUTPUT_DIR)
 
 # The base64 decode trick
 B64_DEC_TRICK = " | base64 -d "
@@ -55,7 +57,7 @@
 SEPARATORS = ["",";","&","|","||","&&","%0a","%26","%26%26","%7C","%7C%7C"]
 
 # The command injection prefixes.
-PREFIXES = ["","'",")","')","|","&","%0a","%27","%29","%27%29","%7C","%26"] 
+PREFIXES = ["","'",")","')","|","&","%0a","%27","%29","%27%29","%7C","%26"]
 
 # The command injection suffixes.
 SUFFIXES = ["","'","#","//","\\\\","&","|","%27","%5C%5C","%27%29","%26","%7C"]
@@ -97,4 +99,4 @@
 
 # 'Yes' or enter inputs
 # Accepts YES/Y/yes/y or "enter"
-CHOISE_YES = set(['yes','y', 'ye', ''])
\ No newline at end of file
+CHOISE_YES = set(['yes','y', 'ye', ''])
