diff -ur warvox/src/ruby-kissfft/main.c warvox-build//src/ruby-kissfft/main.c
--- warvox/src/ruby-kissfft/main.c	2010-06-29 17:38:35.182817155 +0200
+++ warvox/src/ruby-kissfft/main.c	2010-06-29 17:51:35.957329208 +0200
@@ -78,11 +78,11 @@
 		return Qnil;
 	}
 
-	if(RARRAY(r_data)->len == 0) {
+	if(RARRAY_LEN(r_data) == 0) {
 		return Qnil;
 	}
 
-	if(TYPE(RARRAY(r_data)->ptr[0]) != T_FIXNUM ) {
+	if(TYPE(RARRAY_PTR(r_data)) != T_FIXNUM ) {
 		return Qnil;
 	}	
 
@@ -95,7 +95,7 @@
 
 	memset(mag2buf,0,sizeof(mag2buf)*nfreqs);
 
-	inp_len = RARRAY(r_data)->len;
+	inp_len = RARRAY_LEN(r_data);
 	inp_idx = 0;
 
 	while(inp_idx < inp_len) {
@@ -105,10 +105,10 @@
 			if(inp_idx + i >= inp_len) {
 				tbuf[i] = 0;
 			} else {
-				if(TYPE(RARRAY(r_data)->ptr[ inp_idx + i ]) != T_FIXNUM) {
+				if(TYPE(RARRAY_PTR(r_data)[ inp_idx + i ]) != T_FIXNUM) {
 					tbuf[i] = 0;
 				} else {
-					tbuf[i] = NUM2INT( RARRAY(r_data)->ptr[ inp_idx + i ] );
+					tbuf[i] = NUM2INT( RARRAY_PTR(r_data)[ inp_idx + i ] );
 				}
 			}
 		}
