diff --git a/cymothoa-1-beta/cymothoa.c b/cymothoa-1-beta.arm/cymothoa.c
index b387d49..d5f4657 100644
--- a/cymothoa-1-beta/cymothoa.c
+++ b/cymothoa-1-beta.arm/cymothoa.c
@@ -160,7 +160,7 @@ int ptrace_infect()
 
         // other variables
         int ptr=0, error=0;
-        struct user_regs_struct reg;
+        struct user_regs reg;
 
         printf("[+] attaching to process %d\n",pid);
 
diff --git a/cymothoa-1-beta/cymothoa.h b/cymothoa-1-beta.arm/cymothoa.h
index b1b1583..6645354 100644
--- a/cymothoa-1-beta/cymothoa.h
+++ b/cymothoa-1-beta.arm/cymothoa.h
@@ -171,3 +171,16 @@ uint32_t persistent_addr;
 
 #endif
 
+// ARM
+#ifdef __arm__
+
+// stack push size
+#define BLOCK_SIZE 4
+
+// registers
+#define AX uregs[0]
+#define BX uregs[1]
+#define INST_POINTER uregs[12]
+#define STACK_POINTER uregs[13]
+
+#endif
