diff --git a/cymothoa-1-beta.old/cymothoa.c b/cymothoa-1-beta/cymothoa.c
index d5f4657..5d3e520 100644
--- a/cymothoa-1-beta.old/cymothoa.c
+++ b/cymothoa-1-beta/cymothoa.c
@@ -176,10 +176,10 @@ int ptrace_infect()
 
         printf("\n register info: \n");
         printf(" -----------------------------------------------------------\n");
-        printf(" eax value: 0x%lx\t", reg.AX);
-        printf(" ebx value: 0x%lx\n", reg.BX);
-        printf(" esp value: 0x%lx\t", reg.STACK_POINTER);
-        printf(" eip value: 0x%lx\n", reg.INST_POINTER);
+        printf(" r0 value: 0x%lx\t", reg.R0);
+        printf(" r1 value: 0x%lx\n", reg.R1);
+        printf(" sp value: 0x%lx\t", reg.STACK_POINTER);
+        printf(" ip value: 0x%lx\n", reg.INST_POINTER);
         printf(" ------------------------------------------------------------\n\n");
 
         reg.STACK_POINTER -= BLOCK_SIZE; // decrement STACK_POINTER
diff --git a/cymothoa-1-beta.old/cymothoa.h b/cymothoa-1-beta/cymothoa.h
index 6645354..8bf8fc0 100644
--- a/cymothoa-1-beta.old/cymothoa.h
+++ b/cymothoa-1-beta/cymothoa.h
@@ -178,8 +178,8 @@ uint32_t persistent_addr;
 #define BLOCK_SIZE 4
 
 // registers
-#define AX uregs[0]
-#define BX uregs[1]
+#define R0 uregs[0]
+#define R1 uregs[1]
 #define INST_POINTER uregs[12]
 #define STACK_POINTER uregs[13]
 
