diff --git a/cymothoa-1-beta/cymothoa.c b/cymothoa-1-beta.warn/cymothoa.c
index 5d3e520..7c35432 100644
--- a/cymothoa-1-beta/cymothoa.c
+++ b/cymothoa-1-beta.warn/cymothoa.c
@@ -190,13 +190,13 @@ int ptrace_infect()
 
         // print preamble options
         if (need_fork)
-            printf("[+] payload preamble: fork\n", reg.STACK_POINTER);
+            printf("[+] payload preamble: fork 0x%lx\n", reg.STACK_POINTER);
         else if (need_thread)
-            printf("[+] payload preamble: thread\n", reg.STACK_POINTER);
+            printf("[+] payload preamble: thread 0x%lx\n", reg.STACK_POINTER);
         else if (need_alarm)
-            printf("[+] payload preamble: alarm\n", reg.STACK_POINTER);
+            printf("[+] payload preamble: alarm 0x%lx\n", reg.STACK_POINTER);
         else if (need_setitimer || use_setitimer)
-            printf("[+] payload preamble: setitimer\n", reg.STACK_POINTER);
+            printf("[+] payload preamble: setitimer 0x%lx\n", reg.STACK_POINTER);
 
         // get the address for our shellcode
         ptr = search_lib_region(pid, args.lib_name, MEM_EXEC);
diff --git a/cymothoa-1-beta/cymothoa.h b/cymothoa-1-beta.warn/cymothoa.h
index 8bf8fc0..b5f0bbc 100644
--- a/cymothoa-1-beta/cymothoa.h
+++ b/cymothoa-1-beta.warn/cymothoa.h
@@ -11,6 +11,8 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 
+pid_t waitpid(pid_t pid, int *stat_loc, int options);
+
 
 char banner[] =
 "                              _                 \n\
