--- IDEA.xs	2013-07-21 13:03:47.137634587 -0400
+++ IDEA.xs	2013-07-21 13:03:44.764301152 -0400
@@ -71,8 +71,8 @@
 			output = sv_newmortal();
 		output_len = 8;
 
-		if (!SvUPGRADE(output, SVt_PV))
-			croak("cannot use output argument as lvalue");
+		/*if (!SvUPGRADE(output, SVt_PV))*/
+			/*croak("cannot use output argument as lvalue");*/
 
 		idea_crypt((u_int16_t *)input, (u_int16_t *)SvGROW(output, output_len), (u_int16_t *)ks);
