--- src/avet/setup.sh	2024-11-24 01:55:02.130000078 +0100
+++ setup.sh	2024-11-24 01:55:55.896666770 +0100
@@ -4,23 +4,11 @@
 
 echo "+++ Preparing AVET for use..."
 
-echo "+++ Installing wine and wine32"
-# add missing sources for wine32
-sudo dpkg --add-architecture i386
-sudo apt update
-sudo apt install wine -y
-sudo apt install wine32 -y
-
-# Install mingw crosscompiler
-sudo apt install mingw-w64 -y
-sudo apt install jq -y
-
 echo "+++ Downloading and installing tdm-gcc via wine"
 wget https://github.com/jmeubank/tdm-gcc/releases/download/v9.2.0-tdm64-1/tdm64-gcc-9.2.0.exe
 wine tdm64-gcc-9.2.0.exe
 rm tdm64-gcc-9.2.0.exe
 
-
 while true; do
     read -p "Do you want to download the dependencies AVET needs? [y/n]" yn
     case $yn in
