--- src/commix/src/utils/settings.py	2015-03-31 12:20:22.656168011 +0200
+++ settings.py	2015-03-31 12:21:08.129254995 +0200
@@ -10,13 +10,15 @@
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.
- 
+
  For more see the file 'doc/COPYING' for copying permission.
 """
 
 import sys
 import os
 
+from os.path import expanduser
+
 """
  The global variables.
 """
@@ -27,7 +29,7 @@
 AUTHOR  = "Anastasios Stasinopoulos"
 VERSION = "v0.01b"
 YEAR    = "2015"
-TWITTER = "@ancst" 
+TWITTER = "@ancst"
 
 # Inject Tag
 INJECT_TAG = "INJECT_HERE"
@@ -38,14 +40,14 @@
     COMMIT_ID = "-" + f.readline()[0:7]
 else:
     COMMIT_ID = ""
-    
+
 # Output Directory
-OUTPUT_DIR = ".output/"
+OUTPUT_DIR = expanduser('~') + '/commix-output/'
 dir = os.path.dirname(OUTPUT_DIR)
 try:
     os.stat(OUTPUT_DIR)
 except:
-    os.mkdir(OUTPUT_DIR)       
+    os.mkdir(OUTPUT_DIR)
 
 # The base64 decode trick
 B64_DEC_TRICK = " | base64 -d "
