diff -ru attackctrl.py attackctrl.py
--- attackctrl.py       2013-09-04 20:42:01.699888418 +0300
+++ attackctrl.py       2013-09-04 20:57:08.283193747 +0300
@@ -217,7 +217,7 @@
     temp = ''
     gw.append(temp)
     for interface in result:
-       f = os.popen("ifconfig " + interface + " | grep \"inet addr\" | sed -e \'s/.*addr://;s/ .*//\'")
+       f = os.popen("ifconfig " + interface + " | grep \"inet \" | awk \'{print $3}\'")
        temp2 = ''
        temp3 = ''
        temp = f.readline().rstrip('\n')
@@ -237,7 +237,7 @@
     conf = f.readlines()
          
         #Get the Local IP Address
-    f = os.popen("ifconfig " + result[0] + " | grep \"inet addr\" | sed -e \'s/.*addr://;s/ .*//\'")
+    f = os.popen("ifconfig " + result[0] + " | grep \"inet \" | awk \'{print $3}\'")
     temp2 = ''
     temp3 = ''
     temp = f.readline().rstrip('\n')

diff -ru main/views.py main/views.py
--- main/views.py       2013-03-17 05:07:04.000000000 +0200
+++ main/views.py       2013-09-04 20:59:04.733190150 +0300
@@ -292,7 +292,7 @@
 
          #Get the Local IP Address
       try:
-         f = os.popen("ifconfig " + request.POST["iface"] + " | grep \"inet addr\" | sed -e \'s/.*addr://;s/ .*//\'")
+         f = os.popen("ifconfig " + request.POST["iface"] + " | grep \"inet \" | awk \'{print $3}\'")
          temp2 = ''
          temp3 = ''
          temp = f.readline().rstrip('\n')
@@ -445,7 +445,7 @@
          temp = ''
          gw.append(temp)
          for interface in result:
-           f = os.popen("ifconfig " + interface + " | grep \"inet addr\" | sed -e \'s/.*addr://;s/ .*//\'")
+           f = os.popen("ifconfig " + interface + " | grep \"inet \" | awk \'{print $3}\'")
            temp2 = ''
            temp3 = ''
            try:
@@ -522,7 +522,7 @@
 def gate(request):
    if request.is_ajax():
       print "Loading Default Gateway"
-      f = os.popen("ifconfig " + interface + " | grep \"inet addr\" | sed -e \'s/.*addr://;s/ .*//\'")
+      f = os.popen("ifconfig " + interface + " | grep \"inet \" | awk \'{print $3}\'")
       temp = ''
       temp2 = ''
       temp3 = ''
