--- src/plasma-disasm/plasma/lib/arch/x86/analyzer.c	2026-01-15 18:13:13.887125024 +0100
+++ analyzer.c	2026-01-15 18:13:45.592315387 +0100
@@ -17,7 +17,7 @@
  */
 
 
-typedef char bool;
+#include <stdbool.h>
 #define true 1
 #define false 0
 #define MEM_ACCESS_NOT_COMPLETED 2
@@ -590,7 +590,7 @@
 // out : value, is_stack
 // return true if there is an error (example: a register is invalid or
 // not defined)
-static bool get_op_value(struct regs_context *regs, PyObject *insn, 
+static bool get_op_value(struct regs_context *regs, PyObject *insn,
                          PyObject *op, long *value, bool *is_stack)
 {
     int r, base, index, scale, disp;
