--- src/plasma-disasm/plasma/lib/arch/mips/analyzer.c	2026-01-15 18:07:42.244327147 +0100
+++ analyzer-mips.c	2026-01-15 18:08:32.455061216 +0100
@@ -20,7 +20,7 @@
 // TODO MIPS64: warning to all casts to int
 
 
-typedef char bool;
+#include <stdbool.h>
 #define true 1
 #define false 0
 
@@ -412,7 +412,7 @@
 // out : value, is_stack
 // return true if there is an error (example: a register is invalid or
 // not defined)
-static bool get_op_value(struct regs_context *regs, PyObject *insn, 
+static bool get_op_value(struct regs_context *regs, PyObject *insn,
                          PyObject *op, long *value, bool *is_stack,
                          bool use_real_gp)
 {
