--- src/plasma/setup.py	2020-12-22 00:10:57.536665399 +0100
+++ setup.py	2020-12-22 00:10:44.453332063 +0100
@@ -13,19 +13,6 @@
 from distutils.core import Extension
 import plasma
 
-requirements = parse_requirements('requirements.txt', session=False)
-
-requires = []
-for item in requirements:
-    # we want to handle package names and also repo urls
-    if getattr(item, 'url', None):  # older pip has url
-        links.append(str(item.url))
-    if getattr(item, 'link', None): # newer pip has link
-        links.append(str(item.link))
-    if item.req:
-        requires.append(str(item.req))
-
-
 x86_analyzer = Extension('plasma.lib.arch.x86.analyzer',
     sources = ['plasma/lib/arch/x86/analyzer.c'])
 
@@ -59,7 +46,12 @@
               'plasma.scripts',
     ],
     package_dir={'plasma':'plasma'},
-    install_requires=requires,
+    install_requires=[
+      'pefile',
+      'pyelftools',
+      'msgpack',
+      'nose',
+    ],
     entry_points = {
         "console_scripts": [
             "plasma = plasma.main:console_entry",
