--- src/semgrep-0.85.0/setup.py	2022-03-27 16:26:28.646665698 +0200
+++ setup.py	2022-03-27 16:27:31.956665697 +0200
@@ -115,18 +115,18 @@
     # Flexibility is achieved by, in order of preference:
     # 1. x.0~= operator pinning to x major version
     # 2. >=x,<y operator pinning to multiple major versions
-    "attrs~=21.3",
-    "colorama~=0.4.0",
-    "click~=8.0",
-    "click-option-group~=0.5",
-    "requests~=2.22",
-    "ruamel.yaml>=0.16.0,<0.18",
-    "tqdm~=4.46",
-    "packaging~=21.0",
-    "jsonschema~=3.2",
-    "wcmatch~=8.3",
-    "peewee~=3.14",
-    "defusedxml~=0.7.1",
+    "attrs",
+    "colorama",
+    "click",
+    "click-option-group",
+    "requests",
+    "ruamel.yaml",
+    "tqdm",
+    "packaging",
+    "jsonschema",
+    "wcmatch",
+    "peewee",
+    "defusedxml",
     # Include 'setuptools' for 'pkg_resources' usage. We shouldn't be
     # overly prescriptive and pin the version for two reasons: 1) because
     # it may interfere with other 'setuptools' installs on the system,
