--- src/artillery/src/core.py	2015-08-23 16:23:00.837369736 +0200
+++ core.py	2015-08-23 16:23:18.447274335 +0200
@@ -490,17 +490,17 @@
 # format the ip addresses and check to ensure they aren't duplicates
 def format_ips(url):
   try:
-      req = urllib2.Request(url)      
+      req = urllib2.Request(url)
       f = urllib2.urlopen(req).readlines()
   except urllib2.HTTPError, err:
       if err.code == '404':
           # Error 404, page not found!
           write_log("HTTPError: Error 404, URL {} not found.".format(url))
-      return 
-  except urllib2.URLError, err: 
+      return
+  except urllib2.URLError, err:
         # Name or service not found known, DNS unreachable, try again later!
         write_log("Received URL Error, Reason: {}".format(err.reason))
-      return
+        return
   else:
       fileopen = file("/usr/share/artillery/banlist.txt", "r").read()
       # write the file
@@ -509,7 +509,7 @@
       for line in f:
           line=line.rstrip()
           if not "#" in line:
-              if not "//" in line:  
+              if not "//" in line:
                   # if we don't have the IP yet
                   if not line in fileopen:
                       filewrite.write(line + "\n")
@@ -522,9 +522,9 @@
 		# pull source feeds
 		url = ['http://rules.emergingthreats.net/blockrules/compromised-ips.txt','https://zeustracker.abuse.ch/blocklist.php?download=badips','https://palevotracker.abuse.ch/blocklists.php?download=ipblocklist','http://malc0de.com/bl/IP_Blacklist.txt']
 		for urls in url:
-			format_ips(urls)	
+			format_ips(urls)
 		time.sleep(7200) # sleep for 2 hours
-	
+
 def sort_banlist():
 	ips = file("/usr/share/artillery/banlist.txt", "r").read()
 	banner = """#
