--- src/wayback-0.4.5/setup.py	2024-02-01 19:55:24.000000000 +0100
+++ setup.py	2024-05-12 01:33:47.640000679 +0200
@@ -71,11 +71,6 @@
         ]
     },
     install_requires=read_requirements('requirements.txt'),
-    extras_require={
-        'dev': read_requirements('requirements-dev.txt'),
-        'docs': read_requirements('requirements-docs.txt'),
-        'test': read_requirements('requirements-test.txt'),
-    },
     license="BSD (3-clause)",
     project_urls={
         'Documentation': 'https://wayback.readthedocs.io/en/stable/',
