--- src/DIE-engine/build_console_lin.sh	2019-02-17 17:22:52.529999504 +0100
+++ build_console_lin.sh	2019-02-17 17:23:50.009999500 +0100
@@ -1,14 +1,12 @@
-#/bin/bash -x
-
-QMAKE=/home/hors/QT/4.8.7/bin/qmake
-
-$QMAKE DIE_source/console_source/console_source.pro -r -spec linux-g++
-make clean
-make
-
-rm -f Makefile
-find . -type f -name "*.o" -exec rm -f {} \;
-find . -type f -name "moc_*" -exec rm -f {} \;
-
-
-
+#!/bin/bash
+
+QMAKE=/usr/bin/qmake-qt4
+
+$QMAKE die_source/console_source/console_source.pro -r -spec linux-g++
+make clean
+make
+
+rm -f Makefile
+find . -type f -name "*.o" -exec rm -f {} \;
+find . -type f -name "moc_*" -exec rm -f {} \;
+
