--- src/beef/Gemfile	2018-09-23 16:10:05.660000059 +0200
+++ Gemfile	2018-09-23 16:10:58.380000054 +0200
@@ -8,9 +8,9 @@
 
 gem 'eventmachine'
 gem 'thin'
-gem 'sinatra', '~> 2.0'
-gem 'rack', '~> 2.0'
-gem 'rack-protection', '~> 2.0'
+gem 'sinatra'
+gem 'rack'
+gem 'rack-protection'
 gem 'em-websocket' # WebSocket support
 gem 'uglifier'
 gem 'mime-types'
@@ -20,10 +20,11 @@
 gem 'dm-core'
 gem 'json'
 gem 'data_objects'
-gem 'rubyzip', '>= 1.2.1'
-gem 'espeak-ruby', '>= 1.0.4' # Text-to-Voice
-gem 'nokogiri', '>= 1.7'
+gem 'rubyzip'
+gem 'espeak-ruby' # Text-to-Voice
+gem 'nokogiri'
 gem 'rake'
+gem 'pkg-config'
 
 # SQLite support
 group :sqlite do
