diff --git a/Gemfile b/Gemfile
index ebdbbc80..a4afac03 100644
--- a/Gemfile
+++ b/Gemfile
@@ -22,6 +22,7 @@ gem 'data_objects'
 gem 'rubyzip', '~> 1.2.1'
 gem 'espeak-ruby', '~> 1.0.4' # Text-to-Voice
 gem 'nokogiri', '~> 1.7.1'
+gem 'rake'
 
 if RUBY_PLATFORM.downcase.include?('linux')
   gem 'therubyracer'
@@ -79,7 +80,6 @@ end
 # For running unit tests
 group :test do
 if ENV['BEEF_TEST']
-  gem 'rake'
   gem 'test-unit'
   gem 'test-unit-full'
   gem 'curb'
