--- src/whatportis/setup.py	2021-12-14 14:32:30.573329719 +0100
+++ setup.py	2021-12-14 14:33:04.853329715 +0100
@@ -18,11 +18,11 @@
     packages=['whatportis'],
     include_package_data=True,
     install_requires=[
-        "simplejson==3.17.2",
-        "tinydb==4.4.0",
-        "requests==2.25.1",
-        "prettytable==2.1.0",
-        "click==8.0.1"
+        "simplejson",
+        "tinydb",
+        "requests",
+        "prettytable",
+        "click"
     ],
     extras_require={
         "dev": [
@@ -31,7 +31,7 @@
             "black"
         ],
         "server": [
-            "flask==1.1.2"
+            "flask"
         ]
     },
     entry_points=entry_points,
