--- WhatWaf/content/__init__.py	2022-06-04 14:33:51.458508913 -0400
+++ /usr/share/whatwaf/content/__init__.py	2022-06-04 14:35:33.378695105 -0400
@@ -4,15 +4,8 @@
 import importlib
 import random
 import threading
-try:
-    import urlparse
-except ImportError:
-    # python 2.x doesn't have a ModuleNotFoundError so we'll just catch the exception I guess
-    import urllib.parse as urlparse
-try:
-    import queue
-except ImportError:
-    import Queue as queue
+import urllib.parse as urlparse
+import queue
 
 import lib.settings
 import lib.formatter
@@ -314,13 +307,15 @@
                     break
             # simple sloppy little fix for issue #376, we'll just continue if we hit the problem
             # i honestly have no idea if this will cause future issues or not
+            # @dualfade --
+            # ^^ it does; updated for python3 only --
             except RuntimeError:
                 pass
             except Exception as e:
                 if "'NoneType' object is not iterable" in str(e):
                     pass
                 else:
-                    raise e.__class__("Exception caught: {} ~~> {}".format(e.__class__, e.message))
+                   raise e.__class__("Exception caught: {} ~~> {}".format(e.__class__, e))
         if len(working_tampers) == max_successful_payloads:
             break
     return working_tampers
