Index: log4cpp-1.0/src/BasicLayout.cpp
===================================================================
--- log4cpp-1.0.orig/src/BasicLayout.cpp	2008-03-14 16:06:57.000000000 -0400
+++ log4cpp-1.0/src/BasicLayout.cpp	2008-03-14 16:07:11.000000000 -0400
@@ -15,6 +15,8 @@
 #include <sstream>
 #endif
 
+#include <memory>
+
 namespace log4cpp {
 
     BasicLayout::BasicLayout() {
Index: log4cpp-1.0/src/PatternLayout.cpp
===================================================================
--- log4cpp-1.0.orig/src/PatternLayout.cpp	2008-03-14 16:07:33.000000000 -0400
+++ log4cpp-1.0/src/PatternLayout.cpp	2008-03-14 16:07:49.000000000 -0400
@@ -370,7 +370,7 @@
                         literal = "";
                     }
                     if ((minWidth != 0) || (maxWidth != 0)) {
-                        component = new FormatModifierComponent(component, std::abs(minWidth), maxWidth, minWidth < 0);
+                        component = new FormatModifierComponent(component, std::abs((float)minWidth), maxWidth, minWidth < 0);
                         minWidth = maxWidth = 0;
                     }
                     _components.push_back(component);
