--- src/Shiboken-1.2.4/libshiboken/basewrapper.cpp	2015-10-14 20:12:33.000000000 +0200
+++ basewrapper.cpp	2025-12-25 15:05:18.420933908 +0100
@@ -265,7 +265,7 @@
     SbkObjectType* sbkType = reinterpret_cast<SbkObjectType*>(pyObj);
 
     PyObject_GC_UnTrack(pyObj);
-    Py_TRASHCAN_SAFE_BEGIN(pyObj);
+    Py_XDECREF(pyObj);
     if (sbkType->d) {
         if(sbkType->d->user_data && sbkType->d->d_func) {
             sbkType->d->d_func(sbkType->d->user_data);
@@ -278,7 +278,6 @@
         delete sbkType->d;
         sbkType->d = 0;
     }
-    Py_TRASHCAN_SAFE_END(pyObj);
 }
 
 PyObject* SbkObjectTypeTpNew(PyTypeObject* metatype, PyObject* args, PyObject* kwds)
