--- src/shiboken-1.1.2/cmake/Modules/FindPython3Libs.cmake	2012-07-04 20:25:37.000000000 +0200
+++ FindPython3Libs.cmake	2012-11-04 01:59:19.125481523 +0100
@@ -27,7 +27,7 @@
 # Search for the python framework on Apple.
 # CMAKE_FIND_FRAMEWORKS(Python)
 
-FOREACH(_CURRENT_VERSION 3.2 3.1 3.0)
+FOREACH(_CURRENT_VERSION 3.3 3.2 3.1 3.0)
   IF(_CURRENT_VERSION GREATER 3.1)
       SET(_32FLAGS "m" "u" "mu" "")
   ELSE()
