--- src/Shiboken-1.2.4/libshiboken/sbkenum.cpp	2025-12-25 15:11:35.368667769 +0100
+++ sbkenum.cpp	2025-12-25 15:12:38.107740669 +0100
@@ -362,11 +362,10 @@
     SbkEnumType* sbkType = reinterpret_cast<SbkEnumType*>(pyObj);

     PyObject_GC_UnTrack(pyObj);
-    Py_TRASHCAN_SAFE_BEGIN(pyObj);
+    Py_XDECREF(pyObj);
     if (sbkType->converter) {
         Shiboken::Conversions::deleteConverter(sbkType->converter);
     }
-    Py_TRASHCAN_SAFE_END(pyObj);
 }

 PyObject* SbkEnumTypeTpNew(PyTypeObject* metatype, PyObject* args, PyObject* kwds)
@@ -534,7 +533,6 @@
     #endif
     type->tp_basicsize = sizeof(SbkEnumObject);
     #ifndef IS_PY3K
-    type->tp_print = &SbkEnumObject_print;
     #endif
     type->tp_repr = &SbkEnumObject_repr;
     type->tp_str = &SbkEnumObject_repr;
