--- src/Shiboken-1.2.4/libshiboken/sbkstring.cpp	2015-10-14 20:12:33.000000000 +0200
+++ sbkstring.cpp	2025-12-25 15:27:28.743076079 +0100
@@ -20,6 +20,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <Python.h>
 #include "sbkstring.h"
 #include "autodecref.h"
 
@@ -87,7 +88,7 @@
         if (len) {
             // We need to encode the unicode string into utf8 to know the size of returned char*.
             Shiboken::AutoDecRef uniStr(PyUnicode_AsUTF8String(str));
-            *len = PyBytes_GET_SIZE(uniStr.object());
+            *len = PyBytes_GET_LENGTH(uniStr.object());
         }
         // Return unicode from str instead of uniStr, because the lifetime of the returned pointer
         // depends on the lifetime of str.
@@ -96,7 +97,7 @@
 #endif
     if (PyBytes_Check(str)) {
         if (len)
-            *len = PyBytes_GET_SIZE(str);
+            *len = PyBytes_GET_LENGTH(str);
         return PyBytes_AS_STRING(str);
     }
     return 0;
@@ -173,10 +174,10 @@
         return 0;
 
     if (PyUnicode_Check(str))
-        return PyUnicode_GET_SIZE(str);
+        return PyUnicode_GET_LENGTH(str);
 
     if (PyBytes_Check(str))
-        return PyBytes_GET_SIZE(str);
+        return PyBytes_GET_LENGTH(str);
     return 0;
 }
 
