diff -uNr hping2-rc3.orig/bytesex.h hping2-rc3/bytesex.h
--- hping2-rc3.orig/bytesex.h	2004-07-29 15:46:47.572596272 +0000
+++ hping2-rc3/bytesex.h	2004-07-29 15:48:14.557372576 +0000
@@ -9,11 +9,14 @@
 
 #if 	defined(__i386__) \
 	|| defined(__alpha__) \
+	|| defined(__x86_64__) \
+	|| defined(__ia64__) \
 	|| (defined(__mips__) && (defined(MIPSEL) || defined (__MIPSEL__)))
 #define BYTE_ORDER_LITTLE_ENDIAN
 #elif 	defined(__mc68000__) \
 	|| defined (__sparc__) \
 	|| defined (__sparc) \
+	|| defined (__hppa__) \
 	|| defined (__PPC__) \
 	|| defined (__BIG_ENDIAN__) \
 	|| (defined(__mips__) && (defined(MIPSEB) || defined (__MIPSEB__)))
