--- src/SFlock-0.3.6/setup.py	2018-09-19 10:56:24.000000000 +0200
+++ setup.py	2018-09-23 12:15:39.509999837 +0200
@@ -25,10 +25,10 @@
     description="Sample staging and detonation utility",
     include_package_data=True,
     install_requires=[
-        "click==6.6",
-        "cryptography>=2.1",
-        "olefile==0.43",
-        "peepdf>=0.4.1",
-        "python-magic==0.4.12",
+        "click",
+        "cryptography",
+        "olefile",
+        "peepdf",
+        "python-magic",
     ],
 )
