--- src/pcileech-4.2/pcileech/Makefile	2019-07-30 13:00:53.000000000 +0200
+++ Makefile	2019-07-31 12:18:23.533333174 +0200
@@ -8,12 +8,9 @@
 	$(CC) -c -o $@ $< $(CFLAGS)
 
 pcileech: $(OBJ)
-	cp ../files/leechcore.so . |true
-	cp ../../LeechCore-dev/files/leechcore.so . |true
 	$(CC) -o $@ $^ $(CFLAGS) $(LDFLAGS)
 	rm *.o
 	mv pcileech ../files/ |true
-	mv leechcore.so ../files/ |true
 
 clean:
 	rm *.o
