--- allthevhosts.py	2015-02-17 22:30:13.821814743 -0500
+++ allthevhosts.patched.py	2015-02-17 22:35:33.577725292 -0500
@@ -80,9 +80,12 @@

 def yougetsignal():
 	print "[-] searching yougetsignal...",
-	req = urllib2.Request(url='http://www.yougetsignal.com/tools/web-sites-on-web-server/php/get-web-sites-on-web-server-json-data.php',data='remoteAddress='+ip+'&key=')
+	req = urllib2.Request(url='http://domains.yougetsignal.com/domains.php',data='remoteAddress='+ip+'&key=&_=')
 	req.add_header('Referer', 'http://www.yougetsignal.com/tools/web-sites-on-web-server/')
-	req.add_header('User-Agent', 'firefox')
+	req.add_header('Origin', 'http://www.yougetsignal.com')
+	req.add_header('User-Agent', 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.111 Safari/537.36')
+	req.add_header('X-Prototype-Version', '1.6.0')
+	req.add_header('X-Requested-With', 'XMLHttpRequest')
 	search = urllib2.urlopen(req)
 	results = json.load(search)
 	try:
