--- src/cisco-ocs/OCS-0.2.c	2025-12-22 01:06:36.314387539 +0100
+++ OCS-0.2.c	2025-12-22 01:06:27.836122370 +0100
@@ -37,10 +37,11 @@
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <arpa/inet.h>
 #include <unistd.h>
 #include <string.h>
 #include <signal.h>
-
+#include <pthread.h>
 
 int i=0;
 int j=0;
@@ -55,6 +56,7 @@
 
 pthread_t threadname;
 
+int scanna(char *);
 
 void callScan()        // scanning
 {
@@ -336,4 +338,4 @@
 	return byte;
 }
 
- 
+
