--- src/0trace/0trace.sh	2007-01-06 18:22:12.000000000 -0500
+++ 0trace/0trace.sh	2013-10-25 04:21:33.624030797 -0400
@@ -9,7 +9,7 @@
   exit 1
 fi
 
-if [ ! -x /bin/usleep ]; then
+if [ ! -x /usr/bin/usleep ]; then
   echo "[-] /bin/usleep not found on this system, sorry." 1>&2
   exit 1
 fi
@@ -18,8 +18,7 @@
   echo "[-] WARNING: Only Linux is believed to work fine with this utility." 1>&2
 fi
 
-make sendprobe >/dev/null
-test -f ./sendprobe || exit 1
+test -f /usr/bin/sendprobe || exit 1
 
 echo "0trace v0.01 PoC by <lcamtuf@coredump.cx>"
 
@@ -28,7 +27,7 @@
 
 echo "[+] Waiting for traffic from target on $1..."
 
-/usr/sbin/tcpdump -c 1 -s 200 -S -q -i "$1" -n -x "$RULE" >"$T" 2>/dev/null
+/usr/bin/tcpdump -c 1 -s 200 -S -q -i "$1" -n -x "$RULE" >"$T" 2>/dev/null
 
 if [ ! -s "$T" ]; then
   echo "[-] Something went wrong with tcpdump (check parameters)."
@@ -42,9 +41,9 @@
 WAITTIME=0
 
 while [ "$WAITTIME" -lt "80" ]; do
-  /usr/sbin/tcpdump -c 1 -s 200 -S -q -i "$1" -n -x "$RULE" >"$T-2" 2>/dev/null &
+  /usr/bin/tcpdump -c 1 -s 200 -S -q -i "$1" -n -x "$RULE" >"$T-2" 2>/dev/null &
   TPID="$!"
-  usleep 100000
+  /usr/bin/usleep 100000
 
   while kill -0 "$TPID" 2>/dev/null; do
     WAITING=$[WAITING+1]
@@ -52,7 +51,7 @@
       kill "$TPID" 2>/dev/null
       break
     fi
-    usleep 100000
+    /usr/bin/usleep 100000
   done
   
   test -s "$T-2" || break
@@ -80,12 +79,12 @@
 
 echo "[+] Setting up a sniffer..."
 
-/usr/sbin/tcpdump -l -s 200 -S -q -i "$1" -n -x "icmp or ($RULE)" >"$T" 2>/dev/null &
+/usr/bin/tcpdump -l -s 200 -S -q -i "$1" -n -x "icmp or ($RULE)" >"$T" 2>/dev/null &
 TPID="$!"
 
 echo "[+] Sending probes..."
 
-./sendprobe $SADDR $DADDR $SPORT $DPORT $SEQ $ACK
+/usr/bin/sendprobe $SADDR $DADDR $SPORT $DPORT $SEQ $ACK
 sleep 2
 kill "$TPID" 2>/dev/null
 
