--- src/obexftp-0.24.2-Source/swig/python/CMakeLists.txt	2013-08-11 20:26:37.000000000 +0200
+++ CMakeLists.txt	2024-11-17 19:42:36.313332015 +0100
@@ -1,4 +1,3 @@
-
 find_package ( PythonLibs REQUIRED )
 find_package ( PythonInterp REQUIRED )
 
@@ -31,25 +30,8 @@
   obexftp
 )
 
-#
-# Get the install directory for ruby modules and strip the ruby install prefix
-# so that we can use our own prefix.
-#
-execute_process (
-  COMMAND ${PYTHON_EXECUTABLE} -c
-  	"import site, sys; sys.stdout.write(site.PREFIXES[-1])"
-  OUTPUT_VARIABLE PYTHON_PREFIX
-)
-file ( TO_CMAKE_PATH "${PYTHON_PREFIX}" PYTHON_PREFIX )
-execute_process (
-  COMMAND ${PYTHON_EXECUTABLE} -c
-  	"import site, sys; sys.stdout.write(site.getsitepackages()[-1])"
-  OUTPUT_VARIABLE PYTHON_SITE_DIR
-)
-file ( TO_CMAKE_PATH "${PYTHON_SITE_DIR}" PYTHON_SITE_DIR )
-string ( REGEX REPLACE "^${PYTHON_PREFIX}/" ""
-  PYTHON_SITE_DIR "${PYTHON_SITE_DIR}"
-)
+# Statischer Installationspfad für Python-Module
+set(PYTHON_SITE_DIR "/usr/lib/python2.7/site-packages")
 
 install ( TARGETS ${SWIG_MODULE_obexftp-python_REAL_NAME}
   LIBRARY
@@ -61,3 +43,4 @@
   DESTINATION ${PYTHON_SITE_DIR}
   COMPONENT library
 )
+
