diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index b5c84e4..63af134 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -5,14 +5,23 @@ add_definitions( -DVERSION="${obexftp_VERSION}" )
 add_definitions ( -DHAVE_USB )
 
 add_executable ( obexftp_app obexftp.c )
-target_link_libraries ( obexftp_app obexftp )
+target_link_libraries ( obexftp_app
+  PRIVATE multicobex
+  PRIVATE bfb
+  obexftp
+)
 set_target_properties ( obexftp_app PROPERTIES
   OUTPUT_NAME obexftp
 )
 
 
 add_executable ( obexftpd_app obexftpd.c )
-target_link_libraries ( obexftpd_app obexftp openobex )
+target_link_libraries ( obexftpd_app
+  PRIVATE multicobex
+  PRIVATE bfb
+  obexftp
+  openobex
+)
 set_target_properties ( obexftpd_app PROPERTIES
   OUTPUT_NAME obexftpd
 )
diff --git a/fuse/CMakeLists.txt b/fuse/CMakeLists.txt
index 0c80d3d..ff5e3cd 100644
--- a/fuse/CMakeLists.txt
+++ b/fuse/CMakeLists.txt
@@ -13,6 +13,8 @@ if ( Fuse_FOUND )
   )
 
   target_link_libraries ( obexfs
+    PRIVATE multicobex
+    PRIVATE bfb
     obexftp
     ${Fuse_LIBRARIES}
     ${EXPAT_LIBRARIES}
@@ -23,6 +25,8 @@ if ( Fuse_FOUND )
   )
 
   target_link_libraries ( obexautofs
+    PRIVATE multicobex
+    PRIVATE bfb
     obexftp
     ${Fuse_LIBRARIES}
   )

