diff -rupN Honeyd-honeyd-1.6.7.orig/Makefile Honeyd-honeyd-1.6.7/Makefile
--- Honeyd-honeyd-1.6.7.orig/Makefile	2013-11-16 19:09:13.467618562 -0500
+++ Honeyd-honeyd-1.6.7/Makefile	2013-11-16 19:09:57.180952740 -0500
@@ -426,7 +426,7 @@ PACKAGE_STRING =
 PACKAGE_TARNAME = 
 PACKAGE_URL = 
 PACKAGE_VERSION = 
-PATH_PYTHON = /usr/bin/python
+PATH_PYTHON = $(which python2)
 PATH_PYTHON24 = 
 PATH_RRDTOOL = 
 PATH_SEPARATOR = :
