--- src/xsscon/lib/helper/helper.py	2019-05-26 01:09:59.946666635 +0200
+++ helper.py	2019-05-26 01:10:51.696666637 +0200
@@ -1,23 +1,23 @@
 '''
 XSSCon - 2019/2020
-This project was created by menkrep1337 with 407Aex team. 
+This project was created by menkrep1337 with 407Aex team.
 Copyright under the MIT license
 '''
 import requests
-##### Warna ####### 
+##### Warna #######
 N = '\033[0m'
-W = '\033[1;37m' 
-B = '\033[1;34m' 
-M = '\033[1;35m' 
-R = '\033[1;31m' 
-G = '\033[1;32m' 
-Y = '\033[1;33m' 
-C = '\033[1;36m' 
+W = '\033[1;37m'
+B = '\033[1;34m'
+M = '\033[1;35m'
+R = '\033[1;31m'
+G = '\033[1;32m'
+Y = '\033[1;33m'
+C = '\033[1;36m'
 ##### Styling ######
 underline = "\033[4m"
 ##### Default ######
-agent = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'} 
-line="—————————————————" 
+agent = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
+line="—————————————————"
 #####################
 def session(proxies,headers,cookie):
 	r=requests.Session()
@@ -31,10 +31,5 @@
  \  /\___ \___ \| |   / _ \| '_ \ %s
  /  \ ___) |__) | |__| (_) | | | | %s
 /_/\_\____/____/ \____\___/|_| |_|
-<<<<<<< HEAD
 """%(R+"{v0.5B}"+G,underline+C+"https://github.com/menkrep1337/XSSCon"+N+G)
-	
-=======
-"""%(R+"{v0.4B}"+G,underline+C+"https://github.com/menkrep1337/XSSCon"+N+G)
-	
->>>>>>> branch 'master' of https://github.com/menkrep1337/XSSCon
+
