--- rsakeyfind.cpp	2008-07-18 08:18:40.000000000 +0200
+++ rsakeyfind_patch.cpp	2013-12-17 02:20:28.528066804 +0100
@@ -1,14 +1,14 @@
 // RSAKeyFinder 1.0 (2008-07-18)
 // By Nadia Heninger and J. Alex Halderman
 
-#include <fcntl.h>
-#include <stdlib.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
-#include <string>
+#include <string.h>
 #include <iostream>
 
 #ifdef __FreeBSD__
