*** setup.py.orig	2009-08-02 17:16:31.425667179 +0800
--- setup.py	2009-08-02 17:18:53.985653155 +0800
***************
*** 266,273 ****
  
          self.set_perms()
          self.set_modules_path()
!         self.fix_paths()
!         self.create_uninstaller()
          self.finish_banner()
  
      def create_uninstaller(self):
--- 266,273 ----
  
          self.set_perms()
          self.set_modules_path()
!         #self.fix_paths()
!         #self.create_uninstaller()
          self.finish_banner()
  
      def create_uninstaller(self):
***************
*** 330,336 ****
          if self.pkgmaintainer and self.root:
              modules = os.path.join('/', modules[len(self.root):])
  
!         ucontent.insert(uline, "sys.path.insert(0,'%s')\n" % modules)
  
          ufile = open(umit, "w")
          ufile.writelines(ucontent)
--- 330,337 ----
          if self.pkgmaintainer and self.root:
              modules = os.path.join('/', modules[len(self.root):])
  
!         modules = "/usr/share/umit"
!         ucontent.insert(uline, "sys.path = ['%s'] + sys.path\n" % modules)
  
          ufile = open(umit, "w")
          ufile.writelines(ucontent)
