--- src/msldap-0.3.20/setup.py	2020-10-30 00:18:16.000000000 +0100
+++ setup.py	2020-11-15 23:54:29.749999416 +0100
@@ -49,10 +49,10 @@
 	install_requires=[
 		'asn1crypto',
 		'winsspi>=0.0.9;platform_system=="Windows"',
-		'minikerberos>=0.2.4',
-		'asysocks>=0.0.10',
-		'winacl>=0.1.0',
-		'prompt-toolkit>=3.0.2',
+		'minikerberos',
+		'asysocks',
+		'winacl',
+		'prompt-toolkit',
 		'tqdm',
 	],
 	entry_points={
@@ -61,4 +61,4 @@
 			'msldapcompdns = msldap.examples.msldapcompdnslist:main',
 		],
 	}
-)
\ No newline at end of file
+)
