--- src/msldap-0.3.4/setup.py	2020-04-21 10:26:17.000000000 +0200
+++ setup.py	2020-05-20 22:50:16.469999314 +0200
@@ -48,12 +48,12 @@
 	install_requires=[
 		'asn1crypto',
 		'winsspi>=0.0.9;platform_system=="Windows"',
-		'minikerberos>=0.2.1',
+		'minikerberos',
 		'aiocmd',
 		'asciitree',
 		'asysocks',
-		'winacl>=0.0.2',
-		'prompt-toolkit<3.0.0',
+		'winacl',
+		'prompt-toolkit',
 		'tqdm',
 	],
 	entry_points={
@@ -61,4 +61,4 @@
 			'msldap = msldap.examples.msldapclient:main',
 		],
 	}
-)
\ No newline at end of file
+)
