diff -aur owtf/profiles/general/default.cfg owtf.orig/profiles/general/default.cfg
+++ owtf/profiles/general/default.cfg	2014-10-28 23:24:25.145848976 -0700
--- owtf.orig/profiles/general/default.cfg	2014-10-28 23:06:23.545716862 -0700
@@ -57,12 +57,12 @@
 TOOL_METAGOOFIL_DIR: /usr/share/metagoofil
 # Backtrack 5 R1, no longer working in Backtrack 5 R2:
 #TOOL_HTTPRINT_DIR: /pentest/enumeration/www/httprint/linux
+TOOL_HTTPRINT_DIR: /usr/share/httprint
-TOOL_HTTPRINT_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/httprint/httprint_301/linux
 TOOL_WAFW00F: /usr/bin/wafw00f
 TOOL_NMAP: /usr/bin/nmap
 TOOL_WHATWEB: /usr/bin/whatweb
 #TOOL_WHATWEB: @@@FRAMEWORK_DIR@@@/tools/restricted/whatweb/whatweb-0.4.7/whatweb
+TOOL_CMS_EXPLORER_DIR: /usr/share/cms-explorer/cms-explorer
-TOOL_CMS_EXPLORER_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/cms-explorer/cms-explorer-1.0
 TOOL_LBD: /usr/bin/lbd
 #TOOL_REV_HOSTS_DIR: /pentest/enumeration/web/revhosts
 TOOL_DNS_RECON: /usr/bin/dnsrecon
@@ -73,8 +73,8 @@
 #Ran out of time to give this a proper shot:
 #TOOL_ARACHNI_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/arachni-v0.4.1/arachni-0.4.1/bin
 #TOOL_ARACHNI_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/arachni-v0.3-cde
+TOOL_ARACHNI_BIN: /usr/bin/arachni
+TOOL_ARACHNI_REPORTER_BIN: /usr/bin/arachni_reporter
-TOOL_ARACHNI_BIN: /usr/share/arachni/bin/arachni
-TOOL_ARACHNI_REPORTER_BIN: /usr/share/arachni/bin/arachni_reporter
 #TOOL_NIKTO_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/nikto
 # We are patching nikto's poor default user agent in the install script now, 1 tool less to install (in Backtrack!):
 TOOL_NIKTO_BIN: /usr/bin/nikto
@@ -86,7 +86,7 @@
 # Kali Linux:
 TOOL_DIRBUSTER_BIN: /usr/share/dirbuster/DirBuster-1.0-RC1.jar
 #TOOL_DIRBUSTER_DIR: /usr/share/dirbuster
+TOOL_METASPLOIT_DIR: /usr/share/metasploit
-TOOL_METASPLOIT_DIR: /opt/metasploit/apps/pro/msf3
 TOOL_HOPPY_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/hoppy-1.8.1/hoppy-1.8.1
 TOOL_WEBSECURIFY: @@@FRAMEWORK_DIR@@@/tools/restricted/websecurify/Websecurify_Scanner_0.9/runner
 TOOL_WPSCAN: /usr/bin/wpscan
@@ -96,7 +96,7 @@
 TOOL_HTTP_TRACEROUTE: @@@FRAMEWORK_DIR@@@/tools/discovery/web/rev_proxy/HTTP-Traceroute.py
 TOOL_HTTP_DOS_SLOWLORIS: @@@FRAMEWORK_DIR@@@/tools/dos/web/slowloris.pl
 TOOL_HTTP_DOS_HASH_COLLISION: @@@FRAMEWORK_DIR@@@/tools/dos/web/HashCollision-DOS-POC/HashtablePOC.py
+TOOL_PANOPTIC_DIR: /usr/share/panoptic
-TOOL_PANOPTIC_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/Panoptic
 TOOL_DNSPIDER: @@@FRAMEWORK_DIR@@@/tools/restricted/dnspider/dnsspider-0.6.py
 TOOL_DNSPIDER_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/dnspider/
 
