--- src/owtf/profiles/general/default.cfg	2014-06-08 08:42:02.454648031 -0400
+++ src/owtf/profiles/general/default.cfg	2014-06-08 08:41:48.764647438 -0400
@@ -36,7 +36,14 @@
 UPDATE_REPORT_AFTER_EACH_COMMAND: No
 
 # ************************* File system tool locations *************************************
-# Kali Linux paths:
+
+# [*] WARNING: Tool path not found for: /usr/share/owtf/tools/restricted/ssl/ssl-cipher-check/ssl-cipher-check.pl
+#	http://www.unspecific.com/ssl/ssl-cipher-check.pl
+# [*] WARNING: Tool path not found for: /usr/share/owtf/tools/restricted/websecurify/Websecurify Scanner 0.9/runner
+# [*] WARNING: Tool path not found for: /usr/share/owtf/tools/restricted/hoppy-1.8.1/hoppy-1.8.1
+#	https://labs.portcullis.co.uk/tools/hoppy/
+# [*] WARNING: Tool path not found for: /usr/share/owtf/tools/restricted/Panoptic
+# 	https://github.com/lightos/Panoptic
 
 TOOL_SET_DIR: /usr/share/set
 TOOL_THEHARVESTER_BIN: /usr/bin/theharvester
@@ -44,13 +51,13 @@
 TOOL_METAGOOFIL_DIR: /usr/share/metagoofil
 # Backtrack 5 R1, no longer working in Backtrack 5 R2:
 #TOOL_HTTPRINT_DIR: /pentest/enumeration/www/httprint/linux
-TOOL_HTTPRINT_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/httprint/httprint_301/linux
+TOOL_HTTPRINT_DIR: /usr/share/httprint
 TOOL_WAFW00F: /usr/bin/wafw00f
 TOOL_NMAP: /usr/bin/nmap
 TOOL_WHATWEB: /usr/bin/whatweb
 #TOOL_WHATWEB: @@@FRAMEWORK_DIR@@@/tools/restricted/whatweb/whatweb-0.4.7/whatweb
-TOOL_CMS_EXPLORER_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/cms-explorer/cms-explorer-1.0
-TOOL_LBD: @@@FRAMEWORK_DIR@@@/tools/restricted/lbd/lbd.sh
+TOOL_CMS_EXPLORER_DIR: /usr/share/cms-explorer
+TOOL_LBD: /usr/bin/lbd
 #TOOL_REV_HOSTS_DIR: /pentest/enumeration/web/revhosts
 TOOL_DNS_RECON: /usr/bin/dnsrecon
 TOOL_UA_TESTER: /usr/bin/ua-tester
@@ -72,7 +79,7 @@
 # Kali Linux:
 TOOL_DIRBUSTER_BIN: /usr/share/dirbuster/DirBuster-1.0-RC1.jar
 #TOOL_DIRBUSTER_DIR: /usr/share/dirbuster
-TOOL_METASPLOIT_DIR: /opt/metasploit/apps/pro/msf3
+TOOL_METASPLOIT_DIR: /usr/share/metasploit
 TOOL_HOPPY_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/hoppy-1.8.1/hoppy-1.8.1
 TOOL_WEBSECURIFY: @@@FRAMEWORK_DIR@@@/tools/restricted/websecurify/Websecurify Scanner 0.9/runner
 TOOL_WPSCAN: /usr/bin/wpscan
