--- src/owtf/profiles/resources/default.cfg	2014-06-08 08:42:02.454648031 -0400
+++ src/owtf/profiles/resources/default.cfg	2014-06-08 08:41:57.481314483 -0400
@@ -133,29 +133,29 @@
 ActiveInfrastructureConfigurationManagement_____wafw00f WAF Check_____@@@TOOL_WAFW00F@@@ @@@TARGET_URL@@@
 ActiveInfrastructureConfigurationManagement_____Nmap WAF Check (HOST-NAME)_____@@@TOOL_NMAP@@@ -P0 -v -sV --script=http-waf-detect,http-waf-fingerprint --script-args http-waf-fingerprint.intensive=1,http.useragent='@@@USER_AGENT@@@',http-waf-detect.aggro=1,http-waf-detect.detectBodyChanges=1 -p @@@PORT_NUMBER@@@ @@@HOST_NAME@@@
 ActiveInfrastructureConfigurationManagement_____Nmap WAF Check (HOST-IP)_____@@@TOOL_NMAP@@@ -P0 -v -sV --script=http-waf-detect,http-waf-fingerprint --script-args http-waf-fingerprint.intensive=1,http.useragent='@@@USER_AGENT@@@',http-waf-detect.aggro=1,http-waf-detect.detectBodyChanges=1 -p @@@PORT_NUMBER@@@ @@@HOST_IP@@@
-ActiveInfrastructureConfigurationManagement_____LBD Check_____bash @@@TOOL_LBD@@@ @@@HOST_NAME@@@ @@@PORT_NUMBER@@@ @@@URL_SCHEME@@@
+ActiveInfrastructureConfigurationManagement_____LBD Check_____@@@TOOL_LBD@@@ @@@HOST_NAME@@@ @@@PORT_NUMBER@@@ @@@URL_SCHEME@@@
 ActiveInfrastructureConfigurationManagement_____HTTP-Traceroute Reverse proxy Check (using TRACE)_____python @@@TOOL_HTTP_TRACEROUTE@@@ -m TRACE -s @@@URL_SCHEME@@@ -t @@@HOST_NAME@@@ -p @@@PORT_NUMBER@@@ -f 5
 ActiveInfrastructureConfigurationManagement_____HTTP-Traceroute Reverse proxy Check (using GET)_____python @@@TOOL_HTTP_TRACEROUTE@@@ -m GET -s @@@URL_SCHEME@@@ -t @@@HOST_NAME@@@ -p @@@PORT_NUMBER@@@ -f 5
 ActiveInfrastructureConfigurationManagement_____UA-Tester Check_____@@@TOOL_UA_TESTER@@@ -v -u @@@TARGET_URL@@@
 #ActiveFingerPrint_____whatweb Active Check (aggresion 4)_____@@@TOOL_WHATWEB@@@ --user-agent '@@@USER_AGENT@@@' --color=never --aggression 4 @@@TARGET_URL@@@ | sed "s/],/]\n/g"
 ActiveFingerPrint_____whatweb Active Check (aggresion 4)_____@@@TOOL_WHATWEB@@@ --user-agent '@@@USER_AGENT@@@' --color=never --aggression 4 @@@TARGET_URL@@@ | sed "s/],/]\n/g"
 #ActiveFingerPrintProxified_____whatweb Active Check (aggresion 4)_____@@@TOOL_WHATWEB@@@ --user-agent '@@@USER_AGENT@@@' --proxy @@@PROXY@@@ --color=never --aggression 4 @@@TARGET_URL@@@ | sed "s/],/]\n/g"
-ActiveFingerPrint_____httprint FingerPrint Check_____@@@TOOL_HTTPRINT_DIR@@@/httprint -h @@@IP_URL@@@ -s @@@TOOL_HTTPRINT_DIR@@@/signatures.txt -P0
-CMS_FingerPrint_Drupal_____CMS Explorer Plugin Enumeration (Drupal)_____cd @@@TOOL_CMS_EXPLORER_DIR@@@; perl cms-explorer.pl -v 1 -url @@@TARGET_URL@@@ -type Drupal
-CMS_FingerPrint_Drupal_____DirBuster Drupal All_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Drupal_____DirBuster Drupal Plugins_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Drupal_____DirBuster Drupal Themes_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+ActiveFingerPrint_____httprint FingerPrint Check_____httprint -h @@@IP_URL@@@ -s @@@TOOL_HTTPRINT_DIR@@@/signatures.txt -P0
+CMS_FingerPrint_Drupal_____CMS Explorer Plugin Enumeration (Drupal)_____cms-explorer -v 1 -url @@@TARGET_URL@@@ -type Drupal
+CMS_FingerPrint_Drupal_____DirBuster Drupal All_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Drupal_____DirBuster Drupal Plugins_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Drupal_____DirBuster Drupal Themes_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_DRUPAL_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
 CMS_FingerPrint_WordPress_____WPScan Plugin Enumeration (WordPress)_____ruby @@@TOOL_WPSCAN@@@ --url @@@TARGET_URL@@@ --enumerate p --threads 20
-CMS_FingerPrint_WordPress_____CMS Explorer Plugin Enumeration (WordPress)_____cd @@@TOOL_CMS_EXPLORER_DIR@@@; perl cms-explorer.pl -v 1 -url @@@TARGET_URL@@@ -type Wordpress
-CMS_FingerPrint_WordPress_____DirBuster WordPress All_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_WordPress_____DirBuster WordPress Plugins_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_WordPress_____DirBuster WordPress Themes_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Joomla_____CMS Explorer Plugin Enumeration (Joomla)_____cd @@@TOOL_CMS_EXPLORER_DIR@@@; perl cms-explorer.pl -v 1 -url @@@TARGET_URL@@@ -type Joomla
-CMS_FingerPrint_Joomla_____DirBuster Joomla All_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Joomla_____DirBuster Joomla Plugins_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Joomla_____DirBuster Joomla Themes_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-CMS_FingerPrint_Mambo_____CMS Explorer Plugin Enumeration (Mambo)_____cd @@@TOOL_CMS_EXPLORER_DIR@@@; perl cms-explorer.pl -v 1 -url @@@TARGET_URL@@@ -type Mambo
-CMS_FingerPrint_All_____DirBuster All CMS_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_WordPress_____CMS Explorer Plugin Enumeration (WordPress)_____cms-explorer -v 1 -url @@@TARGET_URL@@@ -type Wordpress
+CMS_FingerPrint_WordPress_____DirBuster WordPress All_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_WordPress_____DirBuster WordPress Plugins_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_WordPress_____DirBuster WordPress Themes_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_WORDPRESS_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Joomla_____CMS Explorer Plugin Enumeration (Joomla)_____cms-explorer -v 1 -url @@@TARGET_URL@@@ -type Joomla
+CMS_FingerPrint_Joomla_____DirBuster Joomla All_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Joomla_____DirBuster Joomla Plugins_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_PLUGINS_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Joomla_____DirBuster Joomla Themes_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_JOOMLA_THEMES_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+CMS_FingerPrint_Mambo_____CMS Explorer Plugin Enumeration (Mambo)_____cms-explorer -v 1 -url @@@TARGET_URL@@@ -type Mambo
+CMS_FingerPrint_All_____DirBuster All CMS_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_CMS_ALL_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
 Testing_for_Path_Traversal_All_____Panoptic_____cd @@@TOOL_PANOPTIC_DIR@@@ ; python panoptic.py -a -v -u @@@TARGET_URL@@@ --proxy @@@PROXY@@@ --user-agent '@@@USER_AGENT@@@' -w '###PLUGIN_OUTPUT_DIR###'
 #ActiveDiscovery_____revhosts Virtual Host Check_____cd @@@TOOL_REV_HOSTS_DIR@@@; ./revhosts pig vhh @@@HOST_NAME@@@ | grep -v "^Plugin \["|grep -v ^go$|grep -v ^$|grep -v "^\["
 ActiveDiscovery_____dnsrecon Virtual Host Check_____@@@TOOL_DNS_RECON@@@ --type std,rvl,goo -d @@@TOP_DOMAIN@@@ -r @@@HOST_IP@@@-@@@HOST_IP@@@
@@ -171,10 +171,10 @@
 Nikto_Unauth_____Nikto_____@@@FRAMEWORK_DIR@@@/scripts/run_nikto.sh @@@TOOL_NIKTO_DIR@@@ @@@HOST_IP@@@ @@@PROXY@@@ @@@PORT_NUMBER@@@ @@@HOST_NAME@@@
 Nikto_Unauth_____Extract URLs_____@@@FRAMEWORK_DIR@@@/scripts/extract_urls.sh nikto Nikto @@@TARGET_URL@@@
 Nikto_Verify_Unauth_____Verify Nikto.html_____@@@FRAMEWORK_DIR@@@/scripts/VerifyNikto.py Nikto @@@TARGET_URL@@@
-Wapiti_Unauth_____Wapiti_____cd @@@TOOL_WAPITI_DIR@@@ ; python wapiti.py @@@TARGET_URL@@@
+Wapiti_Unauth_____Wapiti_____wapiti @@@TARGET_URL@@@
 # COMMAND LINE ONLY = CANNOT INTERACT WITH THREAD and other settings ON THE FLY: 
-DirBusterNotInteractive_____DirBuster Not Interactive_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -H -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_GENERIC_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
-DirBusterInteractive_____DirBuster_____java -jar @@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_GENERIC_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+DirBusterNotInteractive_____DirBuster Not Interactive_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -H -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_GENERIC_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
+DirBusterInteractive_____DirBuster_____@@@TOOL_DIRBUSTER_BIN@@@ -u @@@TARGET_URL@@@ -t 20 -R -r '###PLUGIN_OUTPUT_DIR###' -l @@@DICT_GENERIC_DIRBUSTER@@@ | grep -v "^java." | tr "\t" " " | grep -v "^ at" # Remove java exception garbage at the end
 DirBuster_Extract_URLs_____Extract URLs_____@@@FRAMEWORK_DIR@@@/scripts/extract_urls.sh dirbuster DirBuster @@@TARGET_URL@@@
 W3AF_Unauth_____W3AF_____@@@FRAMEWORK_DIR@@@/scripts/run_w3af.sh @@@TOOL_W3AF_DIR@@@ @@@TARGET_URL@@@ @@@PROXY@@@ '@@@USER_AGENT_#@@@'
 W3AF_Unauth_____Extract URLs_____@@@FRAMEWORK_DIR@@@/scripts/extract_urls.sh w3af W3AF @@@TARGET_URL@@@
