--- src/hercules-payload/SOURCE/HERCULES.go	2017-03-07 15:16:58.826666661 -0500
+++ HERCULES.go	2017-03-07 15:24:10.300000000 -0500
@@ -129,20 +129,6 @@
 
 func CheckSetup()  (bool){
 
-  DirList, _ := exec.Command("sh", "-c", "cd $HERCULES_PATH/src && ls").Output()
-  GoVer, _ := exec.Command("sh", "-c", "go version").Output()
-  UPX, _ := exec.Command("sh", "-c", "upx").Output()
-
-  if (!(strings.Contains(string(DirList), "EGESPLOIT"))) {
-    return false
-  }
-
-  if !(strings.Contains(string(GoVer), "version")) {
-    return false
-  }
-  if !(strings.Contains(string(UPX), "Markus")) {
-    return false
-  }
   return true
 }
 
