diff -ruN dnswalk.orig/dnswalk dnswalk/dnswalk
--- dnswalk.orig/dnswalk	2008-10-17 15:38:25.000000000 +0200
+++ dnswalk/dnswalk	2008-10-17 15:38:36.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/contrib/bin/perl
+#!/usr/bin/perl
 #
 # dnswalk    Walk through a DNS tree, pulling out zone data and
 # dumping it in a directory tree
diff -ruN dnswalk.orig/makereports dnswalk/makereports
--- dnswalk.orig/makereports	2008-10-17 15:38:25.000000000 +0200
+++ dnswalk/makereports	2008-10-17 15:38:44.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/contrib/bin/perl
+#!/usr/bin/perl
 # This takes output from dnswalk and makes a "rep.orts" directory
 # with one file per contact.  Great for sending mail to all the admins.
 
