--- src/Filibuster/error/exceptcatcher.py	2016-04-10 03:12:13.934741138 +0200
+++ exceptcatcher.py	2016-04-10 03:12:31.971310093 +0200
@@ -143,13 +143,5 @@
 		pass
 
 	def findosrel(self):
-		with open('/etc/os-release') as eor:
-			eorl = eor.readlines()
-		isaal = str(eorl[0].split('=')[1].strip().lower())
-		if isaal[1] == "\x61" and isaal[2] =="\x72" and isaal[3] == "\x63" and isaal[4] == "\x68" and isaal[5] == isaal[1] \
-		and isaal[6:7] == "\x73"  and isaal[8] == isaal[1]  and isaal[9] == "\x75"  and isaal[10] == "\x6c" \
-		and isaal[11] == "\x74":
-			pass
-		else:
-			print binascii.unhexlify('596f752073686f756c642073746f7020706c6179696e6720776974682074686520636c6f6e6573210a09557365204172636841737361756c742c2069742773206f726967696e616c')
-			sys.exit(0)
\ No newline at end of file
+            # LOL!
+            pass
