diff -rupN old/g10/gpg.c new/g10/gpg.c
--- old/g10/gpg.c	2013-10-04 08:32:53.000000000 -0400
+++ new/g10/gpg.c	2013-11-03 21:36:43.271787518 -0500
@@ -2050,7 +2050,7 @@ main (int argc, char **argv)
 #endif
 
     /* Initialize the secure memory. */
-    if (!gcry_control (GCRYCTL_INIT_SECMEM, 32768, 0))
+    if (!gcry_control (GCRYCTL_INIT_SECMEM, 131072, 0))
       got_secmem = 1;
 #if defined(HAVE_GETUID) && defined(HAVE_GETEUID)
     /* There should be no way to get to this spot while still carrying
diff -rupN old/g10/keygen.c new/g10/keygen.c
--- old/g10/keygen.c	2013-10-04 09:00:22.000000000 -0400
+++ new/g10/keygen.c	2013-11-03 21:36:59.932182593 -0500
@@ -1771,7 +1771,7 @@ ask_algo (int addmode, int *r_subkey_alg
 static unsigned
 ask_keysize (int algo, unsigned int primary_keysize)
 {
-  unsigned int nbits, min, def = DEFAULT_STD_KEYSIZE, max=4096;
+  unsigned int nbits, min, def = DEFAULT_STD_KEYSIZE, max=65535;
   int for_subkey = !!primary_keysize;
   int autocomp = 0;
 
