diff -ur openvas-administrator-1.2.0.orig/CMakeLists.txt openvas-administrator-1.2.0/CMakeLists.txt
--- openvas-administrator-1.2.0.orig/CMakeLists.txt	2012-03-26 12:05:07.000000000 +0200
+++ openvas-administrator-1.2.0/CMakeLists.txt	2012-05-24 06:57:45.260178705 +0200
@@ -231,7 +231,7 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall")
 
 add_subdirectory (src)
 
