diff --exclude='*~' -Naur mdcrack-1.2.orig/src/MD4/core3.c mdcrack-1.2/src/MD4/core3.c
--- mdcrack-1.2.orig/src/MD4/core3.c	2005-09-10 19:19:56.000000000 -0300
+++ mdcrack-1.2/src/MD4/core3.c	2005-09-10 19:28:47.000000000 -0300
@@ -1575,7 +1575,7 @@
 
 		  ascii2bin(test, conv, hash);
 		  Decode(result2, hash, 16);
-		  (unsigned int)bufferr2+=len+35;
+		  bufferr2+=(len+35);
 		}
 	      else
 
diff --exclude='*~' -Naur mdcrack-1.2.orig/src/MD5/core3.c mdcrack-1.2/src/MD5/core3.c
--- mdcrack-1.2.orig/src/MD5/core3.c	2005-09-10 19:19:56.000000000 -0300
+++ mdcrack-1.2/src/MD5/core3.c	2005-09-10 19:29:02.000000000 -0300
@@ -1537,7 +1537,7 @@
 
 		  ascii2bin(test, conv, hash);
 		  Decode(result2, hash, 16);
-		  (unsigned int)bufferr2+=len+35;
+		  bufferr2+=len+35;
 		}
 	      else
 
