--- src/thcrut-1.2.5/src/discover_main.c	2003-05-29 08:07:51.000000000 +0200
+++ discover_main.c	2021-07-28 21:39:32.263334125 +0200
@@ -22,7 +22,6 @@
 extern struct _opt opt;
 extern char ip_tcp_sync[];
 
-struct sockaddr_in ip_tcp_sync_addr;
 int rawsox;
 
 #define DFL_HOSTS_PARALLEL	(5000)
@@ -374,7 +373,7 @@
 testsend(int myip)
 {
 	struct ip *ip = (struct ip *)(ip_tcp_sync);
-	
+
 	ip->ip_dst.s_addr = htonl(myip);
 	net_send(rawsox, ip_tcp_sync, 40);
 }
