--- obex.c.old	2013-11-12 19:17:49.739632684 -0500
+++ obex.c	2013-11-12 19:20:32.842969713 -0500
@@ -126,6 +126,44 @@
 #define OBEX_RSP_DATABASE_LOCKED        0x61
 #endif
 
+int OBEX_CharToUnicode(uint8_t *uc, const uint8_t *c, int size)
+{
+/*#ifdef HAVE_ICONV*/
+#if 0
+    iconv_t cd;
+    size_t c_len;
+    size_t uc_len;
+#endif
+    int len, n;
+    /*printf("\n");*/
+
+
+    len = n = strlen((char *) c);
+    /*obex_return_val_if_fail(n*2+2 <= size, -1);*/
+
+/*#ifdef HAVE_ICONV*/
+#if 0
+    setlocale(LC_CTYPE, "");
+    cd=iconv_open ("UCS-2BE", nl_langinfo(CODESET));
+    obex_return_val_if_fail(cd!=(iconv_t)-1, -1);
+    c_len = n;
+    uc_len = size;
+    iconv(cd, &c, &c_len, &uc, &uc_len);
+    obex_return_val_if_fail(iconv_close(cd)==0, -1);
+    obex_return_val_if_fail(c_len==0, -1);
+    obex_return_val_if_fail(uc_len>=2, -1);
+    memset(uc, 0, uc_len);
+    
+    return size-uc_len+2;
+#else
+/*#if 0*/
+    uc[n*2+1] = 0;
+    uc[n*2] = 0;
+
+    return (len*2)+2 ;
+#endif
+}
+
 static void obex_request_done(obex_t *handle, obex_object_t *object, int command, int response)
 {
 	struct obex_context *context;

