--- src/gspoof-3.2/configure.ac	2003-12-23 11:05:15.000000000 +0100
+++ configure.ac	2019-06-15 22:39:39.706666679 +0200
@@ -1,13 +1,13 @@
 #
 # This file is part of Gspoof-3 (a console/gtk+ tcp/ip packet forger)
-#  
+#
 #  $Name: configure.ac $
 #  $Version: 3.2 $
 #  $Date: 2003/12/22 16:30:03 $
 #  $Author: Embyte <embyte@madlab.it> $
 #  $Copyright: Copyright (C) 2002-2003 by embyte $
 #  $License: This software is under GPL version 2 of license $
-# 
+#
 
 AC_PREREQ(2.50)
 
@@ -31,7 +31,7 @@
 if test "$prefix" = "NONE"; then
    prefix="/usr/local"
    fi
-   
+
 
 #######################
 # Checks for compiler #
@@ -54,16 +54,8 @@
 	                   [search for libnet in DIR/include and DIR/lib]),
             [ LIBNET_PREFIX=$withval ],
             [ LIBNET_PREFIX=/usr ])
-	    
-AC_CHECK_FILE([$LIBNET_PREFIX/lib/libnet.a],
-              [LDFLAGS="-L$LIBNET_PREFIX/lib"
-	       CPPFLAGS="-I$LIBNET_PREFIX/include" ],
-              AC_MSG_ERROR([
-Libnet-1.1.1 Packet Shaping Library is required!
-Verify to have installed it and pass --with-libnet-prefix to configure.
-Or download it from http://www.packetfactory.net/libnet ])) 
-              
-AC_CHECK_LIB(net, libnet_name2addr4,, 
+
+AC_CHECK_LIB(net, libnet_name2addr4,,
 AC_MSG_ERROR([
 Libnet-1.1.1 Packet Shaping Library not found! It's required.
 Download it from http://www.packetfactory.net/libnet and install. ]))
@@ -80,12 +72,12 @@
       	   [
       	      AC_DEFINE(HAVE_GTK)
       	      ENABLE_GTK=yes
-	      GTKOBJS="interface.o callbacks.o gfuncts.o support.o gtk.o"	      
+	      GTKOBJS="interface.o callbacks.o gfuncts.o support.o gtk.o"
       	   ],
 	   [
 		AC_MSG_WARN([
-Library requirements (gtk+-2.0 >= 2.0.0 pango >= 1.0 atk >= 1.0) not met; 
-consider adjusting the PKG_CONFIG_PATH environment variable if your libraries 
+Library requirements (gtk+-2.0 >= 2.0.0 pango >= 1.0 atk >= 1.0) not met;
+consider adjusting the PKG_CONFIG_PATH environment variable if your libraries
 are in a nonstandard prefix so pkg-config can find them.
 Perhaps you want visit www.gtk.org and download them.
 
@@ -121,16 +113,16 @@
 # Print results #
 #################
 
-echo 
-echo "Ok. I have done with configuration." 
-echo 
+echo
+echo "Ok. I have done with configuration."
+echo
 echo "Results are:"
-echo 
+echo
 echo "PLATFORM ...... : `uname -mp`"
 echo "O.S. .......... : `uname -rs` (`uname -n`)"
 echo "COMPILER ...... : ${CC}"
 echo "CFLAGS ........ : -Wall -O2"
-echo "CPPFLAGS ...... : ${GTK_CFLAGS} ${CPPFLAGS}" 
+echo "CPPFLAGS ...... : ${GTK_CFLAGS} ${CPPFLAGS}"
 echo "LDFLAGS ....... : ${LDFLAGS}"
 echo "LIBS .......... : ${LIBS} ${GTK_LIBS}"
 echo "PREFIX ........ : ${prefix}"
