--- src/gspoof-3.2/gtk.c	2003-12-22 16:48:47.000000000 +0100
+++ gtk.c	2019-06-16 02:43:10.653333350 +0200
@@ -24,9 +24,7 @@
    gtk_set_locale ();
    gtk_init (&_argc, &_argv);
 
-   add_pixmap_directory ("pixmap");
-   add_pixmap_directory ("/usr/local/share/gspoof/pixmap");
-   add_pixmap_directory ("/usr/share/gspoof/pixmap");
+   add_pixmap_directory ("/usr/share/pixmaps");
 
    MainWin = create_MainWin();
 
