diff -ur openvas-libraries-4.0.7.orig/CMakeLists.txt openvas-libraries-4.0.7/CMakeLists.txt
--- openvas-libraries-4.0.7.orig/CMakeLists.txt	2012-03-26 09:43:34.000000000 +0200
+++ openvas-libraries-4.0.7/CMakeLists.txt	2012-04-12 22:03:42.023114984 +0200
@@ -240,7 +240,8 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror")
+# Disable -Werror since gnutls have deprecated function calls
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall")
 
 add_subdirectory (base)
 add_subdirectory (misc)
diff -ur openvas-libraries-4.0.7.orig/misc/CMakeLists.txt openvas-libraries-4.0.7/misc/CMakeLists.txt
--- openvas-libraries-4.0.7.orig/misc/CMakeLists.txt	2012-03-26 09:43:34.000000000 +0200
+++ openvas-libraries-4.0.7/misc/CMakeLists.txt	2012-04-12 22:03:54.686643951 +0200
@@ -70,7 +70,7 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror -fPIC")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -fPIC")
 
 if (MINGW)
   set (FILES openvas_auth.c openvas_server.c proctitle.c)
diff -ur openvas-libraries-4.0.7.orig/nasl/CMakeLists.txt openvas-libraries-4.0.7/nasl/CMakeLists.txt
--- openvas-libraries-4.0.7.orig/nasl/CMakeLists.txt	2012-03-26 09:43:34.000000000 +0200
+++ openvas-libraries-4.0.7/nasl/CMakeLists.txt	2012-04-12 22:04:14.876956371 +0200
@@ -39,7 +39,7 @@
 # The "-D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1" is necessary for GPGME!
 # The "-fno-strict-aliasing" silences warnings caused by macros defined in byteorder.h.
 # Once the warnings have been addressed this flag should be removed.
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1 -fno-strict-aliasing")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -D_FILE_OFFSET_BITS=64 -DLARGEFILE_SOURCE=1 -fno-strict-aliasing")
 
 # The "-Wno-error=unprototyped-calls" silences warnings produced by an
 # openSUSE-patched gcc caused by code in regex.c.
diff -ur openvas-libraries-4.0.7.orig/omp/CMakeLists.txt openvas-libraries-4.0.7/omp/CMakeLists.txt
--- openvas-libraries-4.0.7.orig/omp/CMakeLists.txt	2012-03-26 09:43:34.000000000 +0200
+++ openvas-libraries-4.0.7/omp/CMakeLists.txt	2012-04-12 22:04:04.630130704 +0200
@@ -27,7 +27,7 @@
 set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -Wl,-z,relro -Wl,-z,now")
 
 set (CMAKE_C_FLAGS_DEBUG        "${CMAKE_C_FLAGS_DEBUG}")
-set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall -Werror")
+set (CMAKE_C_FLAGS              "${CMAKE_C_FLAGS} ${HARDENING_FLAGS} -Wall")
 
 set (FILES xml.c omp.c)
 
