--- src/demjson-2.2.4/setup.py	2023-08-31 23:08:41.793332080 +0200
+++ setup.py	2023-08-31 23:09:48.276665405 +0200
@@ -36,7 +36,8 @@
 if py_major >= 3:
     # Make sure 2to3 gets run
     if distmech == 'setuptools':
-        py3extra['use_2to3'] = True
+        #py3extra['use_2to3'] = True
+        foo = True
         #py3extra['convert_2to3_doctests'] = ['src/your/module/README.txt']
         #py3extra['use_2to3_fixers'] = ['your.fixers']
     elif distmech == 'distutils':
