diff -urbN qwtplot3d/include/qwt3d_io_gl2ps.h qwtplot3d.patched/include/qwt3d_io_gl2ps.h
--- qwtplot3d/include/qwt3d_io_gl2ps.h	2005-07-10 16:54:52.000000000 +0300
+++ qwtplot3d.patched/include/qwt3d_io_gl2ps.h	2012-01-21 22:15:20.000000000 +0200
@@ -2,12 +2,7 @@
 #define qwt3d_io_gl2ps_h__2004_05_07_01_16_begin_guarded_code
 
 #include <time.h>
-
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
 
 #include "qwt3d_types.h"
 #include "qwt3d_io.h"
diff -urbN qwtplot3d/include/qwt3d_openglhelper.h qwtplot3d.patched/include/qwt3d_openglhelper.h
--- qwtplot3d/include/qwt3d_openglhelper.h	2005-07-19 17:40:28.000000000 +0300
+++ qwtplot3d.patched/include/qwt3d_openglhelper.h	2012-01-21 22:29:56.000000000 +0200
@@ -2,11 +2,8 @@
 #define __openglhelper_2003_06_06_15_49__
 
 #include "qglobal.h"
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
+#include <GL/glu.h>
 
 namespace Qwt3D
 {
