--- src/PRET/helper.py	2017-02-11 08:42:14.553333335 -0500
+++ helper.py	2017-02-11 08:42:46.313333321 -0500
@@ -122,7 +122,7 @@
     if msg: print(Back.RED + msg + info)
 
   # show printer and status
-  def discover(self, (ipaddr, (device, uptime, status, prstat))):
+  def discover(self, ipaddr, device, uptime, status, prstat):
     ipaddr = output().strfit(ipaddr, 15)
     device = output().strfit(device, 27)
     uptime = output().strfit(uptime,  8)
@@ -144,7 +144,7 @@
 
   # show directory listing
   def psdir(self, isdir, size, mtime, name, otime):
-    otime = Style.DIM + "(created " + otime + ")" + Style.RESET_ALL 
+    otime = Style.DIM + "(created " + otime + ")" + Style.RESET_ALL
     vol = Style.DIM + Fore.YELLOW + item(re.findall("^(%.*%)", name)) + Style.RESET_ALL
     name = re.sub("^(%.*%)", '', name) # remove volume information from filename
     name = Style.BRIGHT + Fore.BLUE + name + Style.RESET_ALL if isdir else name
@@ -179,7 +179,7 @@
       self.info("%-35s %-12s %-7s %-7s %-7s" % ((path, cmd) + opt))
 
   # show captured jobs
-  def joblist(self, (date, size, user, name, soft)):
+  def joblist(self, date, size, user, name, soft):
     user = output().strfit(user, 13)
     name = output().strfit(name, 22)
     soft = output().strfit(soft, 20)
@@ -274,7 +274,7 @@
     num = int(num)
     for unit in ['B','K','M']:
       if abs(num) < 1024.0:
-        return (("%4.1f%s" if unit == 'M' else "%4.0f%s") % (num, unit)) 
+        return (("%4.1f%s" if unit == 'M' else "%4.0f%s") % (num, unit))
       num /= 1024.0
 
   # remove carriage return from line breaks
